diff --git a/build.go b/build.go
index 3b6531f..90e245c 100644
--- a/build.go
+++ b/build.go
@@ -38,6 +38,7 @@ var (
 	goarch        string
 	goos          string
 	targetcc      string
+	gccgo         string
 	noupgrade     bool
 	version       string
 	goVersion     float64
@@ -346,6 +347,7 @@ func parseFlags() {
 	flag.StringVar(&goarch, "goarch", runtime.GOARCH, "GOARCH")
 	flag.StringVar(&goos, "goos", runtime.GOOS, "GOOS")
 	flag.StringVar(&targetcc, "targetcc", "", "CC")
+	flag.StringVar(&gccgo, "gccgo", "", "GCCGO")
 	flag.BoolVar(&noupgrade, "no-upgrade", noupgrade, "Disable upgrade functionality")
 	flag.StringVar(&version, "version", getVersion(), "Set compiled in version string")
 	flag.BoolVar(&race, "race", race, "Use race detector")
@@ -425,6 +427,9 @@ func install(target target, tags []string) {
 	if targetcc != "" {
 		os.Setenv("CC", targetcc)
 	}
+	if gccgo != "" {
+		os.Setenv("GCCGO", gccgo)
+	}
 
 	// On Windows generate a special file which the Go compiler will
 	// automatically use when generating Windows binaries to set things like
@@ -455,6 +460,9 @@ func build(target target, tags []string) {
 	if targetcc != "" {
 		os.Setenv("CC", targetcc)
 	}
+	if gccgo != "" {
+		os.Setenv("GCCGO", gccgo)
+	}
 
 	// On Windows generate a special file which the Go compiler will
 	// automatically use when generating Windows binaries to set things like
