--- bitlbee-1.2.5/configure.orig	2010-03-17 01:13:55.000000000 +0000
+++ bitlbee-1.2.5/configure	2010-03-17 15:10:39.456149997 +0000
@@ -115,7 +115,7 @@
 
 DESTDIR=
 LFLAGS=
-EFLAGS=
+EFLAGS=$LDFLAGS
 EOF
 
 cat<<EOF>config.h
@@ -305,7 +305,7 @@
 
 detect_resolv_static()
 {
-	for i in $systemlibdirs; do
+	for i in ; do
 		if [ -f $i/libresolv.a ]; then
 			echo "$RESOLV_TESTCODE" | $CC -o /dev/null -x c - -Wl,$i/libresolv.a >/dev/null 2>/dev/null
 			if [ "$?" = "0" ]; then
