--- Makefile.in.orig	2008-01-29 05:28:31.000000000 +0000
+++ Makefile.in	2008-06-04 23:13:59.000000000 +0000
@@ -59,7 +59,7 @@
 # Install commands...
 #
 
-INSTALL_BIN	=	$(LIBTOOL) $(INSTALL) -m 755 -s
+INSTALL_BIN	=	$(LIBTOOL) $(INSTALL) -m 755
 INSTALL_DATA	=	$(INSTALL) -m 644
 INSTALL_DIR	=	$(INSTALL) -d
 INSTALL_LIB	=	$(LIBTOOL) $(INSTALL) -m 755
@@ -88,7 +88,7 @@
 			mxml-node.o mxml-search.o mxml-set.o
 LIBOBJS		=	$(PUBLIBOBJS) mxml-private.o mxml-string.o
 OBJS		=	mxmldoc.o testmxml.o $(LIBOBJS)
-TARGETS		=	$(LIBMXML) mxmldoc testmxml mxml.xml doc/mxml.man
+TARGETS		=	$(LIBMXML) mxmldoc mxml.xml doc/mxml.man
 
 
 #
@@ -318,6 +318,7 @@
 
 testmxml.o:	mxml.h
 
+MXMLDOC_DIR=.
 
 #
 # mxml.xml
@@ -325,7 +326,7 @@
 
 mxml.xml:	mxmldoc-static mxml.h $(PUBLIBOBJS:.o=.c)
 	$(RM) mxml.xml
-	./mxmldoc-static --header doc/reference.heading mxml.xml mxml.h $(PUBLIBOBJS:.o=.c) >doc/reference.html
+	$(MXMLDOC_DIR)/mxmldoc-static --header doc/reference.heading mxml.xml mxml.h $(PUBLIBOBJS:.o=.c) >doc/reference.html
 
 valgrind:	mxmldoc-static
 	$(RM) valgrind.xml
@@ -340,7 +341,7 @@
 
 doc/mxml.man: mxmldoc-static mxml.xml
 	$(RM) doc/mxml.man
-	./mxmldoc-static --man mxml --title "Mini-XML API" \
+	$(MXMLDOC_DIR)/mxmldoc-static --man mxml --title "Mini-XML API" \
 		--intro doc/intro.man --footer doc/footer.man \
 		mxml.xml >doc/mxml.man
 
