--- wxbase/include/wx/filename.h.orig	2007-03-03 12:00:34.000000000 -0800
+++ wxbase/include/wx/filename.h	2007-03-03 12:01:43.000000000 -0800
@@ -390,7 +390,7 @@
 
     // get the canonical path separator for this format
     static wxChar GetPathSeparator(wxPathFormat format = wxPATH_NATIVE)
-        { return GetPathSeparators(format)[0u]; }
+        { return GetPathSeparators(format)[(int) 0u]; }
 
     // is the char a path separator for this format?
     static bool IsPathSeparator(wxChar ch, wxPathFormat format = wxPATH_NATIVE);
