--- cherokee-0.4.25-orig/cherokee/Makefile.in	2005-08-02 09:24:46.000000000 -0700
+++ cherokee/cherokee/Makefile.in	2005-08-13 20:53:26.980411168 -0700
@@ -331,11 +331,11 @@
 am_cherokee_admin_OBJECTS = cherokee_admin.$(OBJEXT)
 cherokee_admin_OBJECTS = $(am_cherokee_admin_OBJECTS)
 cherokee_admin_DEPENDENCIES = libcherokee-config.la \
-	libcherokee-base.la libcherokee-client.la
+	libcherokee-base.la libcherokee-client.la libcherokee-server.la
 am_cherokee_logrotate_OBJECTS = cherokee_logrotate.$(OBJEXT)
 cherokee_logrotate_OBJECTS = $(am_cherokee_logrotate_OBJECTS)
 cherokee_logrotate_DEPENDENCIES = libcherokee-config.la \
-	libcherokee-base.la libcherokee-client.la
+	libcherokee-base.la libcherokee-client.la libcherokee-server.la
 binSCRIPT_INSTALL = $(INSTALL_SCRIPT)
 SCRIPTS = $(bin_SCRIPTS)
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
@@ -915,7 +915,7 @@
 
 libplugin_admin_la_LDFLAGS = -module -export-dynamic -avoid-version  
 libplugin_admin_la_SOURCES = $(handler_admin)
-libplugin_admin_la_LIBADD = -lcherokee-config -lcherokee-client
+libplugin_admin_la_LIBADD = -lcherokee-config -lcherokee-client -lcherokee-server
 @STATIC_HANDLER_ADMIN_TRUE@static_handler_admin_src = $(handler_admin)
 @STATIC_HANDLER_ADMIN_TRUE@static_handler_admin_lib = -lcherokee-config
 @STATIC_HANDLER_ADMIN_FALSE@dynamic_handler_admin_lib = libplugin_admin.la
@@ -1322,9 +1322,9 @@
 
 cherokee_LDFLAGS = -export-dynamic
 cherokee_logrotate_SOURCES = cherokee_logrotate.c
-cherokee_logrotate_LDADD = libcherokee-config.la libcherokee-base.la libcherokee-client.la 
+cherokee_logrotate_LDADD = libcherokee-config.la libcherokee-base.la libcherokee-client.la libcherokee-server.la
 cherokee_admin_SOURCES = cherokee_admin.c 
-cherokee_admin_LDADD = libcherokee-config.la libcherokee-base.la libcherokee-client.la 
+cherokee_admin_LDADD = libcherokee-config.la libcherokee-base.la libcherokee-client.la libcherokee-server.la
 
 #noinst_PROGRAMS = cherokee_modules cherokee_base64 cherokee_headers cherokee_fastcgi
 
