--- a/qa/conf.py.pre
+++ b/qa/conf.py.pre
@@ -28,19 +28,22 @@
 PHPCGI_PATH       = "%phpcgi%"
 PHP_FCGI_PORT     = 1980
 
-PHP_DIRS          = ["/usr/lib/cgi-bin/",
+PHP_DIRS          = ["/opt/bin/",
+                     "/usr/lib/cgi-bin/",
                      "/usr/bin/",
                      "/usr/sbin/",
                      "/usr/local/bin/",
                      "/opt/csw/php5/bin/",
                      "/opt/local/bin"]
 
-PYTHON_DIRS       = ["/usr/bin",
+PYTHON_DIRS       = ["/opt/bin",
+                     "/usr/bin",
                      "/usr/local/bin",
                      "/opt/csw/bin",
                      "/opt/local/bin"]
 
-PHP_NAMES         = ["php5-cgi",
+PHP_NAMES         = ["php-fcgi",
+                     "php5-cgi",
                      "php-cgi",
                      "php4-cgi",
                      "php5",
