Index: cherokee/ChangeLog
===================================================================
--- cherokee/ChangeLog (revision 183)
+++ cherokee/ChangeLog (revision 184)
@@ -1,2 +1,11 @@
+2006-02-06  Alvaro Lopez Ortega  <alvaro@alobbs.com>
+
+	* qa/128-ValidMethod-common.py, qa/Makefile.am (EXTRA_DIST): Added
+	a new test case to cover the previous bug fix.
+
+	* cherokee/handler_common.c: valid_methods should be
+	http_all_methods.  This handler relays in other ones, so the final
+	handler should be the one which checks it.
+
 2006-02-03  Alvaro Lopez Ortega  <alvaro@alobbs.com>
 
Index: cherokee/cherokee/handler_common.c
===================================================================
--- cherokee/cherokee/handler_common.c (revision 182)
+++ cherokee/cherokee/handler_common.c (revision 184)
@@ -56,5 +56,5 @@
 	.module.type     = cherokee_handler,                /* type         */
 	.module.new_func = cherokee_handler_common_new,     /* new func     */
-	.valid_methods   = http_get | http_head             /* http methods */
+	.valid_methods   = http_all_methods                 /* http methods */
 };
 
