--- conflib.pl.orig	2007-01-16 20:21:40.000000000 +0200
+++ conflib.pl	2015-02-08 22:07:14.110008632 +0200
@@ -356,7 +356,7 @@
  return 0;
 }
 ';
- $test=RunGCCTest($cc,'c',$test,'');
+ $test="OK\n"; # RunGCCTest($cc,'c',$test,'');
  $test=~s/\W//g;
  if ($test ne "OK")
    {
@@ -397,7 +397,7 @@
  STD(cout) << "OK" << STD(endl);
  return 0;
 }';
- $test=RunGCCTest($cc,'cc',$test,$stdcxx);
+ $test="OK\n"; #RunGCCTest($cc,'cc',$test,$stdcxx);
  $test=~s/\W//g;
  if ($test eq "OK")
    {
@@ -478,7 +478,7 @@
    }
  foreach $i (@list)
    {
-    $res=RunGCCTest($i,'cc',$test,$stdcxx);
+    $res="OK\n"; #RunGCCTest($i,'cc',$test,$stdcxx);
     $res=~s/\W//g;
     if ($res eq "OK")
       {
@@ -600,6 +600,7 @@
 {
  my $ret;
 
+ $conf{'LDExtraDirs'}=$ENV{LDExtraDirs};
  $ret=@conf{'LDExtraDirs'};
  if ($ret)
    {
@@ -789,7 +790,7 @@
     $OSf='Linux';
     $Compf='';
     $stdcxx='-lstdc++';
-    $defaultCXX='g++';
+    $defaultCXX="$ENV{TARGET_CXX}";
     $supportDir='linux';
    }
  elsif ($os=~/FreeBSD/)
@@ -1347,7 +1348,7 @@
  return 0;
 }
 ';
- $test=RunGCCTest($GCC,'c',$test,'');
+ $test="$ENV{TARGET_ARCH}\n"; #RunGCCTest($GCC,'c',$test,'');
  $test=~s/\W//g;
  $CPU=$conf{'TV_CPU'}=$test;
  print "$test\n";
