--- config.pl.orig	2009-07-28 18:40:30.000000000 +0300
+++ config.pl	2015-02-08 22:29:32.813864499 +0200
@@ -545,14 +545,16 @@
 
  print 'Checking endianess: ';
 
- if (@conf{'TV_BIG_ENDIAN'} eq 'yes')
+ if ($ENV{'TV_BIG_ENDIAN'} eq 'yes')
    {
     print "big endian (cached)\n";
+    $conf{'TV_BIG_ENDIAN'}='yes';
     return;
    }
- if (@conf{'TV_BIG_ENDIAN'} eq 'no')
+ if ($ENV{'TV_BIG_ENDIAN'} eq 'no')
    {
     print "little endian (cached)\n";
+    $conf{'TV_BIG_ENDIAN'}='no';
     return;
    }
  $test='
@@ -862,6 +864,9 @@
  my ($result,$test);
 
  print 'Looking for ncurses library: ';
+ $conf{'ncurses'} = 'yes';
+ $conf{'NameCurses'} = 'ncurses';
+ $conf{'HAVE_DEFINE_KEY'} = 1;
  if (@conf{'ncurses'})
    {
     print "@conf{'ncurses'} (cached) OK\n";
@@ -1053,6 +1058,7 @@
  my $test;
 
  print 'Looking for pointer size: ';
+ $conf{'HAVE_64BITS_POINTERS'}=$ENV{'HAVE_64BITS_POINTERS'};
  if ($conf{'HAVE_64BITS_POINTERS'})
    {
     print ($conf{'HAVE_64BITS_POINTERS'} eq 'yes' ? '64' : '32')."bits (cached)\n";
@@ -1254,7 +1260,7 @@
     # Darwin: doesn't have ldconfig.
     if ($OSf ne 'Darwin')
       {
-       $rep.=($OSf eq 'FreeBSD') ? "\t-ldconfig -m \$(libdir)\n" : "\t-ldconfig\n";
+       # $rep.=($OSf eq 'FreeBSD') ? "\t-ldconfig -m \$(libdir)\n" : "\t-ldconfig\n";
       }
    }
  if ($internac)
