--- bitlbee/configure.orig	2008-03-02 17:10:16.000000000 +0000
+++ bitlbee/configure	2008-03-19 05:30:26.000000000 +0000
@@ -112,7 +112,7 @@
 
 DESTDIR=
 LFLAGS=
-EFLAGS=
+EFLAGS=$LDFLAGS
 EOF
 
 cat<<EOF>config.h
@@ -322,7 +322,7 @@
 
 echo 'SSL_CLIENT=ssl_'$ssl'.o' >> Makefile.settings
 
-for i in /lib /usr/lib /usr/local/lib; do
+for i in ; do
 	if [ -f $i/libresolv.a ]; then
 		echo '#define HAVE_RESOLV_A' >> config.h
 		echo 'EFLAGS+='$i'/libresolv.a' >> Makefile.settings
