--- common/cpu.c.orig	2009-02-14 21:45:09.000000000 +0000
+++ common/cpu.c	2010-02-15 02:05:36.735976708 +0000
@@ -285,10 +285,7 @@
 #if !defined(HAVE_PTHREAD)
     return 1;
 
-#elif defined(_WIN32)
-    return pthread_num_processors_np();
-
-#elif defined(SYS_LINUX)
+#elif defined(SYS_LINUX) && defined(__i686__)
     unsigned int bit;
     int np;
     cpu_set_t p_aff;
@@ -298,25 +295,6 @@
         np += (((uint8_t *)&p_aff)[bit / 8] >> (bit % 8)) & 1;
     return np;
 
-#elif defined(SYS_BEOS)
-    system_info info;
-    get_system_info( &info );
-    return info.cpu_count;
-
-#elif defined(SYS_MACOSX) || defined(SYS_FREEBSD) || defined(SYS_OPENBSD)
-    int numberOfCPUs;
-    size_t length = sizeof( numberOfCPUs );
-#ifdef SYS_OPENBSD
-    int mib[2] = { CTL_HW, HW_NCPU };
-    if( sysctl(mib, 2, &numberOfCPUs, &length, NULL, 0) )
-#else
-    if( sysctlbyname("hw.ncpu", &numberOfCPUs, &length, NULL, 0) )
-#endif
-    {
-        numberOfCPUs = 1;
-    }
-    return numberOfCPUs;
-
 #else
     return 1;
 #endif
