--- cherokee/cherokee/util.h.orig	2008-09-21 08:24:28.000000000 +0000
+++ cherokee/cherokee/util.h	2008-09-29 16:27:35.910060024 +0000
@@ -125,10 +125,10 @@
 
 /* IO vectors
  */
-ret_t cherokee_iovec_skip_sent (struct iovec orig[], uint16_t  orig_len,
-				struct iovec dest[], uint16_t *dest_len,
+ret_t cherokee_iovec_skip_sent (struct iovec *orig, uint16_t  orig_len,
+				struct iovec *dest, uint16_t *dest_len,
 				size_t sent);
-ret_t cherokee_iovec_was_sent  (struct iovec orig[], uint16_t orig_len, size_t sent);
+ret_t cherokee_iovec_was_sent  (struct iovec *orig, uint16_t orig_len, size_t sent);
 
 /* Debug
  */
