--- cherokee-0.11.3/qa/conf.py.orig	2008-11-26 14:09:33.000000000 +0000
+++ cherokee-0.11.3/qa/conf.py	2008-12-03 22:09:08.786164401 +0000
@@ -30,16 +30,19 @@
 PHPCGI_PATH       = "auto"
 PHP_FCGI_PORT     = 1980
 
-PHP_DIRS          = ["/usr/lib/cgi-bin/",
+PHP_DIRS          = ["/opt/bin/",
+		     "/usr/lib/cgi-bin/",
                      "/usr/bin/",
                      "/usr/local/bin/",
                      "/opt/csw/php5/bin/"]

-PYTHON_DIRS       = ["/usr/bin",
+PYTHON_DIRS       = ["/opt/bin",
+		     "/usr/bin",
                      "/usr/local/bin",
                      "/opt/csw/bin"]

-PHP_NAMES         = ["php5-cgi",
+PHP_NAMES         = ["php-fcgi",
+		     "php5-cgi",
                      "php-cgi",
                      "php4-cgi"
                      "php5",
