--- cherokee/cherokee/server.c.orig	2006-12-13 21:26:05.000000000 -0800
+++ cherokee/cherokee/server.c	2006-12-13 21:29:20.000000000 -0800
@@ -1093,6 +1093,7 @@
 {
 	time_t       newtime;
 	static long *this_timezone = NULL;
+        int z;
 
 	/* Read the time
 	 */
@@ -1111,6 +1112,7 @@
 	 */
 	if (unlikely (this_timezone == NULL))
 		this_timezone = cherokee_get_timezone_ref();
+        z = - (*this_timezone / 60);
 		
 	cherokee_buffer_clean  (srv->bogo_now_string);
 	cherokee_buffer_add_va (srv->bogo_now_string,
@@ -1122,8 +1124,8 @@
 				srv->bogo_now_tm.tm_hour,
 				srv->bogo_now_tm.tm_min,
 				srv->bogo_now_tm.tm_sec,
-				srv->bogo_now_tm.tm_gmtoff < 0 ? '-' : '+',
-				abs(srv->bogo_now_tm.tm_gmtoff / 3600));
+				z < 0 ? '-' : '+',
+				(z / 60));
 	
 	CHEROKEE_RWLOCK_UNLOCK (&srv->bogo_now_mutex);      /* 2.- release */
 }
