*** coreutils/src/Makefile.in.orig	2005-01-07 12:10:37.000000000 +0100
--- coreutils/src/Makefile.in	2005-01-07 12:11:14.000000000 +0100
***************
*** 39,45 ****
  POST_UNINSTALL = :
  host_triplet = @host@
  EXTRA_PROGRAMS = chroot$(EXEEXT) df$(EXEEXT) hostid$(EXEEXT) \
! 	nice$(EXEEXT) pinky$(EXEEXT) stty$(EXEEXT) su$(EXEEXT) \
  	uname$(EXEEXT) uptime$(EXEEXT) users$(EXEEXT) who$(EXEEXT)
  bin_PROGRAMS = [$(EXEEXT) chgrp$(EXEEXT) chown$(EXEEXT) chmod$(EXEEXT) \
  	cp$(EXEEXT) dd$(EXEEXT) dircolors$(EXEEXT) du$(EXEEXT) \
--- 39,45 ----
  POST_UNINSTALL = :
  host_triplet = @host@
  EXTRA_PROGRAMS = chroot$(EXEEXT) df$(EXEEXT) hostid$(EXEEXT) \
! 	nice$(EXEEXT) pinky$(EXEEXT) stty$(EXEEXT) \
  	uname$(EXEEXT) uptime$(EXEEXT) users$(EXEEXT) who$(EXEEXT)
  bin_PROGRAMS = [$(EXEEXT) chgrp$(EXEEXT) chown$(EXEEXT) chmod$(EXEEXT) \
  	cp$(EXEEXT) dd$(EXEEXT) dircolors$(EXEEXT) du$(EXEEXT) \
***************
*** 60,65 ****
--- 60,66 ----
  	kill$(EXEEXT) logname$(EXEEXT) pathchk$(EXEEXT) \
  	printenv$(EXEEXT) printf$(EXEEXT) pwd$(EXEEXT) seq$(EXEEXT) \
  	sleep$(EXEEXT) tee$(EXEEXT) test$(EXEEXT) true$(EXEEXT) \
+ 	su$(EXEEXT) \
  	tty$(EXEEXT) whoami$(EXEEXT) yes$(EXEEXT) $(am__EXEEXT_1) \
  	$(am__EXEEXT_2)
  noinst_PROGRAMS = setuidgid$(EXEEXT)
