--- src/utils/parse.cc.orig	2007-04-06 11:39:44.000000000 -0700
+++ src/utils/parse.cc	2007-04-06 11:42:23.000000000 -0700
@@ -45,7 +45,7 @@
 
 const char*
 parse_skip_wspace(const char* first, const char* last) {
-  while (first != last && std::iswspace(*first))
+  while (first != last && ::iswspace(*first))
     first++;
 
   return first;
@@ -67,7 +67,7 @@
         return ++first;
 
     } else {
-      if (parse_is_seperator(*first) || std::iswspace(*first))
+      if (parse_is_seperator(*first) || ::iswspace(*first))
         return first;
     }
         
