diff -ur sqlite-2.8.16.dist/Makefile.in sqlite-2.8.16/Makefile.in
--- sqlite-2.8.16.dist/Makefile.in	2005-02-13 18:22:04.000000000 -0600
+++ sqlite-2.8.16/Makefile.in	2005-03-16 23:12:40.000000000 -0600
@@ -20,7 +20,7 @@
 # C Compiler and options for use in building executables that
 # will run on the platform that is doing the build.
 #
-BCC = @BUILD_CC@ @BUILD_CFLAGS@
+BCC = gcc
 
 # C Compile and options for use in building executables that 
 # will run on the target platform.  (BCC and TCC are usually the
@@ -38,7 +38,7 @@
 
 # libtool compile/link/install
 LTCOMPILE = $(LIBTOOL) --mode=compile $(TCC)
-LTLINK = $(LIBTOOL) --mode=link $(TCC)
+LTLINK = $(LIBTOOL) --mode=link $(TCC) @LDFLAGS@
 LTINSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 # Compiler options needed for programs that use the TCL library.
diff -ur sqlite-2.8.16.dist/configure sqlite-2.8.16/configure
--- sqlite-2.8.16.dist/configure	2005-02-13 18:22:04.000000000 -0600
+++ sqlite-2.8.16/configure	2005-03-09 22:05:53.000000000 -0600
@@ -19150,11 +19150,11 @@
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
-  if test "$cross_compiling" = "yes"; then
-    { { echo "$as_me:$LINENO: error: unable to find a compiler for building build tools" >&5
-echo "$as_me: error: unable to find a compiler for building build tools" >&2;}
-   { (exit 1); exit 1; }; }
-  fi
+#  if test "$cross_compiling" = "yes"; then
+#    { { echo "$as_me:$LINENO: error: unable to find a compiler for building build tools" >&5
+#echo "$as_me: error: unable to find a compiler for building build tools" >&2;}
+#   { (exit 1); exit 1; }; }
+#  fi
   BUILD_CC=$CC
   default_build_cflags=$CFLAGS
 else
@@ -19988,11 +19988,11 @@
 echo $ECHO_N "checking for $dir/include/tcl.h... $ECHO_C" >&6
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
+#else
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
 if test -r "$dir/include/tcl.h"; then
   eval "$as_ac_File=yes"
 else
@@ -20391,11 +20391,11 @@
 echo $ECHO_N "checking for $dir/include/readline.h... $ECHO_C" >&6
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
+#else
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
 if test -r "$dir/include/readline.h"; then
   eval "$as_ac_File=yes"
 else
@@ -20417,11 +20417,11 @@
 echo $ECHO_N "checking for $dir/include/readline/readline.h... $ECHO_C" >&6
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
+#else
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
 if test -r "$dir/include/readline/readline.h"; then
   eval "$as_ac_File=yes"
 else
