--- ctcs.old/ctcs	2007-04-25 18:11:04.000000000 -0300
+++ ctcs/ctcs	2007-04-25 18:09:13.000000000 -0300
@@ -119,8 +119,6 @@
 }
 
 socket(Server, PF_INET, SOCK_STREAM, $proto)	|| die "socket: $!";
-setsockopt(Server, SOL_SOCKET, SO_REUSEADDR,
-	pack("l", 1)) 	|| die "setsockopt: $!";
 bind(Server, sockaddr_in($port, INADDR_ANY))	|| die "bind: $!";
 listen(Server,SOMAXCONN) 				|| die "listen: $!";
 
