--- src/main.c~	2007-03-30 22:40:23.000000000 +0930
+++ src/main.c	2007-08-04 20:37:31.000000000 +0930
@@ -26,6 +26,8 @@
 #include <getopt.h>
 #include <usb.h>
 #include <errno.h>
+#include <byteswap.h>
+#include <endian.h>
 
 #include "dfu.h"
 #include "usb_dfu.h"
@@ -638,7 +640,12 @@
 			"descriptor: %s\n", usb_strerror());
 		exit(1);
 	}
-	/* FIXME: Endian! */
+
+	/* Correct the endianess of the transfer size before using it */
+#if __BYTE_ORDER == __BIG_ENDIAN
+	func_dfu.wTransferSize = bswap_16(func_dfu.wTransferSize);
+#endif
+
 	if (!transfer_size)
 		transfer_size = func_dfu.wTransferSize;
 
