From c62fb4a103968661968b81ebee9215c87f475ab7 Mon Sep 17 00:00:00 2001
From: Martin Pitt <mpitt@debian.org>
Date: Tue, 9 Aug 2016 18:11:41 +0200
Subject: Do not mess with the permissions of cupsd.conf.

Last-Update: 2015-02-10

Patch-Name: confdirperms.patch
---
 scheduler/conf.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scheduler/conf.c b/scheduler/conf.c
index ab98e1c7d..cd79a1324 100644
--- a/scheduler/conf.c
+++ b/scheduler/conf.c
@@ -1134,10 +1134,12 @@ cupsdReadConfiguration(void)
 			     Group, 1, 1) < 0 ||
        cupsdCheckPermissions(ServerRoot, "ssl", 0700, RunUser,
 			     Group, 1, 0) < 0 ||
+       /* Never alter permissions of central conffile
        cupsdCheckPermissions(ConfigurationFile, NULL, ConfigFilePerm, RunUser,
 			     Group, 0, 0) < 0 ||
        cupsdCheckPermissions(CupsFilesFile, NULL, ConfigFilePerm, RunUser,
 			     Group, 0, 0) < 0 ||
+       */
        cupsdCheckPermissions(ServerRoot, "classes.conf", 0600, RunUser,
 			     Group, 0, 0) < 0 ||
        cupsdCheckPermissions(ServerRoot, "printers.conf", 0600, RunUser,
