Description: Drop dangling references from cups-lpd.man
Author: Bastien ROUCARIÈS <bastien.roucaries@u-cergy.fr>
Bug-Debian: http://bugs.debian.org/570157
Last-Update: 2015-10-02
--- a/man/cups-lpd.man.in
+++ b/man/cups-lpd.man.in
@@ -111,9 +111,7 @@
 .B cups-lpd
 service using the corresponding control program.
 .SH SEE ALSO
-.BR cups (1),
 .BR cupsd (8),
-.BR inetconv (1m),
 .BR inetd (8),
 .BR launchd (8),
 .BR xinetd (8),
