Description: Do not write VCS tags into installed conffiles
Bug: http://www.cups.org/str.php?L2369
Origin: vendor
Author: Kenshi Muto <kmuto@debian.org>
Author: Didier Raboud <odyx@debian.org>
Last-Update: 2014-01-16

--- a/conf/Makefile
+++ b/conf/Makefile
@@ -75,8 +75,12 @@
 			$(INSTALL_CONFIG) -g $(CUPS_GROUP) $$file $(SERVERROOT)/$$file.N ; \
 		else \
 			$(INSTALL_CONFIG) -g $(CUPS_GROUP) $$file $(SERVERROOT) ; \
+			grep -v "\$$Id[:\$$]" $(SERVERROOT)/$$file > $(SERVERROOT)/$$file.n ; \
+			mv $(SERVERROOT)/$$file.n $(SERVERROOT)/$$file ; \
 		fi ; \
 		$(INSTALL_CONFIG) -g $(CUPS_GROUP) $$file $(DATADIR)/$$file.default; \
+		grep -v "\$$Id[:\$$]" $(DATADIR)/$$file.default > $(DATADIR)/$$file.default.n ; \
+		mv $(DATADIR)/$$file.default.n $(DATADIR)/$$file.default ; \
 	done
 	$(INSTALL_DIR) -m 755 $(DATADIR)/mime
 	for file in $(REPLACE); do \
