diff -Bur dropbear-0.43/chansession.c dropbear/chansession.c
--- dropbear-0.44test4/svr-chansession.c	2004-07-16 19:14:40.000000000 +0930
+++ dropbear/svr-chansession.c	2004-09-05 15:30:38.000000000 +0930
@@ -862,6 +862,7 @@
 	addnewvar("LOGNAME", ses.authstate.pw->pw_name);
 	addnewvar("HOME", ses.authstate.pw->pw_dir);
 	addnewvar("SHELL", usershell);
+	addnewvar("PATH", "/bin:/sbin:/usr/bin:/usr/sbin:/opt/bin:/opt/sbin");
 	if (chansess->term != NULL) {
 		addnewvar("TERM", chansess->term);
 	}
