--- a/libc/sysdeps/linux/common/bits/socket.h
+++ b/libc/sysdeps/linux/common/bits/socket.h
@@ -17,8 +17,9 @@
    License along with the GNU C Library; if not, see
    <http://www.gnu.org/licenses/>.  */
 
-#ifndef __BITS_SOCKET_H
+#if !(defined (__BITS_SOCKET_H) || defined (_LINUX_SOCKET_H))
 #define __BITS_SOCKET_H
+#define _LINUX_SOCKET_H
 
 #ifndef _SYS_SOCKET_H
 # error "Never include <bits/socket.h> directly; use <sys/socket.h> instead."
