--- a/ioctl-scan/scan/linux_fs.c
+++ b/ioctl-scan/scan/linux_fs.c
@@ -16,6 +16,8 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <sys/types.h>
+
 #include <libexplain/ac/linux/fs.h>
 #include <libexplain/ac/sys/ioctl.h>
 
--- a/libexplain/ac/linux/cyclades.h
+++ b/libexplain/ac/linux/cyclades.h
@@ -24,6 +24,7 @@
 #include <libexplain/ac/sys/types.h>
 
 #ifdef HAVE_LINUX_CYCLADES_H
+#include <time.h>
 #include <linux/cyclades.h>
 #endif
 
--- a/libexplain/ac/linux/msdos_fs.h
+++ b/libexplain/ac/linux/msdos_fs.h
@@ -24,6 +24,7 @@
 #include <libexplain/ac/sys/types.h>
 
 #ifdef HAVE_LINUX_MSDOS_FS_H
+#include <sys/types.h>
 #include <linux/msdos_fs.h>
 #endif
 
--- a/libexplain/ac/ustat.h
+++ b/libexplain/ac/ustat.h
@@ -19,6 +19,7 @@
 
 #ifndef LIBEXPLAIN_AC_USTAT_H
 #define LIBEXPLAIN_AC_USTAT_H
+#include <sys/stat.h>
 
 /**
   * @file
--- a/libexplain/buffer/enfile.c
+++ b/libexplain/buffer/enfile.c
@@ -17,6 +17,8 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <stdint.h>
+
 #include <libexplain/ac/errno.h>
 #include <libexplain/ac/string.h>
 #include <libexplain/ac/unistd.h>
--- a/libexplain/buffer/errno/fseeko.h
+++ b/libexplain/buffer/errno/fseeko.h
@@ -18,6 +18,7 @@
 
 #ifndef LIBEXPLAIN_BUFFER_ERRNO_FSEEKO_H
 #define LIBEXPLAIN_BUFFER_ERRNO_FSEEKO_H
+#include <sys/stat.h>
 
 #include <libexplain/ac/stdio.h>
 
--- a/libexplain/buffer/loff_t.h
+++ b/libexplain/buffer/loff_t.h
@@ -18,6 +18,7 @@
 
 #ifndef LIBEXPLAIN_BUFFER_LOFF_T_H
 #define LIBEXPLAIN_BUFFER_LOFF_T_H
+#include <sys/types.h>
 
 #include <libexplain/ac/sys/types.h>
 
--- a/libexplain/buffer/ssize_t.h
+++ b/libexplain/buffer/ssize_t.h
@@ -18,6 +18,7 @@
 
 #ifndef LIBEXPLAIN_BUFFER_SSIZE_T_H
 #define LIBEXPLAIN_BUFFER_SSIZE_T_H
+#include <sys/types.h>
 
 #include <libexplain/ac/sys/types.h>
 
--- a/libexplain/fileinfo.h
+++ b/libexplain/fileinfo.h
@@ -19,6 +19,7 @@
 
 #ifndef LIBEXPLAIN_FILEINFO_H
 #define LIBEXPLAIN_FILEINFO_H
+#include <sys/types.h>
 
 #include <libexplain/ac/stddef.h>
 
--- a/libexplain/iocontrol/blkbszget.c
+++ b/libexplain/iocontrol/blkbszget.c
@@ -17,6 +17,8 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <sys/types.h>
+
 #include <libexplain/ac/linux/fs.h>
 #include <libexplain/ac/sys/ioctl.h>
 
--- a/libexplain/iocontrol/blkgetsize64.c
+++ b/libexplain/iocontrol/blkgetsize64.c
@@ -17,6 +17,8 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <sys/types.h>
+
 #include <libexplain/ac/linux/fs.h>
 #include <libexplain/ac/stdint.h>
 #include <libexplain/ac/sys/ioctl.h>
--- a/libexplain/string_buffer.h
+++ b/libexplain/string_buffer.h
@@ -19,6 +19,13 @@
 
 #ifndef LIBEXPLAIN_STRING_BUFFER_H
 #define LIBEXPLAIN_STRING_BUFFER_H
+#include <sys/stat.h>
+#if (defined AF_ROSE) && (!defined HAVE_NETROSE_ROSE_H)
+#undef AF_ROSE
+#endif
+#if (defined AF_ASH) && (!defined HAVE_NETASH_ASH_H)
+#undef AF_ASH
+#endif
 
 #include <libexplain/ac/stdarg.h>
 #include <libexplain/ac/stddef.h>
--- a/libexplain/ustat.h
+++ b/libexplain/ustat.h
@@ -18,6 +18,7 @@
 
 #ifndef LIBEXPLAIN_USTAT_H
 #define LIBEXPLAIN_USTAT_H
+#include <sys/stat.h>
 
 /**
   * @file
