--- git-1.7.0/Makefile.orig	2010-02-13 00:44:43.000000000 +0000
+++ git-1.7.0/Makefile	2010-02-14 18:20:21.717309511 +0000
@@ -236,7 +236,7 @@
 
 CFLAGS = -g -O2 -Wall
 LDFLAGS =
-ALL_CFLAGS = $(CFLAGS)
+ALL_CFLAGS = $(CFLAGS) $(CPPFLAGS)
 ALL_LDFLAGS = $(LDFLAGS)
 STRIP ?= strip
 
@@ -1483,7 +1483,7 @@
 	$(QUIET_GEN)$(RM) $@ $@+ && \
 	INSTLIBDIR=`MAKEFLAGS= $(MAKE) -C perl -s --no-print-directory instlibdir` && \
 	sed -e '1{' \
-	    -e '	s|#!.*perl|#!$(PERL_PATH_SQ)|' \
+	    -e '	s|#!.*perl|#!/opt/bin/perl|' \
 	    -e '	h' \
 	    -e '	s=.*=use lib (split(/$(pathsep)/, $$ENV{GITPERLLIB} || "@@INSTLIBDIR@@"));=' \
 	    -e '	H' \
@@ -1526,7 +1526,7 @@
 	    -e '/@@GITWEB_CSS@@/d' \
 	    -e '/@@GITWEB_JS@@/r gitweb/gitweb.js' \
 	    -e '/@@GITWEB_JS@@/d' \
-	    -e 's|@@PERL@@|$(PERL_PATH_SQ)|g' \
+	    -e 's|@@PERL@@|/opt/bin/perl|g' \
 	    $@.sh > $@+ && \
 	chmod +x $@+ && \
 	mv $@+ $@
@@ -1830,9 +1830,7 @@
 		cp "$$bindir/git$X" "$$execdir/git$X"; } ; } && \
 	{ for p in $(BUILT_INS); do \
 		$(RM) "$$execdir/$$p" && \
-		ln "$$execdir/git$X" "$$execdir/$$p" 2>/dev/null || \
-		ln -s "git$X" "$$execdir/$$p" 2>/dev/null || \
-		cp "$$execdir/git$X" "$$execdir/$$p" || exit; \
+		ln -s "git$X" "$$execdir/$$p" 2>/dev/null || exit; \
 	  done; } && \
 	{ for p in $(REMOTE_CURL_ALIASES); do \
 		$(RM) "$$execdir/$$p" && \
