--- configure.in.orig	2008-06-04 19:08:33.000000000 +0000
+++ configure.in	2008-07-10 02:33:51.000000000 +0000
@@ -283,17 +283,7 @@
 AC_CHECK_FUNCS(wprintf)
 
 AC_MSG_CHECKING([for sizeof(unsigned long)])
-AC_RUN_IFELSE(
-[AC_LANG_PROGRAM(
-[],
-[[
-    if (sizeof(unsigned long) == 8)
-        return 0;
-
-    return 1;
-]])],
-[ulong_is_64_bit="yes"],
-[ulong_is_64_bit="no"])
+ulong_is_64_bit="no"
 if test "x$ulong_is_64_bit" = "xyes"; then
     AC_MSG_RESULT([64 bits])
     AC_DEFINE([ULONGIS64BIT], [], [If unsigned long is 64 bits])
@@ -417,7 +407,7 @@
 # that versions < 0.9 do not accumulate packet statistics whereas >= 0.9 do accumulate.
 # This is Linux only.  The check is done after pcre because the code below uses pcre.
 # It seems Phil Wood's pcap does not accumulate - 0.9x
-pcap_version_check="yes"
+pcap_version_check="no"
 if test "x$linux" = "xyes"; then
     if test "x$pcap_version_check" = "xyes"; then
         AC_MSG_CHECKING([for libpcap version >= 0.9])
