--- configure.in.orig	2009-09-14 13:01:19.000000000 -0700
+++ configure.in	2009-09-24 11:04:52.000000000 -0700
@@ -229,21 +229,7 @@
 AC_CHECK_TYPES([int8_t,int16_t,int32_t,int64_t])
 
 # In case INADDR_NONE is not defined (like on Solaris)
-have_inaddr_none="no"
-AC_MSG_CHECKING([for INADDR_NONE])
-AC_RUN_IFELSE(
-[AC_LANG_PROGRAM(
-[[
-#include <sys/types.h>
-#include <netinet/in.h>
-#include <arpa/inet.h>
-]],
-[[
-	if (inet_addr("10,5,2") == INADDR_NONE);
-    return 0;
-]])],
-[have_inaddr_none="yes"],
-[have_inaddr_none="no"])
+have_inaddr_none="yes"
 AC_MSG_RESULT($have_inaddr_none)
 if test "x$have_inaddr_none" = "xno"; then
 	AC_DEFINE([INADDR_NONE],[-1],[For INADDR_NONE definition])
@@ -349,7 +335,7 @@
 # that versions < 0.9 do not accumulate packet statistics whereas >= 0.9 do accumulate.
 # This is Linux only.  The check is done after pcre because the code below uses pcre.
 # It seems Phil Wood's pcap does not accumulate - 0.9x
-pcap_version_check="yes"
+pcap_version_check="no"
 if test "x$linux" = "xyes"; then
     if test "x$pcap_version_check" = "xyes"; then
         AC_MSG_CHECKING([for libpcap version >= 0.9])
