--- src/cfgparser.h	2006-11-12 13:40:35.000000000 +0100
+++ src/cfgparser.h	2007-01-12 09:03:16.000000000 +0100
@@ -30,8 +30,8 @@
 
 #define USHARE_CONFIG_FILE "ushare.conf"
 #define DEFAULT_USHARE_NAME "uShare"
-#define DEFAULT_USHARE_IFACE "eth0"
-#define DEFAULT_USHARE_LOGFILE "/var/log/ushare.log"
+#define DEFAULT_USHARE_IFACE "ixp0"
+#define DEFAULT_USHARE_LOGFILE "/opt/var/log/ushare.log"
 
 int parse_config_file (struct ushare_t *ut)
     __attribute__ ((nonnull));
