diff -ur ImageMagick-6.3.1/ltdl/ltdl.c imagemagick/ltdl/ltdl.c
--- ImageMagick-6.3.1/ltdl/ltdl.c	2006-07-07 17:59:56.000000000 +0200
+++ imagemagick/ltdl/ltdl.c	2006-12-28 15:29:23.000000000 +0100
@@ -25,6 +25,15 @@
 
 */
 
+#if HAVE_ERRNO_H
+#  include <errno.h>
+#endif
+
+#undef ENOMEM
+#define ENOMEM 12
+
+
+
 #if HAVE_CONFIG_H
 #  include <magick/magick-config.h>
 #endif
@@ -37,6 +46,7 @@
 #  include <stdio.h>
 #endif
 
+
 /* Include the header defining malloc.  On K&R C compilers,
    that's <malloc.h>, on ANSI C and ISO C compilers, that's <stdlib.h>.  */
 #if HAVE_STDLIB_H
@@ -63,9 +73,8 @@
 #  include <memory.h>
 #endif
 
-#if HAVE_ERRNO_H
-#  include <errno.h>
-#endif
+
+
 
 
 #ifndef __WINDOWS__
@@ -126,6 +135,7 @@
 #  define assert(arg)	((void) 0)
 #endif
 
+
 #include "ltdl.h"
 
 #if WITH_DMALLOC
diff -ur ImageMagick-6.3.1/magick/studio.h imagemagick/magick/studio.h
--- ImageMagick-6.3.1/magick/studio.h	2006-04-03 17:09:38.000000000 +0200
+++ imagemagick/magick/studio.h	2006-12-28 15:13:27.000000000 +0100
@@ -39,6 +39,9 @@
 #  endif
 #endif
 
+#include <errno.h>
+  
+  
 #if !defined(vms) && !defined(macintosh) && !defined(__WINDOWS__)
 # define POSIX
 #endif
@@ -156,7 +159,6 @@
 #include <string.h>
 #include <ctype.h>
 #include <locale.h>
-#include <errno.h>
 #include <fcntl.h>
 #include <math.h>
 #include <time.h>
--- ImageMagick-6.3.1/wand/studio.h	2006-04-08 07:18:15.000000000 +0200
+++ imagemagick/wand/studio.h	2006-12-28 15:49:15.000000000 +0100
@@ -39,6 +39,8 @@
 #  endif
 #endif
 
+#include <errno.h>
+
 #if !defined(vms) && !defined(macintosh) && !defined(__WINDOWS__)
 # define POSIX
 #endif
@@ -156,7 +158,6 @@
 #include <string.h>
 #include <ctype.h>
 #include <locale.h>
-#include <errno.h>
 #include <fcntl.h>
 #include <math.h>
 #include <time.h>
