--- xmail-1.25/xmail.orig	2008-01-03 23:27:45.000000000 +0000
+++ xmail-1.25/xmail	2008-09-07 22:48:21.105617102 +0000
@@ -9,17 +9,17 @@
 # Version:	@(#)skeleton  1.8  03-Mar-1998  miquels@cistron.nl
 #
 
-XMAIL_ROOT=/var/MailRoot
-XMAIL_CMD_LINE=""
+XMAIL_ROOT=/opt/var/MailRoot
+XMAIL_CMD_LINE="-SX 1 -Qn 1 -Yt 1 -Ln 1 -PX 1 -CX 1"
 PATH=$XMAIL_ROOT/bin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
-DAEMON=$XMAIL_ROOT/bin/XMail
+DAEMON=/opt/bin/XMail
 NAME=XMail
 DESC="XMail Server"
 
 test -f $DAEMON || exit 0
 
 set -e
-ulimit -c 20000
+ulimit -c 10000
 ulimit -s 128
 
 start_xmail()
