diff -Naur asterisk-1.4.0.orig/configs/agents.conf.sample asterisk-1.4.0/configs/agents.conf.sample
--- asterisk-1.4.0.orig/configs/agents.conf.sample	2006-12-24 01:34:32.000000000 -0500
+++ asterisk-1.4.0/configs/agents.conf.sample	2006-12-28 23:27:23.000000000 -0500
@@ -89,8 +89,8 @@
 ;urlprefix=http://localhost/calls/
 ;
 ; The optional directory to save the conversations in. The default is
-; /var/spool/asterisk/monitor
-;savecallsin=/var/calls
+; /opt/var/spool/asterisk/monitor
+;savecallsin=/opt/var/calls
 ;
 ; An optional custom beep sound file to play to always-connected agents. 
 ;custom_beep=beep
diff -Naur asterisk-1.4.0.orig/configs/festival.conf.sample asterisk-1.4.0/configs/festival.conf.sample
--- asterisk-1.4.0.orig/configs/festival.conf.sample	2006-12-24 01:34:32.000000000 -0500
+++ asterisk-1.4.0/configs/festival.conf.sample	2006-12-28 23:27:42.000000000 -0500
@@ -21,7 +21,7 @@
 ; THIS DIRECTORY *MUST* EXIST and must be writable from the asterisk process.
 ; Defaults to /tmp/
 ;
-;cachedir=/var/lib/asterisk/festivalcache/
+;cachedir=/opt/var/lib/asterisk/festivalcache/
 ;
 ; Festival command to send to the server.
 ; Defaults to: (tts_textasterisk "%s" 'file)(quit)\n
diff -Naur asterisk-1.4.0.orig/configs/iax.conf.sample asterisk-1.4.0/configs/iax.conf.sample
--- asterisk-1.4.0.orig/configs/iax.conf.sample	2006-12-24 01:34:32.000000000 -0500
+++ asterisk-1.4.0/configs/iax.conf.sample	2006-12-28 23:28:15.000000000 -0500
@@ -314,8 +314,8 @@
 ; authentication is used, "inkeys" is a list of acceptable public keys on the 
 ; local system that can be used to authenticate the remote peer, separated by
 ; the ":" character.  "outkey" is a single, private key to use to authenticate
-; to the other side.  Public keys are named /var/lib/asterisk/keys/<name>.pub
-; while private keys are named /var/lib/asterisk/keys/<name>.key.  Private
+; to the other side.  Public keys are named /opt/var/lib/asterisk/keys/<name>.pub
+; while private keys are named /opt/var/lib/asterisk/keys/<name>.key.  Private
 ; keys should always be 3DES encrypted.
 ;
 ;
diff -Naur asterisk-1.4.0.orig/configs/musiconhold.conf.sample asterisk-1.4.0/configs/musiconhold.conf.sample
--- asterisk-1.4.0.orig/configs/musiconhold.conf.sample	2006-12-24 01:34:32.000000000 -0500
+++ asterisk-1.4.0/configs/musiconhold.conf.sample	2006-12-28 23:24:52.000000000 -0500
@@ -34,11 +34,11 @@
 
 [default]
 mode=files
-directory=/var/lib/asterisk/moh
+directory=/opt/var/lib/asterisk/moh
 ;
 ;[native-random]
 ;mode=files
-;directory=/var/lib/asterisk/moh
+;directory=/opt/var/lib/asterisk/moh
 ;random=yes 	; Play the files in a random order
 
 
@@ -50,17 +50,17 @@
 ;mode=custom
 ; Note that with mode=custom, a directory is not required, such as when reading
 ; from a stream.
-;directory=/var/lib/asterisk/mohmp3
+;directory=/opt/var/lib/asterisk/mohmp3
 ;application=/usr/bin/mpg123 -q -r 8000 -f 8192 -b 2048 --mono -s
 
 ;[ulawstream]
 ;mode=custom
-;application=/usr/bin/streamplayer 192.168.100.52 888
+;application=/opt/sbin/streamplayer 192.168.100.52 888
 ;format=ulaw
 
 ; mpg123 on Solaris does not always exit properly; madplay may be a better
 ; choice
 ;[solaris]
 ;mode=custom
-;directory=/var/lib/asterisk/mohmp3
+;directory=/opt/var/lib/asterisk/mohmp3
 ;application=/site/sw/bin/madplay -Q -o raw:- --mono -R 8000 -a -12
diff -Naur asterisk-1.4.0.orig/configs/osp.conf.sample asterisk-1.4.0/configs/osp.conf.sample
--- asterisk-1.4.0.orig/configs/osp.conf.sample	2006-12-24 01:34:32.000000000 -0500
+++ asterisk-1.4.0/configs/osp.conf.sample	2006-12-28 23:28:46.000000000 -0500
@@ -25,7 +25,7 @@
 
 ;[default]
 ;
-; All paths are presumed to be under /var/lib/asterisk/keys unless
+; All paths are presumed to be under /opt/var/lib/asterisk/keys unless
 ; the path begins with '/'
 ;
 ; Specify the private keyfile.  If unspecified, defaults to the name
diff -Naur asterisk-1.4.0.orig/configs/queues.conf.sample asterisk-1.4.0/configs/queues.conf.sample
--- asterisk-1.4.0.orig/configs/queues.conf.sample	2006-12-24 01:34:32.000000000 -0500
+++ asterisk-1.4.0/configs/queues.conf.sample	2006-12-28 23:29:13.000000000 -0500
@@ -229,7 +229,7 @@
 ;
 ; You can specify a post recording command to be executed after the end of
 ; recording by calling
-;   Set(MONITOR_EXEC=mv /var/spool/asterisk/monitor/^{MONITOR_FILENAME} /tmp/^{MONITOR_FILENAME})
+;   Set(MONITOR_EXEC=mv /opt/var/spool/asterisk/monitor/^{MONITOR_FILENAME} /tmp/^{MONITOR_FILENAME})
 ;
 ; The command specified within the contents of MONITOR_EXEC will be executed when
 ; the recording is over. Any strings matching ^{X} will be unescaped to ${X} and
