diff -Naur asterisk-1.4.0.orig/codecs/gsm/Makefile asterisk-1.4.0/codecs/gsm/Makefile
--- asterisk-1.4.0.orig/codecs/gsm/Makefile	2006-12-24 01:34:32.000000000 -0500
+++ asterisk-1.4.0/codecs/gsm/Makefile	2006-12-24 02:21:40.000000000 -0500
@@ -38,7 +38,7 @@
 ######### probably require gcc. 
 
 ifeq (, $(findstring $(OSARCH) , Darwin SunOS ))
-ifeq (, $(findstring $(PROC) , x86_64 amd64 ultrasparc sparc64 arm ppc powerpc ppc64 ia64 s390 bfin mipsel ))
+ifeq (, $(findstring $(PROC) , x86_64 amd64 ultrasparc sparc64 arm armv5b ppc powerpc ppc64 ia64 s390 bfin mipsel ))
 ifeq (, $(findstring $(shell uname -m) , ppc ppc64 alpha armv4l s390 ))
 OPTIMIZE+=-march=$(PROC)
 endif
