--- gtk+-3.10.0/gtk/Makefile.in.orig	2013-09-23 20:48:07.000000000 +0300
+++ gtk+-3.10.0/gtk/Makefile.in	2015-03-08 23:22:44.431378058 +0200
@@ -1016,7 +1016,7 @@ INTROSPECTION_COMPILER_ARGS = \
 
 @USE_QUARTZ_FALSE@GTK_PRINT_PREVIEW_COMMAND = "evince --unlink-tempfile --preview --print-settings %s %f"
 @USE_QUARTZ_TRUE@GTK_PRINT_PREVIEW_COMMAND = "open -a /Applications/Preview.app %f"
-SUBDIRS = a11y native .
+SUBDIRS = a11y .
 @HAVE_CUPS_FALSE@@HAVE_PAPI_CUPS_FALSE@@HAVE_PAPI_FALSE@GTK_PRINT_BACKENDS = file,lpr
 @HAVE_CUPS_FALSE@@HAVE_PAPI_CUPS_FALSE@@HAVE_PAPI_TRUE@GTK_PRINT_BACKENDS = file,papi
 @HAVE_CUPS_TRUE@@HAVE_PAPI_CUPS_FALSE@GTK_PRINT_BACKENDS = file,cups
@@ -6143,13 +6143,6 @@ stamp-icons: $(STOCK_ICONS)
 	) done 								\
 	&& touch stamp-icons
 
-gtkbuiltincache.h: stamp-icons
-@CROSS_COMPILING_TRUE@@USE_EXTERNAL_ICON_CACHE_FALSE@	$(AM_V_at) $(MAKE) $(AM_MAKEFLAGS) -C native/
-@CROSS_COMPILING_FALSE@@USE_EXTERNAL_ICON_CACHE_FALSE@	$(AM_V_at) $(MAKE) $(AM_MAKEFLAGS) gtk-update-icon-cache$(EXEEXT)
-	$(AM_V_GEN) $(gtk_update_icon_cache_program) --quiet --force --ignore-theme-index		\
-	   --source builtin_icons stock-icons > gtkbuiltincache.h.tmp &&        \
-	mv gtkbuiltincache.h.tmp gtkbuiltincache.h
-
 @GSETTINGS_RULES@
 
 install-data-local:
