--- a/quiz/datfiles/pres
+++ b/quiz/datfiles/pres
@@ -79,5 +79,7 @@
 {J{ames} }{D{anforth|an} }Quayle:{W{illiam|m} }{J{efferson} }Clinton
 {W{illiam|m} }{J{efferson} }Clinton:1993-2001:\
 {A{lbert} }{A{rnold} }Gore{{,} J{unio}r}:{G{eorge} }{W{alker} }Bush{{,} J{unio}r}
-{G{eorge} }{W{alker} }Bush{{,} J{unio}r}:2001-:\
-{R{ichard} }{B{ruce} }Cheney:
+{G{eorge} }{W{alker} }Bush{{,} J{unio}r}:2001-{{20}0}9:\
+{R{ichard} }{B{ruce} }Cheney:{B{arack} }{H{ussein} }Obama{ II}
+{B{arack} }{H{ussein} }Obama{ II}:2009-:\
+{J{oe{seph}} }{R{obinette} }{Biden}{{,} J{unio}r}:
