--- vsftpd/sysdeputil.c.orig	2008-02-01 13:31:23.000000000 -0800
+++ vsftpd/sysdeputil.c	2008-02-01 13:34:54.000000000 -0800
@@ -161,7 +161,6 @@
 #include <linux/capability.h>
 #include <errno.h>
 #include <syscall.h>
-_syscall2(int, capset, cap_user_header_t, header, const cap_user_data_t, data)
 /* Gross HACK to avoid warnings - linux headers overlap glibc headers */
 #undef __NFDBITS
 #undef __FDMASK
