diff -u ctorrent-dnh2.1.old/peer.cpp ctorrent-dnh2.1/peer.cpp
--- ctorrent-dnh2.1.old/peer.cpp	2006-08-16 02:02:00.000000000 -0300
+++ ctorrent-dnh2.1/peer.cpp	2006-09-08 12:30:59.000000000 -0300
@@ -526,12 +526,13 @@
 int btPeer::CancelPiece()
 {
   PSLICE ps = request_q.GetHead();
+  PSLICE aux;
   size_t idx;
   int cancel = 1;
   int retval;
 
   idx = ps->index;
-  for( ; ps; ps = ps->next){
+  while( ps ){
     if( ps->index != idx ) break;
     if( ps == request_q.NextSend() ) cancel = 0;
     if( cancel ){
@@ -544,7 +545,9 @@
         m_req_out = 0;
       }
     }
+    aux = ps->next;
     request_q.Remove(ps->index, ps->offset, ps->length);
+    ps = aux;
   }
   if( !m_req_out && g_next_dn == this ) g_next_dn = (btPeer *)0;
 
