--- makedepend.SH.orig	2008-01-31 21:02:32.000000000 -0800
+++ makedepend.SH	2008-01-31 21:03:13.000000000 -0800
@@ -166,7 +166,7 @@
 	    -e '/^#.*<stdin>/d' \
             -e '/^#.*<builtin>/d' \
             -e '/^#.*<built-in>/d' \
-            -e '/^#.*<command line>/d' \
+            -e '/^#.*<command[ -]line>/d' \
 	    -e '/^#.*"-"/d' \
 	    -e '/: file path prefix .* never used$/d' \
 	    -e 's#\.[0-9][0-9]*\.c#'"$file.c#" \
