--- hugs/libraries/Makefile.in~	2006-05-19 22:56:35.000000000 -0700
+++ hugs/libraries/Makefile.in	2006-05-19 22:56:55.000000000 -0700
@@ -93,7 +93,7 @@
 	rmdir $(BUILD_DIR)/packages/cpphs/programs
 
 $(BOOTLIB): hugsbase/Hugs/*.* $(MAKE_BOOT) $(INCLUDES) $(FP_INCLUDES)
-	CPPFLAGS='$(CPPFLAGS)' $(MAKE_BOOT)
+	CPPFLAGS='$(CPPFLAGS)' /opt/bin/bash $(MAKE_BOOT)
 	echo timestamp for bootlib >$@
 
 $(INCLUDES): $(HEADERS)
