diff -ruN 1/albumart.c 2/albumart.c
--- 1/albumart.c	2015-02-28 20:28:01.000000000 +0200
+++ 2/albumart.c	2015-02-28 20:37:07.000000000 +0200
@@ -31,6 +31,9 @@
 #include <errno.h>
 
 #include <jpeglib.h>
+#ifdef THUMBNAIL_CREATION
+#include <libffmpegthumbnailer/videothumbnailerc.h>
+#endif
 
 #include "upnpglobalvars.h"
 #include "albumart.h"
@@ -384,6 +387,50 @@
 	return NULL;
 }
 
+#ifdef THUMBNAIL_CREATION
+char *
+generate_thumbnail(const char * path)
+{
+	char *tfile = NULL;
+	video_thumbnailer *vt = NULL;
+	char cache_dir[MAXPATHLEN];
+
+	if( art_cache_exists(path, &tfile) )
+		return tfile;
+
+	if ( is_video(path) )
+	{
+
+		vt = video_thumbnailer_create();
+		if ( !vt )
+		{
+			free(tfile);
+			return 0;
+		}
+		vt->thumbnail_image_type = Jpeg;
+		vt->thumbnail_image_quality = runtime_vars.thumb_quality;
+		vt->thumbnail_size = runtime_vars.thumb_width;
+		vt->seek_percentage = 20;
+		vt->overlay_film_strip = (GETFLAG(THUMB_FILMSTRIP))?1:0;
+
+		DPRINTF(E_DEBUG, L_METADATA, "generating thumbnail: %s\n", path);
+
+		strncpyt(cache_dir, tfile, sizeof(cache_dir));
+		if ( !make_dir(dirname(cache_dir), S_IRWXU|S_IRGRP|S_IXGRP|S_IROTH|S_IXOTH) &&
+			!video_thumbnailer_generate_thumbnail_to_file(vt, path, tfile) )
+		{
+			video_thumbnailer_destroy(vt);
+			return tfile;
+		}
+
+		video_thumbnailer_destroy(vt);
+	}
+	free(tfile);
+
+	return 0;
+}
+#endif
+
 int64_t
 find_album_art(const char *path, uint8_t *image_data, int image_size)
 {
@@ -392,6 +439,18 @@
 
 	if( (image_size && (album_art = check_embedded_art(path, image_data, image_size))) ||
 	    (album_art = check_for_album_file(path)) )
+	if(image_size)
+		album_art = check_embedded_art(path, image_data, image_size);
+
+	if(!album_art)
+		album_art = check_for_album_file(path);
+
+#ifdef THUMBNAIL_CREATION
+	if(!album_art && GETFLAG(THUMB_MASK))
+		album_art = generate_thumbnail(path);
+#endif
+
+	if(album_art)		
 	{
 		ret = sql_get_int_field(db, "SELECT ID from ALBUM_ART where PATH = '%q'", album_art);
 		if( !ret )
diff -ruN 1/configure.ac 2/configure.ac
--- 1/configure.ac	2015-02-28 20:28:01.000000000 +0200
+++ 2/configure.ac	2015-02-28 20:39:16.000000000 +0200
@@ -619,7 +619,22 @@
         ]
 )
 
-
+AC_ARG_ENABLE(thumbnail,
+	[  --enable-thumbnail       enable video thumbnail generation using libffmpegthumbnailer],[
+	if test "$enableval" = "yes"; then
+		AC_DEFINE([THUMBNAIL_CREATION],[1],[Define to 1 if you want to enable video thumbnail generation])
+		PKG_CHECK_MODULES([LIBFFMPEGTHUMBNAILER], libffmpegthumbnailer, ,
+			AC_MSG_ERROR([Unable to find libffmpegthumbnailer]))
+		AC_SUBST([LIBFFMPEGTHUMBNAILER_CFLAGS])
+		AC_SUBST([LIBFFMPEGTHUMBNAILER_LIBS])
+        else
+                AC_MSG_RESULT([no])
+        fi
+        ],[
+                AC_MSG_RESULT([no])
+        ]
+)
+ 
 case "$target_os" in
 	darwin*)
 		;;
diff -ruN 1/inotify.c 2/inotify.c
--- 1/inotify.c	2015-02-28 20:28:01.000000000 +0200
+++ 2/inotify.c	2015-02-28 20:46:09.000000000 +0200
@@ -600,6 +600,16 @@
 		sql_exec(db, "DELETE from OBJECTS where DETAIL_ID = %lld", detailID);
 	}
 	snprintf(art_cache, sizeof(art_cache), "%s/art_cache%s", db_path, path);
+
+#ifdef THUMBNAIL_CREATION
+	/* Remove video thumbnails */
+	if ( is_video(path) )
+	{
+		char *vthumb = art_cache;
+		strcpy(strchr(vthumb, '\0')-4, ".jpg");
+	}
+#endif
+	
 	remove(art_cache);
 
 	return 0;
@@ -649,6 +659,11 @@
 	int length, i = 0;
 	char * esc_name = NULL;
 	struct stat st;
+
+#ifdef THUMBNAIL_CREATION
+	char renpath_buf[PATH_MAX];
+	int cookie = 0;
+#endif
         
 	pollfds[0].fd = inotify_init();
 	pollfds[0].events = POLLIN;
@@ -710,6 +725,16 @@
 				{
 					DPRINTF(E_DEBUG, L_INOTIFY,  "The directory %s was %s.\n",
 						path_buf, (event->mask & IN_MOVED_TO ? "moved here" : "created"));
+#ifdef THUMBNAIL_CREATION
+					/* We do not want to regenerate the thumbnails if e rename a directory.
+					   We should keep at least four cookies/olddir since IN_MOVED_FROM/IN_MOVED_TO may
+					   not arrive in sequence, but one should cover most cases */
+					if (event->cookie == cookie && event->mask & IN_MOVED_TO)
+					{
+						DPRINTF(E_DEBUG, L_INOTIFY, "Directory rename: %s -> %s \n", renpath_buf, path_buf);
+						rename_artcache_dir(renpath_buf, path_buf);
+					}
+#endif						
 					inotify_insert_directory(pollfds[0].fd, esc_name, path_buf);
 				}
 				else if ( (event->mask & (IN_CLOSE_WRITE|IN_MOVED_TO|IN_CREATE)) &&
@@ -741,7 +766,18 @@
 						(event->mask & IN_ISDIR ? "directory" : "file"),
 						path_buf, (event->mask & IN_MOVED_FROM ? "moved away" : "deleted"));
 					if ( event->mask & IN_ISDIR )
+#ifdef THUMBNAIL_CREATION
+					{
+						if ( event->mask & IN_MOVED_FROM )
+						{
+							strncpy(renpath_buf, path_buf, sizeof(renpath_buf));
+							cookie = event->cookie;
+						}
+#endif						
 						inotify_remove_directory(pollfds[0].fd, path_buf);
+#ifdef THUMBNAIL_CREATION
+					}
+#endif						
 					else
 						inotify_remove_file(path_buf);
 				}
diff -ruN 1/Makefile.am 2/Makefile.am
--- 1/Makefile.am	2015-02-28 20:28:01.000000000 +0200
+++ 2/Makefile.am	2015-02-28 20:33:01.000000000 +0200
@@ -49,7 +49,7 @@
 	@LIBEXIF_LIBS@ \
 	@LIBINTL@ \
 	@LIBICONV@ \
-	-lFLAC  $(flacoggflag) $(vorbisflag) \
+	-lFLAC  $(flacoggflag) $(vorbisflag) @LIBFFMPEGTHUMBNAILER_LIBS@ \
 	@opus_LIBS@
 
 minidlnad_LDFLAGS = @STATIC_LDFLAGS@
diff -ruN 1/minidlna.c 2/minidlna.c
--- 1/minidlna.c	2015-02-28 20:28:01.000000000 +0200
+++ 2/minidlna.c	2015-02-28 20:48:47.000000000 +0200
@@ -654,6 +654,11 @@
 	runtime_vars.max_connections = 50;
 	runtime_vars.root_container = NULL;
 	runtime_vars.ifaces[0] = NULL;
+	
+#ifdef THUMBNAIL_CREATION
+	runtime_vars.thumb_width = 160;
+	runtime_vars.thumb_quality = 8;
+#endif
 
 	/* read options file first since
 	 * command line arguments have final say */
@@ -857,6 +862,30 @@
 			if (strtobool(ary_options[i].value))
 				SETFLAG(MERGE_MEDIA_DIRS_MASK);
 			break;
+#ifdef THUMBNAIL_CREATION
+		case ENABLE_THUMB:
+			if( (strcmp(ary_options[i].value, "yes") == 0) || atoi(ary_options[i].value) )
+				SETFLAG(THUMB_MASK);
+		break;
+		case THUMB_WIDTH:
+			runtime_vars.thumb_width = atoi(ary_options[i].value);
+			if (runtime_vars.thumb_width < 120)
+				runtime_vars.thumb_width = 120;
+			if (runtime_vars.thumb_width > 480)
+				runtime_vars.thumb_width = 480;
+			break;
+		case THUMB_QUALITY:
+			runtime_vars.thumb_quality = atoi(ary_options[i].value);
+			if (runtime_vars.thumb_quality < 5)
+				runtime_vars.thumb_quality = 5;
+			if (runtime_vars.thumb_quality > 30)
+				runtime_vars.thumb_quality = 30;
+		break;
+		case ENABLE_THUMB_FILMSTRIP:
+			if( (strcmp(ary_options[i].value, "yes") == 0) || atoi(ary_options[i].value) )
+				SETFLAG(THUMB_FILMSTRIP);
+		break;
+#endif			
 #ifdef HAVE_LIBPNG
 		case ICON:
 			{
diff -ruN 1/minidlna.conf 2/minidlna.conf
--- 1/minidlna.conf	2015-02-28 20:28:01.000000000 +0200
+++ 2/minidlna.conf	2015-02-28 20:49:58.000000000 +0200
@@ -90,3 +90,15 @@
 # colors indicate that a checkerboard background, in those colors, should
 # be used.
 #bgcolor=666666,999999
+
+# Suport to Movie Thumbnail generation. To use this option, thumbnail generation must be enable at compile time.
+enable_thumbnail=yes
+
+# The width of the thumbnail image. Large images takes more time to generate.  To use this option, thumbnail generation must be enable at compile time.
+thumbnail_width=160
+
+# Thumbnail Image quality. To use this option, thumbnail generation must be enable at compile time.
+thumbnail_quality=8
+
+# Should the thumbnail have a film strip? To use this option, thumbnail generation must be enable at compile time.
+enable_thumbnail_filmstrip=yes
diff -ruN 1/minidlnatypes.h 2/minidlnatypes.h
--- 1/minidlnatypes.h	2015-02-28 20:28:01.000000000 +0200
+++ 2/minidlnatypes.h	2015-02-28 20:50:37.000000000 +0200
@@ -51,6 +51,10 @@
 	int max_connections;	/* max number of simultaneous conenctions */
 	const char *root_container;	/* root ObjectID (instead of "0") */
 	const char *ifaces[MAX_LAN_ADDR];	/* list of configured network interfaces */
+#ifdef THUMBNAIL_CREATION
+	int thumb_width;
+	int thumb_quality;
+#endif	
 };
 
 struct string_s {
diff -ruN 1/options.c 2/options.c
--- 1/options.c	2015-02-28 20:28:01.000000000 +0200
+++ 2/options.c	2015-02-28 20:51:54.000000000 +0200
@@ -64,7 +64,15 @@
 	{ USER_ACCOUNT, "user" },
 	{ FORCE_SORT_CRITERIA, "force_sort_criteria" },
 	{ MAX_CONNECTIONS, "max_connections" },
+#ifndef THUMBNAIL_CREATION	
 	{ MERGE_MEDIA_DIRS, "merge_media_dirs" }
+#else
+	{ MERGE_MEDIA_DIRS, "merge_media_dirs" },
+	{ ENABLE_THUMB, "enable_thumbnail" },
+	{ THUMB_WIDTH, "thumbnail_width" },
+	{ THUMB_QUALITY, "thumbnail_quality" },
+	{ ENABLE_THUMB_FILMSTRIP, "enable_thumbnail_filmstrip" }
+#endif	
 #ifdef HAVE_LIBPNG
 	,{ ICON, "icon" },
 	{ BGCOLOR, "bgcolor" }
diff -ruN 1/options.h 2/options.h
--- 1/options.h	2015-02-28 20:28:01.000000000 +0200
+++ 2/options.h	2015-02-28 20:52:52.000000000 +0200
@@ -57,7 +57,15 @@
 	USER_ACCOUNT,			/* user account to run as */
 	FORCE_SORT_CRITERIA,		/* force sorting by a given sort criteria */
 	MAX_CONNECTIONS,		/* maximum number of simultaneous connections */
+#ifndef THUMBNAIL_CREATION	
 	MERGE_MEDIA_DIRS		/* don't add an extra directory level when there are multiple media dirs */
+#else
+	MERGE_MEDIA_DIRS,		/* don't add an extra directory level when there are multiple media dirs */
+	ENABLE_THUMB,                   /* enable thumbnail generation */
+	THUMB_WIDTH,                    /* thunbnail image with */
+	THUMB_QUALITY,                  /* thumnail image quality */
+	ENABLE_THUMB_FILMSTRIP          /* film strip overlay */
+#endif	
 #ifdef HAVE_LIBPNG
 	,ICON,					/* user-supplied PNG icon */
 	BGCOLOR					/* user-supplied BG color */
diff -ruN 1/upnpglobalvars.h 2/upnpglobalvars.h
--- 1/upnpglobalvars.h	2015-02-28 20:28:01.000000000 +0200
+++ 2/upnpglobalvars.h	2015-02-28 20:56:11.000000000 +0200
@@ -191,6 +191,10 @@
 #define NO_PLAYLIST_MASK      0x0008
 #define SYSTEMD_MASK          0x0010
 #define MERGE_MEDIA_DIRS_MASK 0x0020
+#ifdef THUMBNAIL_CREATION
+#define THUMB_MASK            0x0040
+#define THUMB_FILMSTRIP       0x0080
+#endif
 
 #define SETFLAG(mask)	runtime_flags |= mask
 #define GETFLAG(mask)	(runtime_flags & mask)
diff -ruN 1/utils.c 2/utils.c
--- 1/utils.c	2015-02-28 20:28:01.000000000 +0200
+++ 2/utils.c	2015-02-28 20:57:13.000000000 +0200
@@ -516,3 +516,16 @@
 	return type;
 }
 
+#ifdef THUMBNAIL_CREATION
+int
+rename_artcache_dir(const char * oldpath, const char * newpath)
+{
+	char old_artcache[PATH_MAX];
+	char new_artcache[PATH_MAX];
+
+	snprintf(old_artcache, sizeof(old_artcache), "%s/art_cache%s", db_path, oldpath);
+	snprintf(new_artcache, sizeof(old_artcache), "%s/art_cache%s", db_path, newpath);
+
+	return rename(old_artcache, new_artcache);
+}
+#endif
diff -ruN 1/utils.h 2/utils.h
--- 1/utils.h	2015-02-28 20:28:01.000000000 +0200
+++ 2/utils.h	2015-02-28 20:58:06.000000000 +0200
@@ -95,5 +95,8 @@
 /* Others */
 int make_dir(char * path, mode_t mode);
 unsigned int DJBHash(uint8_t *data, int len);
+#ifdef THUMBNAIL_CREATION
+int rename_artcache_dir(const char * oldpath, const char * newpath);
+#endif
 
 #endif

