--- configure.in.orig	2010-04-17 09:24:28.000000000 +0000
+++ configure.in	2010-04-19 01:38:00.789581788 +0000
@@ -220,7 +220,7 @@
 #fi
 
 if test "x$enable_gtk" = "xyes" ; then
-    AM_PATH_GTK_2_0(2.4.0, LIBGTK_FOUND=1, LIBGTK_FOUND=0)
+    LIBGTK_FOUND=0
     if test "$LIBGTK_FOUND" = "0" ; then
         AC_MSG_WARN([
 *** Gtk library not found!
@@ -255,15 +255,7 @@
       LIBS="$LIBS $ICONV_LFLAGS"
    fi
    AC_MSG_CHECKING(for iconv usability in programs)
-   AC_TRY_RUN([
-        #include <iconv.h>
-        int main(int argc, char **argv) {
-            iconv_t conv = iconv_open("ISO8859-1", "UTF-8");
-            if (conv != (iconv_t) -1) {
-                return 0;
-            }
-            return 1;
-      }],iconv_found="yes")
+   iconv_found="yes"
   if test "x$iconv_found" = "xno" ; then
      AC_MSG_RESULT(no)
   else
