--- mzscheme-orig/src/Makefile.in	2004-12-19 00:38:25.000000000 -0800
+++ mzscheme/src/Makefile.in	2005-04-12 20:39:18.000000000 -0700
@@ -86,10 +86,10 @@
 	if [ -d $(srcdir)/CVS ] ; then cat $(srcdir)/mzscheme/imsg.txt ; fi
 
 copytree:
-	if [ ! -d $(prefix) ] ; then mkdir $(prefix) ; fi
+	if [ ! -d $(prefix) ] ; then mkdir -p $(prefix) ; fi
 	cp -p -r $(srcdir)/../collects $(prefix)/.
 	cp -p -r $(srcdir)/../include $(prefix)/.
 	cp -p -r $(srcdir)/../notes $(prefix)/.
-	cp -p -r $(srcdir)/../man $(prefix)/.
+	cp -p -r $(srcdir)/../man $(prefix)/../..
 	if [ -d $(srcdir)/../teachpack ] ; then  cp -r $(srcdir)/../teachpack $(prefix)/teachpack ; fi
 	cp -p $(srcdir)/../install $(prefix)/install
