--- ocaml/asmcomp/arm/emit.mlp.orig	2005-07-17 10:20:31.000000000 -0700
+++ ocaml/asmcomp/arm/emit.mlp	2005-07-17 10:20:57.000000000 -0700
@@ -645,12 +645,9 @@
 (* Beginning / end of an assembly file *)
 
 let begin_assembly() =
-  `trap_ptr     .req    r11\n`;
   `alloc_ptr    .req    r8\n`;
   `alloc_limit  .req    r9\n`;
-  `sp           .req    r13\n`;
-  `lr           .req    r14\n`;
-  `pc           .req    r15\n`;
+  `trap_ptr     .req    r11\n`;
   let lbl_begin = Compilenv.make_symbol (Some "data_begin") in
   `	.data\n`;
   `	.global	{emit_symbol lbl_begin}\n`;
