--- ocaml-3.11.0/byterun/Makefile.orig	2008-11-08 16:29:02.000000000 +0000
+++ ocaml-3.11.0/byterun/Makefile	2008-12-16 23:58:01.507102928 +0000
@@ -34,6 +34,8 @@
 ocamlrun$(EXE): libcamlrun.a prims.o
 	$(MKEXE) $(BYTECCLINKOPTS) -o ocamlrun$(EXE) \
 		  prims.o libcamlrun.a $(BYTECCLIBS)
+	mv $@ $@.target
+	cp ../../../host/builds/ocaml/byterun/$@ .
 
 ocamlrund$(EXE): libcamlrund.a prims.o
 	$(MKEXE) -g $(BYTECCLINKOPTS) -o ocamlrund$(EXE) \
--- ocaml/yacc/Makefile.orig	2007-02-07 06:49:42.000000000 -0800
+++ ocaml/yacc/Makefile	2008-04-01 20:56:33.000000000 -0700
@@ -26,6 +26,8 @@
 
 ocamlyacc$(EXE): $(OBJS)
 	$(CC) $(CFLAGS) $(CCLINKFLAGS) -o ocamlyacc $(OBJS)
+	mv $@ $@.target
+	cp ../../../host/builds/ocaml/yacc/$@ .
 
 version.h : ../VERSION
 	echo "#define OCAML_VERSION \"`head -1 ../VERSION`\"" >version.h
--- ocaml-3.11.0/Makefile.orig	2008-10-23 15:29:11.000000000 +0000
+++ ocaml-3.11.0/Makefile	2008-12-17 00:07:08.201676096 +0000
@@ -371,12 +371,12 @@
 	sed -e 's|%%LIBDIR%%|$(LIBDIR)|' \
             -e 's|%%BYTERUN%%|$(BINDIR)/ocamlrun|' \
             -e 's|%%CCOMPTYPE%%|cc|' \
-            -e 's|%%BYTECC%%|$(BYTECC) $(BYTECCCOMPOPTS) $(SHAREDCCCOMPOPTS)|' \
-            -e 's|%%NATIVECC%%|$(NATIVECC) $(NATIVECCCOMPOPTS)|' \
-            -e 's|%%PACKLD%%|$(PACKLD)|' \
+            -e 's|%%BYTECC%%|gcc $(BYTECCCOMPOPTS) $(SHAREDCCCOMPOPTS)|' \
+            -e 's|%%NATIVECC%%|gcc $(NATIVECCCOMPOPTS)|' \
+            -e 's|%%PACKLD%%|ld -r|' \
             -e 's|%%BYTECCLIBS%%|$(BYTECCLIBS)|' \
             -e 's|%%NATIVECCLIBS%%|$(NATIVECCLIBS)|' \
-            -e 's|%%RANLIBCMD%%|$(RANLIBCMD)|' \
+            -e 's|%%RANLIBCMD%%|ranlib|' \
             -e 's|%%CC_PROFILE%%|$(CC_PROFILE)|' \
             -e 's|%%ARCH%%|$(ARCH)|' \
             -e 's|%%MODEL%%|$(MODEL)|' \
@@ -387,9 +387,9 @@
             -e 's|%%EXT_DLL%%|.so|' \
             -e 's|%%SYSTHREAD_SUPPORT%%|$(SYSTHREAD_SUPPORT)|' \
             -e 's|%%ASM%%|$(ASM)|' \
-            -e 's|%%MKDLL%%|$(MKDLL)|' \
-            -e 's|%%MKEXE%%|$(MKEXE)|' \
-            -e 's|%%MKMAINDLL%%|$(MKMAINDLL)|' \
+            -e 's|%%MKDLL%%|gcc -shared|' \
+            -e 's|%%MKEXE%%|gcc|' \
+            -e 's|%%MKMAINDLL%%|gcc -shared|' \
             utils/config.mlp > utils/config.ml
 	@chmod -w utils/config.ml

--- ocaml/otherlibs/Makefile.shared.orig	2008-07-15 15:31:32.000000000 +0000
+++ ocaml/otherlibs/Makefile.shared	2008-12-17 00:32:37.828393820 +0000
@@ -46,6 +46,10 @@
 
 $(LIBNAME).cma: $(CAMLOBJS)
 	$(MKLIB) -o $(LIBNAME) -oc $(CLIBNAME) -ocamlc '$(CAMLC)' -linkall $(CAMLOBJS) $(LINKOPTS)
+ifeq ($(LIBNAME), $(filter unix str, $(LIBNAME)))
+	mv dll$(LIBNAME).so dll$(LIBNAME).so.target
+	cp ../../../../host/builds/ocaml/otherlibs/$(LIBNAME)/dll$(LIBNAME).so .
+endif
 
 $(LIBNAME).cmxa: $(CAMLOBJS_NAT)
 	$(MKLIB) -o $(LIBNAME) -oc $(CLIBNAME) -ocamlopt '$(CAMLOPT)' -linkall $(CAMLOBJS_NAT) $(LINKOPTS)
