--- src/inetaddr.c.orig	2007-02-18 23:02:27.000000000 -0800
+++ src/inetaddr.c	2007-02-18 23:05:28.000000000 -0800
@@ -1330,7 +1330,7 @@
 	      ia->sa.ss_family = (size == 4)? AF_INET : AF_INET6;
 	      GNET_INETADDR_SET_SS_LEN(ia);
 	      memcpy(GNET_INETADDR_ADDRP(ia), buf, size);
-	      GNET_INETADDR_PORT(ia) = g_htons(state->port);
+	      GNET_INETADDR_PORT_SET(ia, g_htons(state->port));
 
 	      state->ias = g_list_prepend (state->ias, ia);
 
