--- ipkg-utils-1.7/ipkg-build.old	2005-05-25 07:48:00.103866796 -0700
+++ ipkg-utils-1.7/ipkg-build	2005-05-25 07:49:15.083846628 -0700
@@ -243,8 +243,8 @@
 mkdir $tmp_dir
 
 echo $CONTROL > $tmp_dir/tarX
-( cd $pkg_dir && tar $ogargs -czf $tmp_dir/data.tar.gz . -X $tmp_dir/tarX )
-( cd $pkg_dir/$CONTROL && tar $ogargs -czf $tmp_dir/control.tar.gz . )
+( cd $pkg_dir && tar $ogargs -X $tmp_dir/tarX -czf $tmp_dir/data.tar.gz . ) || exit 1
+( cd $pkg_dir/$CONTROL && tar $ogargs -czf $tmp_dir/control.tar.gz . ) || exit 1
 rm $tmp_dir/tarX
 
 echo "2.0" > $tmp_dir/debian-binary
