--- radiusclient-ng-0.5.6/configure	2007-10-12 00:51:14.070091240 -0400
+++ radiusclient-ng/configure	2007-10-12 00:52:05.363293488 -0400
@@ -23332,8 +23332,7 @@ echo $ECHO_N "checking for field domainn
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+See \`config.log' for more details." >&2;} }
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
