--- a/mx/DateTime/mxDateTime/mxDateTime.c
+++ b/mx/DateTime/mxDateTime/mxDateTime.c
@@ -1753,7 +1753,7 @@ struct tm *mxDateTime_AsTmStruct(mxDateT
 	      mxDateTime_RangeError,
 	      "year out of range for tm struct conversion");
 
-    memset(tm,0,sizeof(tm));
+    memset(tm,0,sizeof(*tm));
     tm->tm_hour = (int)datetime->hour;
     tm->tm_min = (int)datetime->minute;
 #if ROUND_SECONDS_IN_TM_STRUCT
