--- aMule-2.1.3/src/libs/common/MuleDebug.cpp	2006-01-01 06:17:25.000000000 +0100
+++ amule/src/libs/common/MuleDebug.cpp	2007-01-05 21:36:00.000000000 +0100
@@ -33,13 +33,16 @@
 #include "StringFunctions.h"	// Needed for unicode2char
 
 #ifdef __LINUX__
-	#include <execinfo.h>
+#ifdef __UCLIBC__
+#define HAVE_BFD 0
+#else
+ 	#include <execinfo.h>
+#endif
 	#include <cxxabi.h>
 	#include <wx/thread.h>
 	#include <unistd.h> // Seems to be needed at least on Creteil's box
 #endif
 
-
 #if wxUSE_STACKWALKER && defined(__WXMSW__)
 	#include <wx/stackwalk.h>
 #elif HAVE_BFD
@@ -151,7 +154,7 @@
 	return walker.m_trace;
 }
 
-#elif defined(__LINUX__)
+#elif defined(__LINUX__) && !defined(__UCLIBC__)
 
 #if HAVE_BFD
 
