--- modutils-2.4.27/obj/obj_mips.c	2003-04-05 00:47:17.000000000 +0200
+++ modutils/obj/obj_mips.c	2006-12-28 14:28:26.000000000 +0100
@@ -244,7 +244,7 @@
   archdata_sec->header.sh_size = 0;
   sec = obj_find_section(f, "__dbe_table");
   if (sec) {
-    ad = (struct archdata *) (archdata_sec->contents) = xmalloc(sizeof(*ad));
+    ad = (archdata_sec->contents) =  (struct archdata *) xmalloc(sizeof(*ad));
     memset(ad, 0, sizeof(*ad));
     archdata_sec->header.sh_size = sizeof(*ad);
     ad->__start___dbe_table = sec->header.sh_addr;
