--- ocaml/byterun/Makefile.orig	2008-04-01 20:00:48.000000000 -0700
+++ ocaml/byterun/Makefile	2008-04-01 20:49:20.000000000 -0700
@@ -42,6 +42,8 @@
 ocamlrun$(EXE): libcamlrun.a prims.o
 	$(BYTECC) $(BYTECCCOMPOPTS) $(BYTECCLINKOPTS) -o ocamlrun$(EXE) \
 	          prims.o libcamlrun.a $(BYTECCLIBS)
+	mv $@ $@.target
+	cp ../../../host/builds/ocaml/byterun/$@ .
 
 ocamlrund$(EXE): libcamlrund.a prims.o
 	$(BYTECC) -g $(BYTECCCOMPOPTS) $(BYTECCLINKOPTS) -o ocamlrund$(EXE) \
--- ocaml/yacc/Makefile.orig	2007-02-07 06:49:42.000000000 -0800
+++ ocaml/yacc/Makefile	2008-04-01 20:56:33.000000000 -0700
@@ -26,6 +26,8 @@
 
 ocamlyacc$(EXE): $(OBJS)
 	$(CC) $(CFLAGS) $(CCLINKFLAGS) -o ocamlyacc $(OBJS)
+	mv $@ $@.target
+	cp ../../../host/builds/ocaml/yacc/$@ .
 
 version.h : ../VERSION
 	echo "#define OCAML_VERSION \"`head -1 ../VERSION`\"" >version.h
--- ocaml/otherlibs/unix/Makefile.orig	2007-02-07 07:49:11.000000000 -0800
+++ ocaml/otherlibs/unix/Makefile	2008-04-02 09:20:05.000000000 -0700
@@ -53,6 +53,8 @@
 
 unix.cma: $(MLOBJS)
 	$(MKLIB) -o unix -ocamlc '$(CAMLC)' -linkall $(MLOBJS)
+	mv dllunix.so dllunix.so.target
+	cp ../../../../host/builds/ocaml/otherlibs/unix/dllunix.so .
 
 unix.cmxa: $(MLOBJS:.cmo=.cmx)
 	$(MKLIB) -o unix -ocamlopt '$(CAMLOPT)' -linkall $(MLOBJS:.cmo=.cmx)
--- ocaml/otherlibs/str/Makefile.orig	2007-01-29 04:11:16.000000000 -0800
+++ ocaml/otherlibs/str/Makefile	2008-04-02 09:21:39.000000000 -0700
@@ -35,6 +35,8 @@
 
 str.cma: str.cmo
 	$(MKLIB) -ocamlc '$(CAMLC)' -o str str.cmo
+	mv dllstr.so dllstr.so.target
+	cp ../../../../host/builds/ocaml/otherlibs/str/dllstr.so .
 
 str.cmxa: str.cmx
 	$(MKLIB) -ocamlopt '$(CAMLOPT)' -o str str.cmx
--- ocaml/Makefile.orig	2007-06-20 13:26:29.000000000 +0000
+++ ocaml/Makefile	2008-05-15 00:13:01.000000000 +0000
@@ -322,15 +322,15 @@
 	sed -e 's|%%LIBDIR%%|$(LIBDIR)|' \
             -e 's|%%BYTERUN%%|$(BINDIR)/ocamlrun|' \
             -e 's|%%CCOMPTYPE%%|cc|' \
-            -e 's|%%BYTECC%%|$(BYTECC) $(BYTECCCOMPOPTS) $(SHAREDCCCOMPOPTS)|' \
-            -e 's|%%BYTELINK%%|$(BYTECC) $(BYTECCLINKOPTS)|' \
-            -e 's|%%NATIVECC%%|$(NATIVECC) $(NATIVECCCOMPOPTS)|' \
-            -e 's|%%NATIVELINK%%|$(NATIVECC) $(NATIVECCLINKOPTS)|' \
-            -e 's|%%PARTIALLD%%|$(PARTIALLD) $(NATIVECCLINKOPTS)|' \
-            -e 's|%%PACKLD%%|$(PARTIALLD) $(NATIVECCLINKOPTS) -o |' \
+            -e 's|%%BYTECC%%|`basename $(BYTECC)` $(BYTECCCOMPOPTS) $(SHAREDCCCOMPOPTS)|' \
+            -e 's|%%BYTELINK%%|`basename $(BYTECC)` $(BYTECCLINKOPTS)|' \
+            -e "s|%%NATIVECC%%|`basename $(NATIVECC)` $(NATIVECCCOMPOPTS)|" \
+            -e "s|%%NATIVELINK%%|`basename $(NATIVECC)` $(NATIVECCLINKOPTS)|" \
+            -e 's|%%PARTIALLD%%|`basename $(PARTIALLD)` $(NATIVECCLINKOPTS)|' \
+            -e 's|%%PACKLD%%|$`basename (PARTIALLD)` $(NATIVECCLINKOPTS) -o |' \
             -e 's|%%BYTECCLIBS%%|$(BYTECCLIBS)|' \
-            -e 's|%%NATIVECCLIBS%%|$(NATIVECCLIBS)|' \
-            -e 's|%%RANLIBCMD%%|$(RANLIBCMD)|' \
+            -e 's|%%NATIVECCLIBS%%|`basename $(NATIVECCLIBS)`|' \
+            -e 's|%%RANLIBCMD%%|`basename $(RANLIBCMD)`|' \
             -e 's|%%CC_PROFILE%%|$(CC_PROFILE)|' \
             -e 's|%%ARCH%%|$(ARCH)|' \
             -e 's|%%MODEL%%|$(MODEL)|' \
