diff -ur kissdx.conf patched/kissdx.conf
--- kissdx.conf	2006-11-24 22:54:04.250000000 +0100
+++ patched/kissdx.conf	2006-11-30 21:00:36.609375000 +0100
@@ -8,7 +8,7 @@
 # Path for persistent storage of info by kissdx.
 # The [Recently used] feature and the picture caching feature will store data here.
 # If option is omitted, kissdx will use picturepath, etc. for such storage.
-#persistentstoragepath	= /var/kissdx
+#persistentstoragepath	= /opt/var/kissdx
 
 # Allowed extensions for media files (comma separated, no spaces, case insensitive)
 # Omit a setting to send ALL files to the player in the corresponding directory listing.
