diff -ur kissdx.c patch/kissdx.c
--- kissdx.c	2006-11-24 01:36:19.906250000 +0100
+++ patch/kissdx.c	2006-11-24 01:28:10.046875000 +0100
@@ -221,12 +221,12 @@
 			snprintf(buffer, sizeof(buffer), "%s/.kissd.conf", home);
 			configpath = buffer;
 			if ((f = fopen(configpath, "r")) == NULL) {
-				configpath = "/etc/kissdx.conf";
+				configpath = "/opt/etc/kissdx.conf";
 				if ((f = fopen(configpath, "r")) == NULL) {
-					configpath = "/etc/kissd.conf";
+					configpath = "/opt/etc/kissd.conf";
 					if ((f = fopen(configpath, "r")) == NULL) {
 						log0("config file not found!");
-						log0("searched $HOME/ and /etc/ for kissdx.conf and kissd.conf");
+						log0("searched $HOME/ and /opt/etc/ for kissdx.conf and kissd.conf");
 						return 1;
 					}
 				}
