--- configure.ac.orig	2010-04-30 10:50:30.000000000 +0000
+++ configure.ac	2010-05-21 21:27:51.575538142 +0000
@@ -133,22 +133,9 @@
 ac_c_werror_flag=$_savedwerror_flag       
 
 AC_MSG_CHECKING([whether your getopt() correctly understands double colons in option string]) 
-AC_RUN_IFELSE(
-    [AC_LANG_PROGRAM( [[ const char optstring[] = "+:a::b";
-		         char  * myargv [] = {"progname", "-a", "-b"}; 
-		         int myargc = 3;
-			 extern char *optarg;
-		      ]] , [[
-		         int opt = getopt(myargc, myargv, optstring);
-		         if (opt != 'a' || optarg) /* optarg != NULL means that -b is seen as argument of -a */ 
-		           exit (-1);
-		        else
-		           exit(0); ]] )],
-    [AC_DEFINE(GETOPT_GROKS_OPTIONAL_ARGS, 1, Define if your getopt() correctly understands double colons in the option string)
+     AC_DEFINE(GETOPT_GROKS_OPTIONAL_ARGS, 1, Define if your getopt() correctly understands double colons in the option string)
      AC_SUBST(HAVE_OPTIONAL_ARGS,yes)
-     ac_double_colons="yes"],
-    [AC_SUBST(HAVE_OPTIONAL_ARGS,no)
-     ac_double_colons="no"])
+     ac_double_colons="yes"
 AC_MSG_RESULT($ac_double_colons)
 
 # Checks for typedefs, structures, and compiler characteristics.
@@ -326,10 +313,8 @@
 
 # check for existence of myself under /proc/$$/cwd
 echo "Will rlwrap find command's working directory under /proc/<commands pid>/cwd? let's see..."
-AC_CHECK_FILES(/proc/$$/cwd/configure.ac,[
                   AC_DEFINE(HAVE_PROC_PID_CWD,1 ,[Define to 1 if /proc/<pid>/cwd is a link to working directory of process <pid>])
                   AC_SUBST(HAVE_PROC_PID_CWD,[yes])
-               ],[AC_SUBST(HAVE_PROC_PID_CWD,[no])])
 
 
 echo
