--- monotone/configure.ac~	2005-12-31 06:10:31.000000000 +1300
+++ monotone/configure.ac	2006-05-19 12:43:42.000000000 +1200
@@ -497,7 +497,7 @@
 
   return EXIT_SUCCESS;
 }
-])])
+])],,,[true])
 AC_RUN_IFELSE([AC_LANG_SOURCE([
 /*
  * Reads the input file.
@@ -531,7 +531,7 @@
   else
     AC_MSG_RESULT(no)
   fi
-  rm -f _conftest.dat])
+  rm -f _conftest.dat],,[AC_MSG_RESULT(cross compiling - assume yes)])
 
 # allow compiling with different flags/optimisation
 # for monotone versus libs, useful for testing.
