--- vnstat-1.4/src/vnstat.h	2004-03-26 14:23:08.000000000 +0100
+++ vnstat/src/vnstat.h	2007-01-21 23:56:35.000000000 +0100
@@ -11,7 +11,7 @@
 #include <ctype.h>
 
 /* location of the database directory */
-#define DATABASEDIR "/var/lib/vnstat"
+#define DATABASEDIR "/opt/var/lib/vnstat"
 
 /* on which day should months change */
 #define MONTHROTATE "1"
@@ -34,7 +34,7 @@
 #define DEFQMODE "0"
 
 /* how much the boot time can variate between updates (seconds) */
-#define BVAR "15"
+#define BVAR "100"
 
 /* database version */
 /* 1 = 1.0, 2 = 1.1-1.2, 3 = 1.3*/
