--- vnstat-1.5/src/vnstat.h.orig	2007-12-03 19:30:48.000000000 +0100
+++ vnstat-1.5/src/vnstat.h	2007-12-20 14:43:29.000000000 +0100
@@ -22,7 +22,7 @@
 */
 
 /* location of the database directory */
-#define DATABASEDIR "/var/lib/vnstat"
+#define DATABASEDIR "/opt/var/lib/vnstat"
 
 /* on which day should months change */
 #define MONTHROTATE 1
@@ -64,7 +64,7 @@
 #define DEFQMODE 0
 
 /* how much the boot time can variate between updates (seconds) */
-#define BVAR 15
+#define BVAR 100
 
 /* use file locking by default */
 #define USEFLOCK 1
