--- w3m-0.5.1-orig/Makefile.in	2004-04-26 10:04:48.000000000 -0700
+++ w3m/Makefile.in	2005-04-06 09:07:04.000000000 -0700
@@ -168,8 +168,10 @@
 	./mktable$(EXT) 100 functable.tab > $@
 	-rm -f functable.tab
 
+ifeq ($(CROSS_COMPILATION), no)
 mktable$(EXT): mktable.o dummy.o $(ALIB)
 	$(CC) $(CFLAGS) -o mktable mktable.o dummy.o $(LDFLAGS) $(LIBS)
+endif
 
 $(BOOKMARKER): w3mbookmark.o dummy.o $(ALIB)
 	$(CC) $(CFLAGS) -o $(BOOKMARKER) w3mbookmark.o dummy.o $(LDFLAGS) $(LIBS)
--- w3m-0.5.1-orig/libwc/Makefile.in	2004-04-26 10:00:38.000000000 -0700
+++ w3m/libwc/Makefile.in	2005-04-06 09:29:02.000000000 -0700
@@ -6,7 +6,7 @@
 VPATH=$(srcdir):.
 LIBRARY = libwc.a
 CC=@CC@
-CFLAGS = $(OPTS) -I.. -I$(top_srcdir) -I$(srcdir) @CFLAGS@ @WCCFLAGS@ @DEFS@
+CFLAGS = $(OPTS) -I.. -I$(top_srcdir) -I$(srcdir) @CFLAGS@ @WCCFLAGS@ @DEFS@ @CPPFLAGS@
 AR=ar
 RANLIB=@RANLIB@
 RM=rm
