diff -Naur tftp-hpa.old/tftp-xinetd tftp-hpa/tftp-xinetd
--- tftp-hpa.old/tftp-xinetd	2005-06-10 16:21:00.000000000 +0200
+++ tftp-hpa/tftp-xinetd	2005-06-10 16:23:21.000000000 +0200
@@ -1,18 +1,18 @@
-# default: off
-# description: The tftp server serves files using the trivial file transfer \
-#	protocol.  The tftp protocol is often used to boot diskless \
-#	workstations, download configuration files to network-aware printers, \
-#	and to start the installation process for some operating systems.
+#
+# ftp://ftp.kernel.org/pub/software/network/tftp/
+#
 service tftp
 {
-	socket_type		= dgram
-	protocol		= udp
-	wait			= yes
-	user			= root
-	server			= /usr/sbin/in.tftpd
-	server_args		= -s /tftpboot
-	disable			= yes
-	per_source		= 11
-	cps			= 100 2
-	flags			= IPv4
+       flags            = REUSE
+       socket_type      = dgram
+       protocol         = udp
+       instances        = 30
+       wait             = yes
+       user             = root
+       server           = /opt/sbin/in.tftpd
+       server_args	= -s /opt/tftpboot
+       cps		= 100 2
+       log_on_success   = HOST PID
+       log_on_failure   = HOST
+       disable          = no
 }
