--- uClibc++/bin/Makefile	2005-12-11 01:36:58.000000000 +0100
+++ libuclibc++/bin/Makefile	2006-02-27 21:57:39.000000000 +0100
@@ -10,7 +10,12 @@
 
 install:
 	$(INSTALL) -d $(PREFIX)$(UCLIBCXX_RUNTIME_BINDIR)
-	$(INSTALL) -m 755 $(WRAPPER) $(PREFIX)$(UCLIBCXX_RUNTIME_BINDIR)
+	$(INSTALL) -d $(PREFIX)$(UCLIBCXX_RUNTIME_BINDIR)/nowrap
+	ln -sf ../mipsel-uclibc-gcc \
+		$(PREFIX)$(UCLIBCXX_RUNTIME_BINDIR)/nowrap/mipsel-uclibc-g++
+	$(INSTALL) -m 755 $(WRAPPER) $(PREFIX)$(UCLIBCXX_RUNTIME_BINDIR)/mipsel-uclibc-g++
+	sed -i 's|mipsel-uclibc-g++|nowrap/mipsel-uclibc-g++|'\
+		$(PREFIX)$(UCLIBCXX_RUNTIME_BINDIR)/mipsel-uclibc-g++
 
 $(WRAPPER):
 	echo "#!/bin/bash" > $(WRAPPER)
