--- buildroot.r15597/toolchain/uClibc/uclibc.mk	2006-07-14 01:15:53.000000000 +0200
+++ buildroot/toolchain/uClibc/uclibc.mk	2006-07-14 13:30:15.000000000 +0200
@@ -65,6 +65,7 @@
 	mkdir -p $(TOOL_BUILD_DIR)
 	bzcat $(DL_DIR)/$(UCLIBC_SOURCE) | tar -C $(TOOL_BUILD_DIR) $(TAR_OPTIONS) -
 	toolchain/patch-kernel.sh $(UCLIBC_DIR) toolchain/uClibc/ \*.patch
+	cat ../../sources/buildroot/uClibc-patches/*.patch | patch -d $(UCLIBC_DIR) -p1
 	touch $(UCLIBC_DIR)/.unpacked
 
 uclibc-configured: $(UCLIBC_DIR)/.configured
@@ -74,9 +75,9 @@
 		-e 's,# TARGET_$(UCLIBC_TARGET_ARCH) is not set,TARGET_$(UCLIBC_TARGET_ARCH)=y,g' \
 		-e 's,^TARGET_ARCH="none",TARGET_ARCH=\"$(UCLIBC_TARGET_ARCH)\",g' \
 		-e 's,^KERNEL_SOURCE=.*,KERNEL_SOURCE=\"$(LINUX_HEADERS_DIR)\",g' \
-		-e 's,^RUNTIME_PREFIX=.*,RUNTIME_PREFIX=\"/\",g' \
-		-e 's,^DEVEL_PREFIX=.*,DEVEL_PREFIX=\"/usr/\",g' \
-		-e 's,^SHARED_LIB_LOADER_PREFIX=.*,SHARED_LIB_LOADER_PREFIX=\"/lib\",g' \
+		-e 's,^RUNTIME_PREFIX=.*,RUNTIME_PREFIX=\"/opt/\",g' \
+		-e 's,^DEVEL_PREFIX=.*,DEVEL_PREFIX=\"/opt/\",g' \
+		-e 's,^SHARED_LIB_LOADER_PREFIX=.*,SHARED_LIB_LOADER_PREFIX=\"/opt/lib\",g' \
 		$(UCLIBC_DIR)/.config
 ifeq ($(UCLIBC_TARGET_ARCH),arm)
 	$(SED) 's,^.*CONFIG_$(shell echo $(BR2_ARM_TYPE)).*,CONFIG_$(shell echo $(BR2_ARM_TYPE))=y,g' \
@@ -150,25 +151,30 @@
 	touch -c $(STAGING_DIR)/lib/libc.a
 
 ifneq ($(TARGET_DIR),)
-$(TARGET_DIR)/lib/libc.so.0: $(STAGING_DIR)/lib/libc.a
+$(TARGET_DIR)/opt/lib/libc.so.0: $(STAGING_DIR)/lib/libc.a
 	$(MAKE1) -C $(UCLIBC_DIR) \
 		PREFIX=$(TARGET_DIR) \
-		DEVEL_PREFIX=/usr/ \
-		RUNTIME_PREFIX=/ \
+		DEVEL_PREFIX=/opt/ \
+		RUNTIME_PREFIX=/opt/ \
 		install_runtime
-	touch -c $(TARGET_DIR)/lib/libc.so.0
+	touch -c $(TARGET_DIR)/opt/lib/libc.so.0
 
-$(TARGET_DIR)/usr/bin/ldd:
+# utils are installed into $(PREFIX)/usr/bin
+# and must be moved into $(PREFIX)/bin
+$(TARGET_DIR)/opt/bin/ldd:
 	$(MAKE1) -C $(UCLIBC_DIR) $(TARGET_CONFIGURE_OPTS) \
 		PREFIX=$(TARGET_DIR) utils install_utils
+	mkdir -p $(TARGET_DIR)/opt/bin
+	install -c $(TARGET_DIR)/opt/usr/bin/ldd  $(TARGET_DIR)/opt/bin
+	rm -rf $(TARGET_DIR)/opt/usr
 ifeq ($(strip $(BR2_CROSS_TOOLCHAIN_TARGET_UTILS)),y)
 	mkdir -p $(STAGING_DIR)/$(REAL_GNU_TARGET_NAME)/target_utils
-	install -c $(TARGET_DIR)/usr/bin/ldd \
+	install -c $(TARGET_DIR)/opt/bin/ldd \
 		$(STAGING_DIR)/$(REAL_GNU_TARGET_NAME)/target_utils/ldd
 endif
-	touch -c $(TARGET_DIR)/usr/bin/ldd
+	touch -c $(TARGET_DIR)/opt/bin/ldd
 
-UCLIBC_TARGETS=$(TARGET_DIR)/lib/libc.so.0
+UCLIBC_TARGETS=$(TARGET_DIR)/opt/lib/libc.so.0
 endif
 
 uclibc-configured: $(UCLIBC_DIR)/.configured
@@ -187,7 +193,7 @@
 uclibc-dirclean:
 	rm -rf $(UCLIBC_DIR)
 
-uclibc-target-utils: $(TARGET_DIR)/usr/bin/ldd
+uclibc-target-utils: $(TARGET_DIR)/opt/bin/ldd
 
 #############################################################
 #
@@ -196,15 +202,15 @@
 #
 #############################################################
 
-$(TARGET_DIR)/usr/lib/libc.a: $(STAGING_DIR)/$(REAL_GNU_TARGET_NAME)/lib/libc.a
+$(TARGET_DIR)/opt/lib/libc.a: $(STAGING_DIR)/$(REAL_GNU_TARGET_NAME)/lib/libc.a
 	$(MAKE1) -C $(UCLIBC_DIR) \
 		PREFIX=$(TARGET_DIR) \
-		DEVEL_PREFIX=/usr/ \
-		RUNTIME_PREFIX=/ \
+		DEVEL_PREFIX=/opt/ \
+		RUNTIME_PREFIX=/opt/ \
 		install_dev
-	touch -c $(TARGET_DIR)/usr/lib/libc.a
+	touch -c $(TARGET_DIR)/opt/lib/libc.a
 
-uclibc_target: gcc uclibc $(TARGET_DIR)/usr/lib/libc.a $(TARGET_DIR)/usr/bin/ldd
+uclibc_target: gcc uclibc $(TARGET_DIR)/opt/lib/libc.a $(TARGET_DIR)/opt/bin/ldd
 
 uclibc_target-clean:
 	rm -f $(TARGET_DIR)/include
