--- cherokee-0.4.26/cget/Makefile.in	2005-09-19 09:44:32.000000000 -0700
+++ cherokee/cget/Makefile.in	2005-09-20 07:42:50.964048376 -0700
@@ -56,6 +56,7 @@
 cget_OBJECTS = $(am_cget_OBJECTS)
 am__DEPENDENCIES_1 =
 cget_DEPENDENCIES = ../cherokee/libcherokee-client.la \
+	../cherokee/libcherokee-base.la \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
@@ -286,6 +287,7 @@
 
 cget_LDADD = \
 ../cherokee/libcherokee-client.la \
+../cherokee/libcherokee-base.la \
 $(TLS_LIBS) \
 $(PTHREAD_LIBS) \
 $(PTHREAD_CFLAGS)
