--- cherokee-0.5.4/configure.in	2006-07-11 18:08:14.000000000 +0200
+++ cherokee/configure.in	2006-07-19 13:35:19.000000000 +0200
@@ -280,25 +280,25 @@
 if test "x$have_epoll_include" = "xyes"; then
  	AC_MSG_CHECKING(for epoll system call)
 
-     AC_TRY_RUN([
-		#include <stdint.h>
-		#include <sys/param.h>
-		#include <sys/types.h>
-		#include <sys/syscall.h>
-		#include <sys/epoll.h>
-		#include <unistd.h>
-
-		int epoll_create (int size) {
-	        return (syscall(__NR_epoll_create, size));
-		}
-	
-		int main (int argc, char **argv) {
-	        int epfd;
-	        epfd = epoll_create(256);
-	    	   exit (epfd == -1 ? 1 : 0);
-		}
-	], have_epoll=yes)
-	AC_MSG_RESULT($have_epoll)
+#     AC_TRY_RUN([
+#		#include <stdint.h>
+#		#include <sys/param.h>
+#		#include <sys/types.h>
+#		#include <sys/syscall.h>
+#		#include <sys/epoll.h>
+#		#include <unistd.h>
+#
+#		int epoll_create (int size) {
+#	        return (syscall(__NR_epoll_create, size));
+#		}
+#	
+#		int main (int argc, char **argv) {
+#	        int epfd;
+#	        epfd = epoll_create(256);
+#	    	   exit (epfd == -1 ? 1 : 0);
+#		}
+#	], have_epoll=yes)
+	AC_MSG_RESULT(yes)
 fi
 
 dnl
@@ -614,15 +614,7 @@
 
 # Is sendfile broken?
 AC_MSG_CHECKING(if sendfile works)
-AC_TRY_RUN([#include <errno.h>
-            int main() {
-               int o = 0;
-               if (-1 == sendfile(0, 0, &o, 0) && errno == ENOSYS) return -1;
-               return 0;
-            } ], 
-            AC_MSG_RESULT(yes), 
-            [ AC_MSG_RESULT(no) 
-              AC_DEFINE([HAVE_SENDFILE_BROKEN], [1], [broken sendfile]) ])
+            AC_MSG_RESULT(yes)
 
 # readdir_r()
 LIBWWW_READDIR_R_TYPE
