--- a/Makefile.am
+++ b/Makefile.am
@@ -239,13 +239,6 @@
 libexec_PROGRAMS += extras/firmware/firmware
 
 # ------------------------------------------------------------------------------
-# ata_id - ATA identify
-# ------------------------------------------------------------------------------
-extras_ata_id_ata_id_SOURCES = extras/ata_id/ata_id.c
-extras_ata_id_ata_id_LDADD = libudev/libudev-private.la
-libexec_PROGRAMS += extras/ata_id/ata_id
-
-# ------------------------------------------------------------------------------
 # cdrom_id - optical drive/media capability
 # ------------------------------------------------------------------------------
 extras_cdrom_id_cdrom_id_SOURCES = extras/cdrom_id/cdrom_id.c
@@ -261,19 +254,6 @@
 libexec_PROGRAMS += extras/collect/collect
 
 # ------------------------------------------------------------------------------
-# scsi_id - SCSI inquiry to get various serial numbers
-# ------------------------------------------------------------------------------
-extras_scsi_id_scsi_id_SOURCES =\
-	extras/scsi_id/scsi_id.c \
-	extras/scsi_id/scsi_serial.c \
-	extras/scsi_id/scsi.h \
-	extras/scsi_id/scsi_id.h
-extras_scsi_id_scsi_id_LDADD = libudev/libudev-private.la
-libexec_PROGRAMS += extras/scsi_id/scsi_id
-dist_man_MANS += extras/scsi_id/scsi_id.8
-EXTRA_DIST += extras/scsi_id/README
-
-# ------------------------------------------------------------------------------
 # v4l_id - video4linux capabilities
 # ------------------------------------------------------------------------------
 extras_v4l_id_v4l_id_SOURCES = extras/v4l_id/v4l_id.c
