--- a/include/curl/system.h
+++ b/include/curl/system.h
@@ -401,7 +401,7 @@
 
 #elif defined(__GNUC__)
 #  if !defined(__LP64__) && (defined(__ILP32__) || \
-      defined(__i386__) || defined(__ppc__) || defined(__arm__) || \
+      defined(__i386__) || defined(__powerpc__) || defined(__arm__) || \
       defined(__sparc__) || defined(__mips__) || defined(__sh__) || \
       defined(__XTENSA__))
 #    define CURL_SIZEOF_LONG           4
