--- asterisk-1.2.10/codecs/gsm/Makefile	2006-05-30 21:18:30.000000000 +0200
+++ asterisk/codecs/gsm/Makefile	2006-07-20 16:15:15.000000000 +0200
@@ -243,6 +243,10 @@
 ifneq (${PROC},arm)
 ifneq ($(shell uname -m), parisc)
 ifneq ($(shell uname -m),s390)
+ifneq (${PROC}, ppc)
+ifneq (${PROC}, mips1)
+ifneq (${PROC}, mips)
+ifneq (${PROC}, mipsel)
 GSM_SOURCES+= $(SRC)/k6opt.s
 endif
 endif
@@ -255,6 +259,10 @@
 endif
 endif
 endif
+endif
+endif
+endif
+endif
 
 TOAST_SOURCES = $(SRC)/toast.c 		\
 		$(SRC)/toast_lin.c	\
@@ -309,6 +317,11 @@
 ifneq ($(shell uname -m), armv4l)
 ifneq ($(shell uname -m), parisc)
 ifneq ($(shell uname -m),s390)
+ifneq (${PROC},arm)
+ifneq (${PROC},ppc)
+ifneq (${PROC},mips1)
+ifneq (${PROC},mips)
+ifneq (${PROC},mipsel)
 GSM_OBJECTS+= $(SRC)/k6opt.o
 endif
 endif
@@ -319,6 +332,11 @@
 endif
 endif
 endif
+endif
+endif
+endif
+endif
+endif
 
 TOAST_OBJECTS =	$(SRC)/toast.o 		\
 		$(SRC)/toast_lin.o	\
