--- asterisk-1.2.2/codecs/gsm/Makefile~	2005-11-07 19:31:45.000000000 -0800
+++ asterisk-1.2.2/codecs/gsm/Makefile	2005-11-12 16:51:08.000000000 -0800
@@ -297,6 +297,7 @@
 ifneq ($(shell uname -m), sparc64)
 ifneq ($(shell uname -m), armv4l)
 ifneq ($(shell uname -m), parisc)
+ifneq (${PROC},arm)
 GSM_OBJECTS+= $(SRC)/k6opt.o
 endif
 endif
@@ -305,6 +306,7 @@
 endif
 endif
 endif
+endif
 
 TOAST_OBJECTS =	$(SRC)/toast.o 		\
 		$(SRC)/toast_lin.o	\
