--- mysql-5.1.11-beta.org/sql/Makefile.am	2006-06-02 20:00:51.000000000 +0200
+++ mysql5/sql/Makefile.am	2006-08-13 14:00:01.000000000 +0200
@@ -156,8 +156,8 @@
 		@echo "If it fails, re-run configure with --with-low-memory"
 		$(CXXCOMPILE) $(LM_CFLAGS) -c $<
 
-lex_hash.h:	gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@
+lex_hash.h:
+		$(MYSQL5_BUILD_DIR)-native/sql/gen_lex_hash > $@
 
 # the following three should eventually be moved out of this directory
 ha_berkeley.o:	ha_berkeley.cc ha_berkeley.h
