--- emu/Linux/asm-arm.S.orig	2007-11-07 10:38:01.000000000 -0800
+++ emu/Linux/asm-arm.S	2007-11-07 10:48:32.000000000 -0800
@@ -26,7 +26,7 @@
         blx     r1             /* call tramp*/
 
         /* if we return here, tramp didn't do it's job */
-        swi     SYS_exit
+        swi     0x900001
         ldmea   fp, {fp, sp, pc}
         .size   executeonnewstack, .-executeonnewstack
 
@@ -50,7 +50,7 @@
         sub     sp, sp, #4
         mov     r1, #0
         str     r1, [r0]
-        swi     SYS_exit
+        swi     0x900001
         ldmea   fp, {fp, sp, pc}
         .size   unlockandexit, .-unlockandexit
 
