diff -Nru postfix-2.1.5-orig/postfix-install postfix-2.1.5/postfix-install
--- postfix-2.1.5-orig/postfix-install	2004-04-14 11:45:23.000000000 -0400
+++ postfix-2.1.5/postfix-install	2005-04-29 01:41:15.180000000 -0400
@@ -191,11 +191,6 @@
     exit 1
 }
 
-test -x bin/postconf || {
-    echo $0: Error: no bin/postconf file. Did you forget to run \"make\"? 1>&2
-    exit 1
-}
-
 case `uname -s` in
 HP-UX*) FMT=cat;;
      *) FMT=fmt;;
@@ -352,7 +347,7 @@
 
 : ${install_root=/}
 : ${tempdir=`pwd`}
-: ${config_directory=`bin/postconf -h -d config_directory`}
+: ${config_directory=/opt/etc/postfix}
 
 # Find out the location of installed configuration files.
 
@@ -404,54 +399,6 @@
     }
 }
 
-# Extract parameter settings from the installed main.cf file.
-
-test -f $CONFIG_DIRECTORY/main.cf && {
-    for name in daemon_directory command_directory queue_directory mail_owner \
-	setgid_group sendmail_path newaliases_path mailq_path manpage_directory \
-	sample_directory html_directory readme_directory
-    do
-	eval junk=\$$name
-	case "$junk" in
-	"") eval unset $name;;
-	esac
-	eval : \${$name=\`bin/postconf -c $CONFIG_DIRECTORY -h $name\`} ||
-	    exit 1
-    done
-}
-
-# Use built-in defaults as the final source of parameter settings.
-
-for name in daemon_directory command_directory queue_directory mail_owner \
-    setgid_group sendmail_path newaliases_path mailq_path manpage_directory \
-    sample_directory html_directory readme_directory
-do
-    eval junk=\$$name
-    case "$junk" in
-    "") eval unset $name;;
-    esac
-    eval : \${$name=\`bin/postconf -d -h $name\`} || exit 1
-done
-
-# Override settings manually.
-
-test -z "$non_interactive" && for name in daemon_directory command_directory \
-    queue_directory sendmail_path newaliases_path mailq_path mail_owner \
-    setgid_group html_directory manpage_directory readme_directory
-do
-    while :
-    do
-	echo
-	eval echo Please specify \$${name}_prompt | ${FMT}
-	eval echo \$n "$name: [\$$name]\  \$c"
-	read ans
-	case $ans in
-	"") break;;
-	 *) eval $name=$ans; break;;
-	esac
-    done
-done
-
 # Sanity checks
 
 case "$setgid_group" in
@@ -696,26 +643,6 @@
 
     done
 
-# Save the installation parameters to main.cf even when they haven't
-# changed from their current default. Defaults can change between
-# Postfix releases, and software should not suddenly be installed in
-# the wrong place when Postfix is being upgraded.
-
-bin/postconf -c $CONFIG_DIRECTORY -e \
-    "daemon_directory = $daemon_directory" \
-    "command_directory = $command_directory" \
-    "queue_directory = $queue_directory" \
-    "mail_owner = $mail_owner" \
-    "setgid_group = $setgid_group" \
-    "sendmail_path = $sendmail_path" \
-    "mailq_path = $mailq_path" \
-    "newaliases_path = $newaliases_path" \
-    "html_directory = $html_directory" \
-    "manpage_directory = $manpage_directory" \
-    "sample_directory = $sample_directory" \
-    "readme_directory = $readme_directory" \
-|| exit 1
-
 # If Postfix is being installed locally from source code, do the
 # post-install processing now.
 
