--- otp_src_R10B-9/erts/emulator/Makefile.in	2005-12-02 06:43:58.000000000 -0800
+++ erlang/erts/emulator/Makefile.in	2005-12-24 10:35:39.000000000 -0800
@@ -215,7 +215,7 @@
 UNIX_ONLY_BUILDS =
 endif
 
-all: generate erts_lib zlib $(BINDIR)/$(EMULATOR_EXECUTABLE) $(UNIX_ONLY_BUILDS)
+all: generate erts_lib zlib $(BINDIR)/$(EMULATOR_EXECUTABLE) $(BINDIR)/$(EMULATOR_EXECUTABLE_SAE) $(UNIX_ONLY_BUILDS)
 
 zlib:
 	@set -e ; cd zlib && $(MAKE) $(TYPE)
@@ -669,7 +669,7 @@
 	$(CC) $(subst O2,O3, $(CFLAGS)) $(INCLUDES) -c $< -o $@
 
 $(BINDIR)/hipe_mkliterals$(TYPEMARKER):	$(OBJDIR)/hipe_mkliterals.o
-	$(CC) $(CFLAGS) $(INCLUDES) -o $@ $<
+	gcc $(CFLAGS) $(INCLUDES) -o $@ hipe/hipe_mkliterals.c
 
 $(OBJDIR)/hipe_mkliterals.o:	$(TARGET)/hipe_x86_asm.h $(TARGET)/hipe_ppc_asm.h
 
