--- w3m/Makefile.in.orig	2007-05-29 05:01:04.000000000 -0700
+++ w3m/Makefile.in	2007-06-05 20:49:42.000000000 -0700
@@ -170,8 +170,10 @@
 	./mktable$(EXT) 100 functable.tab > $@
 	-rm -f functable.tab
 
+ifeq ($(CROSS_COMPILATION), no)
 mktable$(EXT): mktable.o dummy.o $(ALIB)
 	$(CC) $(CFLAGS) -o mktable mktable.o dummy.o $(LDFLAGS) $(LIBS) $(EXT_LIBS)
+endif
 
 $(BOOKMARKER): w3mbookmark.o dummy.o $(ALIB)
 	$(CC) $(CFLAGS) -o $(BOOKMARKER) w3mbookmark.o dummy.o $(LDFLAGS) $(LIBS) $(EXT_LIBS)
