--- src/killall.c.orig	2007-01-04 15:43:26.000000000 -0800
+++ src/killall.c	2007-01-04 15:43:44.000000000 -0800
@@ -395,8 +395,7 @@
 #ifdef FLASK_LINUX
   fprintf (stderr, "  -S,--Sid            kill only process(es) having sid\n");
   fprintf (stderr, "  -c,--context        kill only process(es) having scontext\n");
-  fprintf(stderr, "   (-s, -c are mutually exclusive and must precede other
-arguments)\n\n");
+  fprintf(stderr, "   (-s, -c are mutually exclusive and must precede other arguments)\n\n");
 #endif /*FLASK_LINUX*/
 }
 
