--- otp_src_R11B-1/erts/emulator/Makefile.in.orig	2006-08-29 02:46:47.000000000 -0700
+++ otp_src_R11B-1/erts/emulator/Makefile.in	2006-08-31 21:46:25.000000000 -0700
@@ -792,9 +792,10 @@
 	$(CC) $(subst O2,O3, $(CFLAGS)) $(INCLUDES) -c $< -o $@
 
 $(BINDIR)/hipe_mkliterals$(TF_MARKER):	$(OBJDIR)/hipe_mkliterals.o
-	$(CC) $(CFLAGS) $(INCLUDES) -o $@ $<
+	gcc $(CFLAGS) $(INCLUDES) -o $@ hipe/hipe_mkliterals.c
 
 $(OBJDIR)/hipe_mkliterals.o:	$(TTF_DIR)/hipe_x86_asm.h $(TTF_DIR)/hipe_ppc_asm.h
+	gcc $(CFLAGS) $(INCLUDES) -c hipe/hipe_mkliterals.c
 
 $(TTF_DIR)/hipe_literals.h:	$(BINDIR)/hipe_mkliterals$(TF_MARKER)
 	$(BINDIR)/hipe_mkliterals$(TF_MARKER) -c > $@
