--- otp_src_R13B/erts/configure.in.orig	2009-04-16 09:23:46.000000000 +0000
+++ otp_src_R13B/erts/configure.in	2009-04-21 23:53:10.335177341 +0000
@@ -483,10 +483,13 @@
     i686)	ARCH=x86;;
     x86_64)	ARCH=amd64;;
     amd64)	ARCH=amd64;;
+    powerpc)	ARCH=ppc;;
     macppc)	ARCH=ppc;;
     ppc)	ARCH=ppc;;
     ppc64)	ARCH=ppc64;;
     "Power Macintosh")	ARCH=ppc;;
+    arm)	ARCH=arm;;
+    armeb)	ARCH=arm;;
     armv5b)	ARCH=arm;;
     armv5teb)	ARCH=arm;;
     armv5tel)	ARCH=arm;;
@@ -3330,6 +3333,7 @@
     SSL_APP=ssl
     CRYPTO_APP=crypto
     SSH_APP=ssh
+    SSL_DYNAMIC_ONLY="yes" 
 fi
 
 if test "x$SSL_APP" != "x" ; then
