--- erlang/erts/emulator/Makefile.in.orig	2006-05-15 06:06:30.000000000 -0700
+++ erlang/erts/emulator/Makefile.in	2006-05-20 10:20:48.000000000 -0700
@@ -782,9 +782,10 @@
 	$(CC) $(subst O2,O3, $(CFLAGS)) $(INCLUDES) -c $< -o $@
 
 $(BINDIR)/hipe_mkliterals$(TYPEMARKER):	$(OBJDIR)/hipe_mkliterals.o
-	$(CC) $(CFLAGS) $(INCLUDES) -o $@ $<
+	gcc $(CFLAGS) $(INCLUDES) -o $@ hipe/hipe_mkliterals.c
 
 $(OBJDIR)/hipe_mkliterals.o:	$(TARGET_TYPE_DIR)/hipe_x86_asm.h $(TARGET_TYPE_DIR)/hipe_ppc_asm.h
+	gcc $(CFLAGS) $(INCLUDES) -c hipe/hipe_mkliterals.c
 
 $(TARGET_TYPE_DIR)/hipe_literals.h:	$(BINDIR)/hipe_mkliterals$(TYPEMARKER)
 	$(BINDIR)/hipe_mkliterals$(TYPEMARKER) -c > $@
