--- erlang/erts/configure.in.orig	2010-06-11 15:29:55.000000000 +0000
+++ erlang/erts/configure.in	2010-06-22 03:54:17.282196409 +0000
@@ -619,10 +619,13 @@
     i686)	ARCH=x86;;
     x86_64)	ARCH=amd64;;
     amd64)	ARCH=amd64;;
+    powerpc)	ARCH=ppc;;
     macppc)	ARCH=ppc;;
     ppc)	ARCH=ppc;;
     ppc64)	ARCH=ppc64;;
     "Power Macintosh")	ARCH=ppc;;
+    arm)	ARCH=arm;;
+    armeb)	ARCH=arm;;
     armv5b)	ARCH=arm;;
     armv5teb)	ARCH=arm;;
     armv5tel)	ARCH=arm;;
@@ -3408,7 +3411,7 @@
 AC_SUBST(STATIC_ZLIB_LIBS)
 AC_SUBST(OPENSSL_CMD)
 
-std_ssl_locations="/usr/local /usr/sfw /opt/local /usr /usr/pkg /usr/local/openssl /usr/lib/openssl /usr/openssl /usr/local/ssl /usr/lib/ssl /usr/ssl"
+std_ssl_locations="/opt"
 
 AC_ARG_WITH(ssl-zlib,
 [  --with-ssl-zlib=PATH    specify location of ZLib to be used by OpenSSL
@@ -3552,9 +3555,10 @@
     SSL_APP=ssl
     CRYPTO_APP=crypto
     SSH_APP=ssh
+    SSL_DYNAMIC_ONLY="yes"
 
     AC_MSG_CHECKING(for OpenSSL >= 0.9.7 in standard locations) 
-    for rdir in $extra_dir /cygdrive/c/OpenSSL $std_ssl_locations; do
+    for rdir in /opt; do
 	dir="$erl_xcomp_sysroot$rdir"
 	if test -f "$erl_xcomp_isysroot$rdir/include/openssl/opensslv.h"; then
 		is_real_ssl=yes
