--- erlang/erts/configure.in.orig	2007-06-11 05:53:45.000000000 -0700
+++ erlang/erts/configure.in	2007-06-14 14:31:49.000000000 -0700
@@ -331,7 +331,8 @@
 FreeBSD) OPSYS=freebsd;;
 esac
 ARCH=noarch
-case `uname -m` in
+test "x$TARGET_ARCH" != x && TEST_ARCH=$TARGET_ARCH || TEST_ARCH=`uname -m`
+case $TEST_ARCH in
 sun4u)	ARCH=ultrasparc;;
 sparc64)	ARCH=sparc64;;
 sun4v)	ARCH=ultrasparc;;
@@ -342,10 +343,13 @@
 i686)	ARCH=x86;;
 x86_64) ARCH=amd64;;
 amd64)	ARCH=amd64;;
+powerpc)	ARCH=ppc;;
 macppc)	ARCH=ppc;;
 ppc)	ARCH=ppc;;
 ppc64)	ARCH=ppc64;;
 "Power Macintosh")	ARCH=ppc;;
+arm)	ARCH=arm;;
+armeb)	ARCH=arm;;
 armv5b)	ARCH=arm;;
 armv5teb)	ARCH=arm;;
 esac
@@ -2232,7 +2236,7 @@
   exit(0);
 #endif
 }
-], poll_works=true, poll_works=false, poll_works=false)
+], poll_works=true, poll_works=false, poll_works=true)
 case $poll_works in
 true)
 	AC_DEFINE(ERTS_USE_POLL, 1, [Define if poll() should be used instead of select()])
@@ -2399,7 +2403,11 @@
     DED_CFLAGS="$DED_CFLAGS -fPIC"
 fi
 
+if test "x$LD" = x; then
 DED_LD=ld
+else
+DED_LD=$LD
+fi
 DED_LD_FLAG_RUNTIME_LIBRARY_PATH="-R"
 STATIC_CFLAGS=""
 
@@ -2430,7 +2438,7 @@
 	;;
 	linux*)
 		DED_LD="$CC"
-		DED_LDFLAGS="-shared"
+		DED_LDFLAGS="-shared $LDFLAGS"
 	;;	
 	osf*)
 		# NOTE! Whitespace after -rpath is important.
@@ -2439,7 +2447,7 @@
 	;;
 	*)
 		# assume GNU linker and ELF
-		DED_LDFLAGS="-shared"
+		DED_LDFLAGS="-shared $LDFLAGS"
 	;;
 esac
 
@@ -2716,6 +2724,7 @@
     SSL_APP=ssl
     CRYPTO_APP=crypto
     SSH_APP=ssh
+    SSL_DYNAMIC_ONLY="yes" 
 fi
 
 if test "x$SSL_APP" != "x" ; then
