--- otp_src_R13B/Makefile.in.orig	2009-03-12 12:19:41.000000000 +0000
+++ otp_src_R13B/Makefile.in	2009-04-21 23:16:12.407282225 +0000
@@ -809,9 +809,10 @@
 # Install erlang base public files
 #
 install.bin:
+	cd $(BINDIR); \
 	for file in $(ERL_BASE_PUB_FILES); do \
 		rm -f $(BINDIR)/$$file; \
-		${LN_S} $(ERLANG_BINDIR)/$$file $(BINDIR)/$$file; \
+		${LN_S} ../lib/erlang/bin/$$file .; \
 	done
 
 #
