--- otp_src_R10B-9/Makefile.in	2005-10-24 02:43:23.000000000 -0700
+++ erlang/Makefile.in	2005-12-24 10:35:39.000000000 -0800
@@ -532,12 +532,12 @@
 install.bin:
 	rm -f $(BINDIR)/erl $(BINDIR)/erlc \
 	      $(BINDIR)/ecc $(BINDIR)/elink $(BINDIR)/ear $(BINDIR)/escript
-	${LN_S} $(ERLANG_BINDIR)/erl      $(BINDIR)/erl
-	${LN_S} $(ERLANG_BINDIR)/erlc     $(BINDIR)/erlc
-	${LN_S} $(ERLANG_BINDIR)/ecc      $(BINDIR)/ecc
-	${LN_S} $(ERLANG_BINDIR)/elink    $(BINDIR)/elink
-	${LN_S} $(ERLANG_BINDIR)/ear      $(BINDIR)/ear
-	${LN_S} $(ERLANG_BINDIR)/escript  $(BINDIR)/escript
+	#${LN_S} $(ERLANG_BINDIR)/erl      $(BINDIR)/erl
+	#${LN_S} $(ERLANG_BINDIR)/erlc     $(BINDIR)/erlc
+	#${LN_S} $(ERLANG_BINDIR)/ecc      $(BINDIR)/ecc
+	#${LN_S} $(ERLANG_BINDIR)/elink    $(BINDIR)/elink
+	#${LN_S} $(ERLANG_BINDIR)/ear      $(BINDIR)/ear
+	#${LN_S} $(ERLANG_BINDIR)/escript  $(BINDIR)/escript
 
 #
 # Directories needed before we can install
