--- a/gotools/configure
+++ b/gotools/configure
@@ -3783,7 +3783,7 @@ ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
- if test "$cross_compiling" = no; then
+ if test "$host" = "$target"; then
   NATIVE_TRUE=
   NATIVE_FALSE='#'
 else
--- a/gotools/configure.ac
+++ b/gotools/configure.ac
@@ -46,7 +46,7 @@ AC_PROG_INSTALL
 AC_PROG_CC
 AC_PROG_GO
 
-AM_CONDITIONAL(NATIVE, test "$cross_compiling" = no)
+AM_CONDITIONAL(NATIVE, test "$host" = "$target")
 
 dnl Test for -lsocket and -lnsl.  Copied from libjava/configure.ac.
 AC_CACHE_CHECK([for socket libraries], gotools_cv_lib_sockets,
