--- a/configure.in
+++ b/configure.in
@@ -992,19 +992,19 @@ else
    RRD_HOME=/usr/local/rrdtool
 fi
 
-RRD_LIB="-L${RRD_HOME}/lib -lrrd_th"
+RRD_LIB="-L${RRD_HOME}/lib -lrrd"
 
-if test -f "$RRD_HOME/lib/librrd_th.so"; then
+if test -f "$RRD_HOME/lib/librrd.so"; then
    AC_MSG_RESULT(checking for rrdtool... yes)
 else
-  if test -f "$RRD_HOME/lib/librrd_th.dylib"; then # OSX
+  if test -f "$RRD_HOME/lib/librrd.dylib"; then # OSX
      AC_MSG_RESULT(checking for rrdtool... yes)
   else
-     if test -f "$RRD_HOME/lib/librrd_th.a"; then
+     if test -f "$RRD_HOME/lib/librrd.a"; then
        AC_MSG_RESULT(checking for rrdtool... yes)
      else 
-       AC_CHECK_LIB([rrd_th], [main])
-       if test ".${ac_cv_lib_rrd_th_main}" != ".yes"; then
+       AC_CHECK_LIB([rrd], [main])
+       if test ".${ac_cv_lib_rrd_main}" != ".yes"; then
          AC_MSG_ERROR(Unable to find RRD at $RRD_HOME: please use --with-rrd-home=DIR);
 	 AC_MSG_ERROR(RRD source can be downloaded from http://www.rrdtool.org/);
        else
