--- otp_src_R10B-6-orig/erts/configure.in	2005-06-21 09:37:14.000000000 -0700
+++ erlang/erts/configure.in	2005-06-23 22:09:58.055415000 -0700
@@ -1277,7 +1277,7 @@
   exit(0);
 #endif
 }
-], poll_works=true, poll_works=false, poll_works=false)
+], poll_works=true, poll_works=false, poll_works=true)
 case $poll_works in
 true)
 	AC_MSG_RESULT(ok)
@@ -1325,7 +1325,11 @@
 	DED_CFLAGS="$ETHR_DEFS $DED_CFLAGS"
 fi
 
+if test "x$LD" = x; then
 DED_LD=ld
+else
+DED_LD=$LD
+fi
 DED_LD_FLAG_RUNTIME_LIBRARY_PATH="-R"
 STATIC_CFLAGS=""
 
