--- ext/posix/posix.c.orig	2014-09-06 15:48:01.873158851 +0300
+++ ext/posix/posix.c	2014-09-06 15:55:35.311458359 +0300
@@ -632,7 +632,7 @@
 	add_assoc_string(return_value, "version",  u.version,  1);
 	add_assoc_string(return_value, "machine",  u.machine,  1);
 
-#if defined(_GNU_SOURCE) && !defined(DARWIN) && defined(HAVE_UTSNAME_DOMAINNAME)
+#if defined(_GNU_SOURCE) && !defined(__UCLIBC__) && !defined(DARWIN) && defined(HAVE_UTSNAME_DOMAINNAME)
 	add_assoc_string(return_value, "domainname", u.domainname, 1);
 #endif
 }
