--- git-core.orig/Makefile	2005-11-14 18:39:18.000000000 -0800
+++ git-core/Makefile	2005-12-11 20:35:42.000000000 -0800
@@ -56,7 +56,7 @@
 
 CFLAGS = -g -O2 -Wall
 LDFLAGS =
-ALL_CFLAGS = $(CFLAGS)
+ALL_CFLAGS = $(CFLAGS) $(CPPFLAGS)
 ALL_LDFLAGS = $(LDFLAGS)
 
 prefix = $(HOME)
@@ -434,6 +434,7 @@
 install: $(PROGRAMS) $(SCRIPTS)
 	$(INSTALL) -d -m755 $(call shellquote,$(DESTDIR)$(bindir))
 	$(INSTALL) $(PROGRAMS) $(SCRIPTS) $(call shellquote,$(DESTDIR)$(bindir))
+	cd $(DESTDIR)$(bindir); $(STRIP) $(PROGRAMS)
 	$(MAKE) -C templates install
 	$(INSTALL) -d -m755 $(call shellquote,$(DESTDIR)$(GIT_PYTHON_DIR))
 	$(INSTALL) $(PYMODULES) $(call shellquote,$(DESTDIR)$(GIT_PYTHON_DIR))
