--- git-1.4.1/templates/Makefile	2006-07-02 08:23:24.000000000 +0200
+++ git-core/templates/Makefile	2006-07-21 09:42:26.000000000 +0200
@@ -44,5 +44,5 @@
 
 install: all
 	$(INSTALL) -d -m755 $(call shellquote,$(DESTDIR)$(template_dir))
-	(cd blt && $(TAR) cf - .) | \
-	(cd $(call shellquote,$(DESTDIR)$(template_dir)) && $(TAR) xf -)
+	$(TAR) -c -C blt -f - . | \
+		$(TAR) -x -C  $(call shellquote,$(DESTDIR)$(template_dir)) -f -
