--- otp_src_R12B-1/erts/emulator/hipe/hipe_process.h.orig	2008-02-05 05:37:55.000000000 -0800
+++ otp_src_R12B-1/erts/emulator/hipe/hipe_process.h	2008-02-07 06:14:13.000000000 -0800
@@ -17,11 +17,11 @@
     Eterm *nstblacklim;		/* Black/gray stack boundary. Must exist if
 				   graylim exists. Ignored if no graylim. */
     void (*ngra)(void);		/* Saved original RA from graylim frame. */
-#if defined(__i386__) || defined(__x86_64__)
+#if 0 /* defined(__i386__) || defined(__x86_64__) */
     Eterm *ncsp;		/* Saved C stack pointer. */
     unsigned int narity;	/* Arity of BIF call, for stack walks. */
 #endif
-#if defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__)
+#if 1 /* defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__) */
     void (*nra)(void);		/* Native code return address. */
     unsigned int narity;	/* Arity of BIF call, for stack walks. */
 #endif
@@ -37,10 +37,10 @@
     p->nstgraylim = NULL;
     p->nstblacklim = NULL;
     p->ngra = NULL;
-#if defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__)
+#if 1 /* defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__) */
     p->nra = NULL;
 #endif
-#if defined(__i386__) || defined(__x86_64__) || defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__)
+#if 1 /* defined(__i386__) || defined(__x86_64__) || defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__) */
     p->narity = 0;
 #endif
 }
--- otp_src_R12B-1/erts/emulator/hipe/hipe_mkliterals.c.orig	2008-02-05 05:37:52.000000000 -0800
+++ otp_src_R12B-1/erts/emulator/hipe/hipe_mkliterals.c	2008-02-07 06:17:00.000000000 -0800
@@ -264,11 +264,11 @@
     { "P_NSP", offsetof(struct process, hipe.nsp) },
     { "P_NCALLEE", offsetof(struct process, hipe.ncallee) },
     { "P_CLOSURE", offsetof(struct process, hipe.closure) },
-#if defined(__i386__) || defined(__x86_64__)
+#if 0 /* defined(__i386__) || defined(__x86_64__) */
     { "P_NSP_LIMIT", offsetof(struct process, hipe.nstack) },
     { "P_CSP", offsetof(struct process, hipe.ncsp) },
     { "P_NARITY", offsetof(struct process, hipe.narity) },
-#elif defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__)
+#elif 1 /* defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__) */
     { "P_NSP_LIMIT", offsetof(struct process, hipe.nstack) },
     { "P_NRA", offsetof(struct process, hipe.nra) },
     { "P_NARITY", offsetof(struct process, hipe.narity) },
--- erlang/erts/emulator/Makefile.in.orig	2008-06-10 05:47:19.000000000 -0700
+++ erlang/erts/emulator/Makefile.in	2008-06-14 13:57:14.000000000 -0700
@@ -544,7 +544,7 @@
 endif
 
 $(OBJDIR)/%.o: beam/%.c
-	$(CC) $(subst -O2, $(GEN_OPT_FLGS), $(CFLAGS)) $(INCLUDES) -c $< -o $@
+	$(CC) $(INCLUDES) $(subst -O2, $(GEN_OPT_FLGS), $(CFLAGS)) -c $< -o $@
 
 else
 
@@ -840,7 +840,7 @@
 	$(CC) $(subst O2,O3, $(CFLAGS)) $(INCLUDES) -c $< -o $@
 
 $(BINDIR)/hipe_mkliterals$(TF_MARKER):	$(OBJDIR)/hipe_mkliterals.o
-	$(CC) $(CFLAGS) $(INCLUDES) -o $@ $<
+	cp `type -p $(@F)` $@
 
 $(OBJDIR)/hipe_mkliterals.o:	$(TTF_DIR)/hipe_x86_asm.h $(TTF_DIR)/hipe_ppc_asm.h
 
