--- otp_src_R13A/erts/emulator/hipe/hipe_process.h.orig	2009-03-12 12:16:17.000000000 +0000
+++ otp_src_R13A/erts/emulator/hipe/hipe_process.h	2009-03-19 16:44:47.097998402 +0000
@@ -35,10 +35,10 @@
     Eterm *nstblacklim;		/* Black/gray stack boundary. Must exist if
 				   graylim exists. Ignored if no graylim. */
     void (*ngra)(void);		/* Saved original RA from graylim frame. */
-#if defined(__i386__) || defined(__x86_64__)
+#if 0 /* defined(__i386__) || defined(__x86_64__) */
     Eterm *ncsp;		/* Saved C stack pointer. */
 #endif
-#if defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__)
+#if 1 /* defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__) */
     void (*nra)(void);		/* Native code return address. */
 #endif
     unsigned int narity;	/* Arity of BIF call, for stack walks. */
@@ -54,7 +54,7 @@
     p->nstgraylim = NULL;
     p->nstblacklim = NULL;
     p->ngra = NULL;
-#if defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__)
+#if 1 /* defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__) */
     p->nra = NULL;
 #endif
     p->narity = 0;
--- otp_src_R13A/erts/emulator/hipe/hipe_mkliterals.c.orig	2009-03-12 12:16:12.000000000 +0000
+++ otp_src_R13A/erts/emulator/hipe/hipe_mkliterals.c	2009-03-19 16:41:21.738286002 +0000
@@ -281,10 +281,10 @@
     { "P_NSP", offsetof(struct process, hipe.nsp) },
     { "P_NCALLEE", offsetof(struct process, hipe.ncallee) },
     { "P_CLOSURE", offsetof(struct process, hipe.closure) },
-#if defined(__i386__) || defined(__x86_64__)
+#if 0 /* defined(__i386__) || defined(__x86_64__) */
     { "P_NSP_LIMIT", offsetof(struct process, hipe.nstack) },
     { "P_CSP", offsetof(struct process, hipe.ncsp) },
-#elif defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__)
+#elif 1 /* defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__powerpc64__) || defined(__arm__) */
     { "P_NSP_LIMIT", offsetof(struct process, hipe.nstack) },
     { "P_NRA", offsetof(struct process, hipe.nra) },
 #endif
--- erlang/erts/emulator/Makefile.in.orig	2008-06-10 05:47:19.000000000 -0700
+++ erlang/erts/emulator/Makefile.in	2008-06-14 13:57:14.000000000 -0700
@@ -544,7 +544,7 @@
 endif
 
 $(OBJDIR)/%.o: beam/%.c
-	$(CC) $(subst -O2, $(GEN_OPT_FLGS), $(CFLAGS)) $(INCLUDES) -c $< -o $@
+	$(CC) $(INCLUDES) $(subst -O2, $(GEN_OPT_FLGS), $(CFLAGS)) -c $< -o $@
 
 else
 
@@ -840,7 +840,7 @@
 	$(CC) $(subst O2,O3, $(CFLAGS)) $(INCLUDES) -c $< -o $@
 
 $(BINDIR)/hipe_mkliterals$(TF_MARKER):	$(OBJDIR)/hipe_mkliterals.o
-	$(CC) $(CFLAGS) $(INCLUDES) -o $@ $<
+	cp HOST_HIPE_MKLITERAL_PATH/$(@F) $@
 
 $(OBJDIR)/hipe_mkliterals.o:	$(TTF_DIR)/hipe_x86_asm.h $(TTF_DIR)/hipe_ppc_asm.h
 
