--- otp_src_R10B-5-orig/erts/boot/src/Makefile	2003-07-07 05:19:02.000000000 -0700
+++ erlang/erts/boot/src/Makefile	2005-06-02 21:28:25.864321000 -0700
@@ -55,26 +55,26 @@
 ## Build escript ear and esh using ecc and elink
 
 escript: escript.beam
-	./elink -t unix -d -o escript -S escript -m escript.beam 
+	elink -t unix -d -o escript -S escript -m escript.beam 
 
 escript.beam: escript.erl
-	./ecc $(ERL_FLAGS) escript.erl
+	ecc $(ERL_FLAGS) escript.erl
 
 ## Build ear
 
 ear: ear.beam elink 
-	./elink -t unix -d -o ear -S ear -m ear.beam
+	elink -t unix -d -o ear -S ear -m ear.beam
 
 ##----------------------------------------------------------------------
 
 esh: esh.beam shell.beam
-	./elink -t unix -d -o esh -S esh -m esh.beam shell.beam
+	elink -t unix -d -o esh -S esh -m esh.beam shell.beam
 
 esh.beam: esh.erl
-	./ecc $(ERL_FLAGS) esh.erl
+	ecc $(ERL_FLAGS) esh.erl
 
 shell.beam: shell.erl
-	./ecc $(ERL_FLAGS) shell.erl
+	ecc $(ERL_FLAGS) shell.erl
 
 ##----------------------------------------------------------------------
 
