--- erlang/erts/configure.in.orig	2010-02-19 10:03:41.000000000 -0800
+++ erlang/erts/configure.in	2010-02-26 23:10:23.000000000 -0800
@@ -611,10 +611,13 @@
     i686)	ARCH=x86;;
     x86_64)	ARCH=amd64;;
     amd64)	ARCH=amd64;;
+    powerpc)	ARCH=ppc;;
     macppc)	ARCH=ppc;;
     ppc)	ARCH=ppc;;
     ppc64)	ARCH=ppc64;;
     "Power Macintosh")	ARCH=ppc;;
+    arm)	ARCH=arm;;
+    armeb)	ARCH=arm;;
     armv5b)	ARCH=arm;;
     armv5teb)	ARCH=arm;;
     armv5tel)	ARCH=arm;;
@@ -3397,7 +3400,7 @@
 		SSL_LINK_WITH_ZLIB=no
 		STATIC_ZLIB_LIBS=
 		AC_MSG_CHECKING(for static ZLib to be used by SSL in standard locations) 
-		for rdir in $std_ssl_locations; do
+		for rdir in /opt; do
 			dir="$erl_xcomp_sysroot$rdir"
 			if test "x$ac_cv_sizeof_void_p" = "x8"; then
 				if test -f "$dir/lib64/libz.a"; then
@@ -3518,9 +3521,10 @@
     SSL_APP=ssl
     CRYPTO_APP=crypto
     SSH_APP=ssh
+    SSL_DYNAMIC_ONLY="yes"
 
     AC_MSG_CHECKING(for OpenSSL >= 0.9.7 in standard locations) 
-    for rdir in $extra_dir /cygdrive/c/OpenSSL $std_ssl_locations; do
+    for rdir in /opt; do
 	dir="$erl_xcomp_sysroot$rdir"
 	if test -f "$erl_xcomp_isysroot$rdir/include/openssl/opensslv.h"; then
 		is_real_ssl=yes
