--- pymssql-0.8.0/setup.py.orig	2006-09-22 14:03:02.000000000 -0700
+++ pymssql-0.8.0/setup.py	2006-12-25 23:20:35.000000000 -0800
@@ -69,16 +69,6 @@
 		library_dirs.insert(0, os.path.join(p, r'DevTools\lib'))
 
 else:	# try some generic paths
-	include_dirs = [
-		'/usr/include', '/usr/local/include',
-		'/usr/include/freetds', '/usr/local/include/freetds',
-		'/usr/pkg/freetds/include'	# netbsd
-	]
-	library_dirs = [
-		'/usr/lib', '/usr/local/lib',
-		'/usr/lib/freetds', '/usr/local/lib/freetds'
-		'/usr/pkg/freetds/lib'		# netbsd
-	]
 	libraries = ["sybdb"]
 	data_files = []
 
@@ -110,8 +100,6 @@
 	url = 'http://pymssql.sourceforge.net',
 	py_modules = [ 'pymssql', ],
 	ext_modules = [Extension('_mssql', ['mssqldbmodule.c'],
-			include_dirs = include_dirs,
-			library_dirs = library_dirs,
 			libraries = libraries)],
 	classifiers = filter(None, classifiers.split('\n')),
 	data_files = data_files
