--- http/utils/Makefile.original	2005-06-12 16:04:00.000000000 +0100
+++ http/utils/Makefile	2005-08-11 23:42:16.000000000 +0100
@@ -4,7 +4,6 @@
 # Copyright (c) Mbedthis Software LLC, 2003-2005. All Rights Reserved.
 #
 
-BLD_NATIVE	:= 1
 COMPILE		:= *.cpp 
 MAKE_IFLAGS	:= -I.. -I../../mpr -I../../esp
 
@@ -14,19 +13,22 @@
 	$(BLD_BIN_DIR)/httpComp$(BLD_EXE) $(BLD_BIN_DIR)/httpPassword$(BLD_EXE)
 
 $(BLD_BIN_DIR)/charGen$(BLD_EXE): charGen$(BLD_OBJ)
-	@bld --native --executable $(BLD_BIN_DIR)/charGen$(BLD_EXE) \
+	@bld  --executable $(BLD_BIN_DIR)/charGen$(BLD_EXE) \
 		--preferStatic \
-		--objects charGen$(BLD_OBJ)
+		--objects charGen$(BLD_OBJ) \
+		--libs "stdc++"
 
 $(BLD_BIN_DIR)/httpComp$(BLD_EXE): httpComp$(BLD_OBJ)
-	@bld --native --executable $(BLD_BIN_DIR)/httpComp$(BLD_EXE) \
+	@bld  --executable $(BLD_BIN_DIR)/httpComp$(BLD_EXE) \
 		--preferStatic \
-		--objects httpComp$(BLD_OBJ)
+		--objects httpComp$(BLD_OBJ) \
+		--libs "stdc++"
 
 $(BLD_BIN_DIR)/httpPassword$(BLD_EXE): httpPassword$(BLD_OBJ)
-	@bld --native --executable $(BLD_BIN_DIR)/httpPassword$(BLD_EXE) \
+	@bld  --executable $(BLD_BIN_DIR)/httpPassword$(BLD_EXE) \
 		--preferStatic \
-		--objects httpPassword$(BLD_OBJ)
+		--objects httpPassword$(BLD_OBJ) \
+		--libs "stdc++"
 
 cleanExtra:
 	@rm -f $(BLD_BIN_DIR)/charGen$(BLD_EXE) 
