--- openldap/libraries/liblunicode/Makefile.in~	2005-01-20 09:01:03.000000000 -0800
+++ openldap/libraries/liblunicode/Makefile.in	2005-02-20 13:45:58.009795269 -0800
@@ -29,8 +29,8 @@
 LDAP_INCDIR= ../../include       
 LDAP_LIBDIR= ../../libraries
 
-ucgendat: $(XLIBS) ucgendat.o
-	$(LTLINK) -o $@ ucgendat.o $(LIBS)
+ucgendat: $(XLIBS) .links
+	$(HOSTCC) -o $@ -I$(top_srcdir)/include ucgendat.c ../liblutil/utils.c
 	./ucgendat $(srcdir)/UnicodeData.txt -x $(srcdir)/CompositionExclusions.txt
 
 DATFILES = case.dat cmbcl.dat comp.dat ctype.dat decomp.dat num.dat kdecomp.dat
