--- openldap/clients/tools/Makefile.in~	2005-01-20 09:00:55.000000000 -0800
+++ openldap/clients/tools/Makefile.in	2005-06-01 08:32:48.000000000 -0700
@@ -98,7 +98,7 @@
 	-$(MKDIR) $(DESTDIR)$(bindir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(bindir);				\
 	    done							\
 	)
--- openldap/servers/slapd/Makefile.in~	2005-01-20 09:01:05.000000000 -0800
+++ openldap/servers/slapd/Makefile.in	2005-06-01 08:35:55.000000000 -0700
@@ -355,7 +355,7 @@
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	-$(MKDIR) $(DESTDIR)$(localstatedir)/run
-	$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 \
+	$(LTINSTALL) $(INSTALLFLAGS) -m 755 \
 		slapd$(EXEEXT) $(DESTDIR)$(libexecdir)
 	@for i in $(SUBDIRS); do \
 	    if test -d $$i -a -f $$i/Makefile ; then \
--- openldap/servers/slurpd/Makefile.in~	2005-01-20 09:01:19.000000000 -0800
+++ openldap/servers/slurpd/Makefile.in	2005-06-01 08:36:30.000000000 -0700
@@ -60,7 +60,7 @@
 	@-$(INSTALL) -m 700 -d $(DESTDIR)$(localstatedir)/openldap-slurp
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) -s -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(libexecdir);				\
 	    done							\
 	)
