--- a/Makefile.in
+++ b/Makefile.in
@@ -40,18 +40,18 @@
 
 # Defaults
 CC = @CC@
-LD = @CC@
+LD = @CC@ $(LDFLAGS)
 STRIP = @STRIP@
 RANLIB = @RANLIB@
 
 # make 'modegg'
 MOD_CC = @MOD_CC@
-MOD_LD = @MOD_LD@
+MOD_LD = @MOD_LD@ $(LDFLAGS)
 MOD_STRIP = @MOD_STRIP@
 
 # make 'modules'
 SHLIB_CC = @SHLIB_CC@
-SHLIB_LD = @SHLIB_LD@
+SHLIB_LD = @SHLIB_LD@ $(LDFLAGS)
 SHLIB_STRIP = @SHLIB_STRIP@
 MOD_EXT = @MOD_EXT@
 
@@ -77,7 +77,7 @@
 
 modconf = $(top_srcdir)/misc/modconfig --top_srcdir=$(top_srcdir)
 
-egg_test_run = EGG_LANGDIR=$(top_srcdir)/language ./$(EGGEXEC) -v
+egg_test_run = EGG_LANGDIR=$(top_srcdir)/language #./$(EGGEXEC) -v
 
 post_config  =  echo "" && \
 		echo "You can now compile the bot, using \"make\"." && \
