--- include/net-snmp/library/tools.h.orig	2006-09-01 10:38:16.000000000 -0700
+++ include/net-snmp/library/tools.h	2007-12-15 22:24:11.000000000 -0800
@@ -50,7 +50,7 @@
 
 /** @def SNMP_FREE(s)
     Frees a pointer only if it is !NULL and sets its value to NULL */
-#define SNMP_FREE(s)    do { if (s) { free((void *)s); s=NULL; } } while(0)
+#define SNMP_FREE(s)    do { if (s) { free(s); s=NULL; } } while(0)
 
 /** @def SNMP_SWIPE_MEM(n, s)
     Frees pointer n only if it is !NULL, sets n to s and sets s to NULL */
