--- a/unix/plugins/CameraPlugin/sqCamera-linux.c
+++ b/unix/plugins/CameraPlugin/sqCamera-linux.c
@@ -49,6 +49,12 @@
 
 #include <linux/videodev2.h>
 
+#ifndef V4L2_PIX_FMT_RGB444
+#define v4l2_fourcc(a, b, c, d)\
+         ((__u32)(a) | ((__u32)(b) << 8) | ((__u32)(c) << 16) | ((__u32)(d) << 24))
+#define V4L2_PIX_FMT_RGB444  v4l2_fourcc('R', '4', '4', '4')
+#endif
+
 
 #define sqInt int
 #define true 1
--- a/unix/plugins/WeDoPlugin/WeDoLinux.c
+++ b/unix/plugins/WeDoPlugin/WeDoLinux.c
@@ -34,7 +34,7 @@
 #include <time.h>
 #include <sys/ioctl.h>
 #include <errno.h>
-#include <sys/types.h>
+#include <linux/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <linux/hiddev.h>
