--- Squeak-3.9-7/platforms/unix/vm/sqUnixMain.c	2006-04-25 05:07:18.000000000 +0200
+++ squeak/platforms/unix/vm/sqUnixMain.c	2006-05-29 13:30:35.000000000 +0200
@@ -161,8 +161,8 @@
 #      else
 	sa.sa_flags= 0;	/* assume we already have BSD behaviour */
 #      endif
-#      if defined(__linux__) && !defined(__ia64)
-	sa.sa_restorer= 0;
+#      if defined(__linux__) && !defined(__ia64) && !defined(__UCLIBC__)
+	sa.sa_restorer= 0; 
 #      endif
 	sigaction(SIGALRM, &sa, 0);
       }
