--- ncftp/configure.orig	2005-01-12 21:30:44.000000000 +0100
+++ ncftp/configure	2005-01-12 21:31:32.000000000 +0100
@@ -6281,47 +6281,7 @@
 
 echo $ac_n "checking whether setpgrp takes no argument""... $ac_c" 1>&6
 echo "configure:6284: checking whether setpgrp takes no argument" >&5
-if eval "test \"`echo '$''{'ac_cv_func_setpgrp_void'+set}'`\" = set"; then
-  echo $ac_n "(cached) $ac_c" 1>&6
-else
-  if test "$cross_compiling" = yes; then
-  { echo "configure: error: cannot check setpgrp if cross compiling" 1>&2; exit 1; }
-else
-  cat > conftest.$ac_ext <<EOF
-#line 6292 "configure"
-#include "confdefs.h"
-
-#ifdef HAVE_UNISTD_H
-#include <unistd.h>
-#endif
-
-/*
- * If this system has a BSD-style setpgrp, which takes arguments, exit
- * successfully.
- */
-main()
-{
-    if (setpgrp(1,1) == -1)
-	exit(0);
-    else
-	exit(1);
-}
-
-EOF
-if { (eval echo configure:6312: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext} && (./conftest; exit) 2>/dev/null
-then
-  ac_cv_func_setpgrp_void=no
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -fr conftest*
-  ac_cv_func_setpgrp_void=yes
-fi
-rm -fr conftest*
-fi
-
-
-fi
+ac_cv_func_setpgrp_void=yes
 
 echo "$ac_t""$ac_cv_func_setpgrp_void" 1>&6
 if test $ac_cv_func_setpgrp_void = yes; then
