--- ipkg-opt/ipkg_download.c~	2008-05-21 14:07:34.000000000 +0930
+++ ipkg-opt/ipkg_download.c	2008-05-21 14:05:19.000000000 +0930
@@ -69,7 +69,7 @@
     }
 
     /* XXX: BUG rewrite to use execvp or else busybox's internal wget -Jamey 7/23/2002 */ 
-    sprintf_alloc(&cmd, "wget --passive-ftp %s %s%s %s%s %s -P %s %s",
+    sprintf_alloc(&cmd, "/opt/bin/wget --passive-ftp %s %s%s %s%s %s -P %s %s",
 		  (conf->http_proxy || conf->ftp_proxy) ? "--proxy=on" : "",
 		  conf->proxy_user ? "--proxy-user=" : "",
 		  conf->proxy_user ? conf->proxy_user : "",
