--- a/scheduler/conf.c
+++ b/scheduler/conf.c
@@ -593,7 +593,7 @@
   cupsdSetStringf(&ServerHeader, "CUPS/%d.%d IPP/2.1", CUPS_VERSION_MAJOR,
                   CUPS_VERSION_MINOR);
   cupsdSetString(&StateDir, CUPS_STATEDIR);
-  cupsdSetString(&PidFile, "/run/cups/cupsd.pid");
+  cupsdSetString(&PidFile, "%OPTWARE_TARGET_PREFIX%/var/run/cups/cupsd.pid");
 
   if (!strcmp(CUPS_DEFAULT_PRINTCAP, "/etc/printers.conf"))
     PrintcapFormat = PRINTCAP_SOLARIS;
