--- lxappearance-0.6.1/src/lxappearance.c.orig	2014-10-23 22:36:13.000000000 +0300
+++ lxappearance-0.6.1/src/lxappearance.c	2015-03-30 17:31:47.920416740 +0300
@@ -240,17 +240,16 @@ static void lxappearance_save_gtkrc()
         "GTK_ICON_SIZE_DIALOG"
     };
 
-    char* file_path = g_build_filename(g_get_home_dir(), ".gtkrc-2.0", NULL);
+    char* file_path = g_build_filename("%OPTWARE_TARGET_PREFIX%/etc/gtk-2.0/gtkrc", NULL);
     GString* content = g_string_sized_new(512);
     g_string_append(content,
         "# DO NOT EDIT! This file will be overwritten by LXAppearance.\n"
-        "# Any customization should be done in ~/.gtkrc-2.0.mine instead.\n\n");
+        "# Any customization should be done in %OPTWARE_TARGET_PREFIX%/etc/gtk-2.0/gtkrc.user instead.\n\n");
 
-    /* include ~/.gtkrc-2.0.mine first to be able to apply changes done
+    /* include %OPTWARE_TARGET_PREFIX%/etc/gtk-2.0/gtkrc.user first to be able to apply changes done
        by LXAppearance if the same settings exist in that file */
     g_string_append_printf(content,
-        "include \"%s/.gtkrc-2.0.mine\"\n",
-        g_get_home_dir());
+        "include \"%OPTWARE_TARGET_PREFIX%/etc/gtk-2.0/gtkrc.user\"\n");
 
     if(app.widget_theme)
         g_string_append_printf(content,
@@ -323,7 +322,7 @@ static void lxappearance_save_gtkrc()
        Content shold be different from the gtk2 one
     */
     GKeyFile *content_gtk3 = g_key_file_new();
-    char* file_path_gtk3 = g_build_filename(g_get_user_config_dir(), "gtk-3.0", NULL);
+    char* file_path_gtk3 = g_build_filename("%OPTWARE_TARGET_PREFIX%/etc/gtk-3.0", NULL);
     char* file_path_settings = g_build_filename(file_path_gtk3, "settings.ini", NULL);
 
     if (!g_file_test(file_path_gtk3, G_FILE_TEST_IS_DIR))
@@ -584,7 +583,7 @@ static void settings_init()
     /* FIXME: we should support other xsettings daemons too */
     else
     {
-        char* gtkrc_file = g_build_filename(g_get_home_dir(), ".gtkrc-2.0", NULL);
+        char* gtkrc_file = "%OPTWARE_TARGET_PREFIX%/etc/gtk-2.0/gtkrc";
         gtkrc_file_get_color_scheme(gtkrc_file, app.color_scheme_hash);
         g_free(gtkrc_file);
         if(g_hash_table_size(app.color_scheme_hash) > 0)
