*** libxml2/configure.orig	Sun Mar 13 17:18:18 2005
--- libxml2/configure	Sun Mar 13 17:23:36 2005
***************
*** 8343,8347 ****
  version_type=none
  dynamic_linker="$host_os ld.so"
! sys_lib_dlsearch_path_spec="/lib /usr/lib"
  if test "$GCC" = yes; then
    sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
--- 8343,8347 ----
  version_type=none
  dynamic_linker="$host_os ld.so"
! sys_lib_dlsearch_path_spec=""
  if test "$GCC" = yes; then
    sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
***************
*** 8356,8360 ****
    fi
  else
!   sys_lib_search_path_spec="/lib /usr/lib /usr/local/lib"
  fi
  need_lib_prefix=unknown
--- 8356,8360 ----
    fi
  else
!   sys_lib_search_path_spec=""
  fi
  need_lib_prefix=unknown
***************
*** 8665,8670 ****
    shlibpath_var=LD_LIBRARY${shlibsuff}_PATH
    shlibpath_overrides_runpath=no
!   sys_lib_search_path_spec="/usr/lib${libsuff} /lib${libsuff} /usr/local/lib${libsuff}"
!   sys_lib_dlsearch_path_spec="/usr/lib${libsuff} /lib${libsuff}"
    hardcode_into_libs=yes
    ;;
--- 8665,8670 ----
    shlibpath_var=LD_LIBRARY${shlibsuff}_PATH
    shlibpath_overrides_runpath=no
!   sys_lib_search_path_spec=""
!   sys_lib_dlsearch_path_spec=""
    hardcode_into_libs=yes
    ;;
***************
*** 8714,8718 ****
    if test -f /etc/ld.so.conf; then
      lt_ld_extra=`$SED -e 's/:,\t/ /g;s/=^=*$//;s/=^= * / /g' /etc/ld.so.conf | tr '\n' ' '`
!     sys_lib_dlsearch_path_spec="/lib${libsuff} /usr/lib${libsuff} $lt_ld_extra"
    fi
  
--- 8714,8718 ----
    if test -f /etc/ld.so.conf; then
      lt_ld_extra=`$SED -e 's/:,\t/ /g;s/=^=*$//;s/=^= * / /g' /etc/ld.so.conf | tr '\n' ' '`
!     sys_lib_dlsearch_path_spec=""
    fi
  
***************
*** 12062,12066 ****
  version_type=none
  dynamic_linker="$host_os ld.so"
! sys_lib_dlsearch_path_spec="/lib /usr/lib"
  if test "$GCC" = yes; then
    sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
--- 12062,12066 ----
  version_type=none
  dynamic_linker="$host_os ld.so"
! sys_lib_dlsearch_path_spec=""
  if test "$GCC" = yes; then
    sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
***************
*** 12075,12079 ****
    fi
  else
!   sys_lib_search_path_spec="/lib /usr/lib /usr/local/lib"
  fi
  need_lib_prefix=unknown
--- 12075,12079 ----
    fi
  else
!   sys_lib_search_path_spec=""
  fi
  need_lib_prefix=unknown
***************
*** 12384,12389 ****
    shlibpath_var=LD_LIBRARY${shlibsuff}_PATH
    shlibpath_overrides_runpath=no
!   sys_lib_search_path_spec="/usr/lib${libsuff} /lib${libsuff} /usr/local/lib${libsuff}"
!   sys_lib_dlsearch_path_spec="/usr/lib${libsuff} /lib${libsuff}"
    hardcode_into_libs=yes
    ;;
--- 12384,12389 ----
    shlibpath_var=LD_LIBRARY${shlibsuff}_PATH
    shlibpath_overrides_runpath=no
!   sys_lib_search_path_spec=""
!   sys_lib_dlsearch_path_spec=""
    hardcode_into_libs=yes
    ;;
***************
*** 12433,12437 ****
    if test -f /etc/ld.so.conf; then
      lt_ld_extra=`$SED -e 's/:,\t/ /g;s/=^=*$//;s/=^= * / /g' /etc/ld.so.conf | tr '\n' ' '`
!     sys_lib_dlsearch_path_spec="/lib${libsuff} /usr/lib${libsuff} $lt_ld_extra"
    fi
  
--- 12433,12437 ----
    if test -f /etc/ld.so.conf; then
      lt_ld_extra=`$SED -e 's/:,\t/ /g;s/=^=*$//;s/=^= * / /g' /etc/ld.so.conf | tr '\n' ' '`
!     sys_lib_dlsearch_path_spec=""
    fi
  
***************
*** 15232,15236 ****
  version_type=none
  dynamic_linker="$host_os ld.so"
! sys_lib_dlsearch_path_spec="/lib /usr/lib"
  if test "$GCC" = yes; then
    sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
--- 15232,15236 ----
  version_type=none
  dynamic_linker="$host_os ld.so"
! sys_lib_dlsearch_path_spec=""
  if test "$GCC" = yes; then
    sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
***************
*** 15245,15249 ****
    fi
  else
!   sys_lib_search_path_spec="/lib /usr/lib /usr/local/lib"
  fi
  need_lib_prefix=unknown
--- 15245,15249 ----
    fi
  else
!   sys_lib_search_path_spec=""
  fi
  need_lib_prefix=unknown
***************
*** 15592,15596 ****
        *64-bit*)
          libsuff=64
!         sys_lib_search_path_spec="/lib${libsuff} /usr/lib${libsuff} /usr/local/lib${libsuff}"
          ;;
        esac
--- 15592,15596 ----
        *64-bit*)
          libsuff=64
!         sys_lib_search_path_spec=""
          ;;
        esac
***************
*** 15603,15607 ****
    if test -f /etc/ld.so.conf; then
      lt_ld_extra=`$SED -e 's/:,\t/ /g;s/=^=*$//;s/=^= * / /g' /etc/ld.so.conf | tr '\n' ' '`
!     sys_lib_dlsearch_path_spec="/lib${libsuff} /usr/lib${libsuff} $lt_ld_extra"
    fi
  
--- 15603,15607 ----
    if test -f /etc/ld.so.conf; then
      lt_ld_extra=`$SED -e 's/:,\t/ /g;s/=^=*$//;s/=^= * / /g' /etc/ld.so.conf | tr '\n' ' '`
!     sys_lib_dlsearch_path_spec=""
    fi
  
***************
*** 17600,17604 ****
  version_type=none
  dynamic_linker="$host_os ld.so"
! sys_lib_dlsearch_path_spec="/lib /usr/lib"
  if test "$GCC" = yes; then
    sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
--- 17600,17604 ----
  version_type=none
  dynamic_linker="$host_os ld.so"
! sys_lib_dlsearch_path_spec=""
  if test "$GCC" = yes; then
    sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
***************
*** 17613,17617 ****
    fi
  else
!   sys_lib_search_path_spec="/lib /usr/lib /usr/local/lib"
  fi
  need_lib_prefix=unknown
--- 17613,17617 ----
    fi
  else
!   sys_lib_search_path_spec=""
  fi
  need_lib_prefix=unknown
***************
*** 17960,17964 ****
        *64-bit*)
          libsuff=64
!         sys_lib_search_path_spec="/lib${libsuff} /usr/lib${libsuff} /usr/local/lib${libsuff}"
          ;;
        esac
--- 17960,17964 ----
        *64-bit*)
          libsuff=64
!         sys_lib_search_path_spec=""
          ;;
        esac
***************
*** 17971,17975 ****
    if test -f /etc/ld.so.conf; then
      lt_ld_extra=`$SED -e 's/:,\t/ /g;s/=^=*$//;s/=^= * / /g' /etc/ld.so.conf | tr '\n' ' '`
!     sys_lib_dlsearch_path_spec="/lib${libsuff} /usr/lib${libsuff} $lt_ld_extra"
    fi
  
--- 17971,17975 ----
    if test -f /etc/ld.so.conf; then
      lt_ld_extra=`$SED -e 's/:,\t/ /g;s/=^=*$//;s/=^= * / /g' /etc/ld.so.conf | tr '\n' ' '`
!     sys_lib_dlsearch_path_spec=""
    fi
  
