diff -Naur mysql-5.1.11-beta.org/configure.in mysql-5.1.11-beta/configure.in
--- mysql-5.1.11-beta.org/configure.in	2006-06-02 20:01:10.000000000 +0200
+++ mysql-5.1.11-beta/configure.in	2006-08-12 23:11:00.000000000 +0200
@@ -231,6 +231,9 @@
   *netware*)
     AC_DEFINE(SPRINTF_RETURNS_INT, [1]) AC_MSG_RESULT("int")
     ;;
+  *linux*)
+    AC_DEFINE(SPRINTF_RETURNS_INT, [1]) AC_MSG_RESULT("int")
+    ;;
   *)
 AC_TRY_RUN([
   int main()
@@ -783,38 +786,6 @@
   AC_LANG_SAVE
   AC_LANG_CPLUSPLUS
 
-  atom_ops=
-  AC_TRY_RUN([
-#include <asm/atomic.h>
-int main()
-{
-  atomic_t v;
-
-  atomic_set(&v, 23);
-  atomic_add(5, &v);
-  return atomic_read(&v) == 28 ? 0 : -1;
-}
-  ],
-  [AC_DEFINE([HAVE_ATOMIC_ADD], [1],
-             [atomic_add() from <asm/atomic.h> (Linux only)])
-   atom_ops="${atom_ops}atomic_add "],
-  )
-  AC_TRY_RUN([
-#include <asm/atomic.h>
-int main()
-{
-  atomic_t v;
-
-  atomic_set(&v, 23);
-  atomic_sub(5, &v);
-  return atomic_read(&v) == 18 ? 0 : -1;
-}
-  ],
-  [AC_DEFINE([HAVE_ATOMIC_SUB], [1],
-             [atomic_sub() from <asm/atomic.h> (Linux only)])
-   atom_ops="${atom_ops}atomic_sub "],
-  )
-
   if test -z "$atom_ops"; then atom_ops="no"; fi
   AC_MSG_RESULT($atom_ops)
 
@@ -876,6 +847,9 @@
   *netware)
     AC_MSG_RESULT([no])
     ;;
+  *linux*)
+    AC_MSG_RESULT([no])
+    ;;
   *)
 AC_TRY_RUN([
 #ifdef HAVE_STDLIB_H
@@ -1194,7 +1168,7 @@
           # old method, check headers
           # Look for LinuxThreads.
           AC_MSG_CHECKING("LinuxThreads in header file comment")
-          res=`grep Linuxthreads /usr/include/pthread.h 2>/dev/null | wc -l`
+          res=1#`grep Linuxthreads /usr/include/pthread.h 2>/dev/null | wc -l`
           if test "$res" -gt 0
           then
             AC_MSG_RESULT("Found")
@@ -1527,12 +1501,12 @@
 
 
 # System characteristics
-case $SYSTEM_TYPE in
-  *netware*) ;;
-  *)
-AC_SYS_RESTARTABLE_SYSCALLS
-    ;;
-esac
+#case $SYSTEM_TYPE in
+#  *netware*) ;;
+#  *)
+#AC_SYS_RESTARTABLE_SYSCALLS
+#    ;;
+#esac
 
 # Build optimized or debug version ?
 # First check for gcc and g++
@@ -1774,20 +1748,6 @@
 MYSQL_CHECK_IN_ADDR_T
 # Do the c++ compiler have a bool type
 MYSQL_CXX_BOOL
-# Check some common bugs with gcc 2.8.# on sparc
-if ! ( expr "$SYSTEM_TYPE" : ".*netware.*" > /dev/null ); then
-MYSQL_CHECK_LONGLONG_TO_FLOAT
-if test "$ac_cv_conv_longlong_to_float" != "yes"
-then
-  AC_MSG_ERROR([Your compiler cannot convert a longlong value to a float!
-If you are using gcc 2.8.# you should upgrade to egcs 1.0.3 or newer and try
-again]);
-fi
-fi
-AC_CHECK_TYPES([sigset_t, off_t], [], [], [#include <sys/types.h>])
-AC_CHECK_TYPES([size_t], [], [], [#include <stdio.h>])
-AC_CHECK_TYPES([u_int32_t])
-
 MYSQL_PTHREAD_YIELD
 
 ######################################################################
