Index: configure.ac
===================================================================
RCS file: /cvs/xlibs/X11/configure.ac,v
retrieving revision 1.31
diff -u -r1.31 configure.ac
--- configure.ac	23 Nov 2004 01:48:35 -0000	1.31
+++ configure.ac	14 Feb 2005 04:50:48 -0000
@@ -149,7 +149,7 @@
 #
 KEYSYMDEF=""
 AC_MSG_CHECKING([keysymdef.h])
-for flag in $XPROTO_CFLAGS -I/usr/include; do
+for flag in $XPROTO_CFLAGS $CPPFLAGS; do
 	case "$KEYSYMDEF" in
 	"")
 		case "$flag" in
