--- cyrus-imapd/tools/config2header.orig	2006-12-28 16:39:16.000000000 +0100
+++ cyrus-imapd/tools/config2header	2006-12-28 16:39:47.000000000 +0100
@@ -214,7 +214,6 @@
 print HFILE <<EOF
   IMAPOPT_LAST
 };
-extern struct imapopt_s imapopts[];
 
 enum enum_value {
   IMAP_ENUM_ZERO = 0,
@@ -260,6 +259,9 @@
 
 print HFILE <<EOF
 };
+
+extern struct imapopt_s imapopts[];
+
 #endif /* INCLUDED_IMAPOPTIONS_H */
 EOF
     ;
