--- units-1.85/Makefile.in	2005-04-24 11:14:54.000000000 -0700
+++ units/Makefile.in	2005-09-05 00:04:02.799912768 -0700
@@ -82,7 +82,9 @@
 	$(INSTALL_DATA) $(srcdir)/units.dat $(datadir)/units.dat
 
 install-strip: units units.dat install-doc
-	$(INSTALL_PROGRAM) -s units $(bindir)/`echo units|sed '$(transform)'`
+	$(srcdir)/mkinstalldirs $(bindir) $(datadir)
+	$(INSTALL_PROGRAM) units $(bindir)/`echo units|sed '$(transform)'`
+	$(STRIP) $(bindir)/units
 	$(INSTALL_DATA) $(srcdir)/units.dat $(datadir)/units.dat
 
 install-doc: install-man install-info
