--- BitchX/Makefile.in	2003-04-10 18:09:04.000000000 -0700
+++ bitchx/Makefile.in	2005-03-21 15:17:10.000000000 -0800
@@ -75,6 +75,7 @@
 # Set this to -g if you want to be able to debug the client, otherwise
 # use -O to have the compiler do some optimization instead.
 CFLAGS = @CFLAGS@
+CPPFLAGS = @CPPFLAGS@
 
 # Set this to -s if you want the binary to be stripped.
 LDFLAGS = @LDFLAGS@
@@ -159,7 +160,7 @@
 	'INSTALL_IRC=$(INSTALL_IRC)'			\
 	'IRCLIB=$(IRCLIB)'				\
 	'CC=$(CC)'					\
-	'CFLAGS=$(CFLAGS)'				\
+	'CFLAGS=$(CFLAGS) $(CPPFLAGS)'				\
 	'HELPDIR=$(HELPDIR)'				\
         'INSTALL_WSERV=$(INSTALL_WSERV)'                \
 	'IRCPATH=$(IRCPATH)'				\
