--- stunnel/configure.orig	2012-03-26 12:52:47.000000000 +0400
+++ stunnel/configure	2013-01-27 00:04:47.985393890 +0400
@@ -11255,14 +11255,8 @@
 if eval "test \"\${$as_ac_File+set}\"" = set; then :
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r ""/dev/ptmx""; then
-  eval "$as_ac_File=yes"
-else
   eval "$as_ac_File=no"
 fi
-fi
 eval ac_res=\$$as_ac_File
 	       { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_res" >&5
 $as_echo "$ac_res" >&6; }
@@ -11278,14 +11272,8 @@
 if eval "test \"\${$as_ac_File+set}\"" = set; then :
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r ""/dev/ptc""; then
-  eval "$as_ac_File=yes"
-else
   eval "$as_ac_File=no"
 fi
-fi
 eval ac_res=\$$as_ac_File
 	       { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_res" >&5
 $as_echo "$ac_res" >&6; }
@@ -11333,13 +11321,7 @@
 if eval "test \"\${$as_ac_File+set}\"" = set; then :
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r ""/dev/urandom""; then
   eval "$as_ac_File=yes"
-else
-  eval "$as_ac_File=no"
-fi
 fi
 eval ac_res=\$$as_ac_File
 	       { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_res" >&5
