--- stunnel/configure.orig	2008-05-22 15:08:14.000000000 +0930
+++ stunnel/configure	2008-07-26 23:38:49.000000000 +0930
@@ -20906,16 +20906,8 @@
 if test "${ac_cv_file___dev_ptmx_+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
-if test -r ""/dev/ptmx""; then
-  ac_cv_file___dev_ptmx_=yes
-else
   ac_cv_file___dev_ptmx_=no
 fi
-fi
 { echo "$as_me:$LINENO: result: $ac_cv_file___dev_ptmx_" >&5
 echo "${ECHO_T}$ac_cv_file___dev_ptmx_" >&6; }
 if test $ac_cv_file___dev_ptmx_ = yes; then
@@ -20930,16 +20922,8 @@
 if test "${ac_cv_file___dev_ptc_+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
-if test -r ""/dev/ptc""; then
-  ac_cv_file___dev_ptc_=yes
-else
   ac_cv_file___dev_ptc_=no
 fi
-fi
 { echo "$as_me:$LINENO: result: $ac_cv_file___dev_ptc_" >&5
 echo "${ECHO_T}$ac_cv_file___dev_ptc_" >&6; }
 if test $ac_cv_file___dev_ptc_ = yes; then
@@ -20980,15 +20964,7 @@
 if test "${ac_cv_file___dev_urandom_+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
-if test -r ""/dev/urandom""; then
   ac_cv_file___dev_urandom_=yes
-else
-  ac_cv_file___dev_urandom_=no
-fi
 fi
 { echo "$as_me:$LINENO: result: $ac_cv_file___dev_urandom_" >&5
 echo "${ECHO_T}$ac_cv_file___dev_urandom_" >&6; }
