--- mc/doc/Makefile.in.orig	2007-07-17 15:50:54.000000000 -0700
+++ mc/doc/Makefile.in	2007-07-17 15:51:23.000000000 -0700
@@ -637,7 +637,7 @@
 
 
 mc.hlp: mc.1 $(srcdir)/xnc.hlp $(top_builddir)/src/man2hlp
-	- $(top_builddir)/src/man2hlp mc.1 $(srcdir)/xnc.hlp mc.hlp
+	- $(top_builddir)/src/man2hlp.host mc.1 $(srcdir)/xnc.hlp mc.hlp
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
