--- mc-4.7.0.8/src/man2hlp.c.orig	2010-08-06 07:57:50.000000000 +0000
+++ mc-4.7.0.8/src/man2hlp.c	2010-09-07 02:17:28.025294076 +0000
@@ -23,15 +23,11 @@
  *  \brief Source: man page to help file converter
  */
 
-#include <config.h>
-
 #include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
-#include <glib.h>
-
 #include "help.h"
 
 #define BUFFER_SIZE 256
@@ -150,7 +146,7 @@
 
     f = fopen (filename, flags);
     if (f == NULL) {
-	g_snprintf (tmp, sizeof (tmp), "man2hlp: Cannot open file \"%s\"",
+	snprintf (tmp, sizeof (tmp), "man2hlp: Cannot open file \"%s\"",
 		    filename);
 	perror (tmp);
 	exit (3);
@@ -291,7 +287,7 @@
     char buffer[BUFFER_SIZE];
 
     va_start (args, format);
-    g_vsnprintf (buffer, sizeof (buffer), format, args);
+    vsnprintf (buffer, sizeof (buffer), format, args);
     va_end (args);
     print_string (buffer);
 }
@@ -584,7 +580,7 @@
     } else {
 	/* Other commands are ignored */
 	char warn_str[BUFFER_SIZE];
-	g_snprintf (warn_str, sizeof (warn_str),
+	snprintf (warn_str, sizeof (warn_str),
 		    "Warning: unsupported command %s", buffer);
 	print_error (warn_str);
 	return;
@@ -615,7 +611,7 @@
 	/* Bold text or italics text */
 	if (buffer[0] == '.' && (buffer[1] == 'I' || buffer[1] == 'B'))
 	    for (buffer += 2; *buffer == ' ' || *buffer == '\t'; buffer++);
-	g_strlcpy (old, buffer, sizeof (old));
+	strncpy (old, buffer, sizeof (old));
 	link_flag = 3;
 	break;
     case 3:
@@ -813,7 +809,7 @@
 	    }
 	}
 	if (!found) {
-	    g_snprintf (buffer, sizeof (buffer), "Stale link \"%s\"",
+	    snprintf (buffer, sizeof (buffer), "Stale link \"%s\"",
 			current_link->linkname);
 	    c_in = current_link->filename;
 	    in_row = current_link->line;
