diff -Nur lha.orig/src/header.c lha/src/header.c
--- lha.orig/src/header.c	2006-08-26 12:46:36.000000000 +0200
+++ lha/src/header.c	2006-08-26 12:47:48.000000000 +0200
@@ -656,8 +656,17 @@
 	}
 
 	if (dir_length) {
+		if ((dir_length + name_length) > sizeof(dirname)) {
+			fprintf(stderr, "Insufficient buffer size\n");
+			exit(112);
+		}
 		strcat(dirname, hdr->name);
-		strcpy(hdr->name, dirname);
+
+		if ((dir_length + name_length) > sizeof(hdr->name)) {
+			fprintf(stderr, "Insufficient buffer size\n");
+			exit(112);
+		}
+		strncpy(hdr->name, dirname, sizeof(hdr->name));
 		name_length += dir_length;
 	}
 
