--- Zend/zend_strtod.c.orig	2006-12-26 18:17:52.000000000 +0100
+++ Zend/zend_strtod.c	2007-03-27 12:31:10.000000000 +0200
@@ -274,7 +274,7 @@
  *  * An alternative that might be better on some machines is
  *   * #define Storeinc(a,b,c) (*a++ = b << 16 | c & 0xffff)
  *    */
-#if defined(IEEE_LITTLE_ENDIAN) + defined(VAX) + defined(__arm__)
+#if defined(IEEE_LITTLE_ENDIAN) + defined(VAX)
 #define Storeinc(a,b,c) (((unsigned short *)a)[1] = (unsigned short)b, \
 		((unsigned short *)a)[0] = (unsigned short)c, a++)
 #else
