--- roundup-0.8.3/setup.py	2005-05-01 23:09:38.000000000 -0700
+++ py-roundup/setup.py	2005-07-08 22:24:37.115415000 -0700
@@ -102,10 +102,7 @@
         if self.python_executable:
             # TODO? allow command-line option
             pass
-        if target == sys.platform:
-            self.python_executable = os.path.normpath(sys.executable)
-        else:
-            self.python_executable = "python"
+        self.python_executable = "/opt/bin/python"
 
         # for windows builds, add ".bat" extension
         if target == "win32":
@@ -115,16 +112,7 @@
             #    for script in self.scripts]
             self.scripts = [script + ".bat" for script in self.scripts]
 
-        # tweak python path for installations outside main python library
-        if cmdopt.get("install", {}).has_key("prefix"):
-            prefix = os.path.expanduser(cmdopt['install']['prefix'][1])
-            version = '%d.%d'%sys.version_info[:2]
-            self.script_preamble = '''
-import sys
-sys.path.insert(1, "%s/lib/python%s/site-packages")
-'''%(prefix, version)
-        else:
-            self.script_preamble = ''
+        self.script_preamble = ''
 
     def copy_scripts(self):
         """ Create each script listed in 'self.scripts'
