--- irssi/configure.in.orig	2006-08-22 16:38:26.000000000 -0700
+++ irssi/configure.in	2006-08-22 16:39:51.000000000 -0700
@@ -281,21 +281,7 @@
 
 dnl * older autoconfs don't include sys/types.h, so do it manually
 AC_MSG_CHECKING([size of off_t])
-AC_TRY_RUN([
-  #include <stdio.h>
-  #include <sys/types.h>
-  int main() {
-    FILE *f=fopen("conftestval", "w");
-    if (!f) exit(1);
-    fprintf(f, "%d\n", sizeof(off_t));
-    return 0;
-  }
-], [
-  sizeof_off_t=`cat conftestval`
-  rm -f conftestval
-], [
-  AC_ERROR([Unsupported off_t size])
-])
+sizeof_off_t=4
 AC_MSG_RESULT($sizeof_off_t)
 
 if test $sizeof_off_t = 8; then
