--- vsftpd/sysdeputil.c.orig	2005-04-02 17:34:15.491583096 +0400
+++ vsftpd/sysdeputil.c	2005-04-02 17:35:10.979147704 +0400
@@ -66,7 +66,9 @@
     #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,2,0))
       #define VSF_SYSDEP_HAVE_CAPABILITIES
       #define VSF_SYSDEP_HAVE_LINUX_SENDFILE
-      #include <sys/prctl.h>
+      #ifndef __UCLIBC__
+        #include <sys/prctl.h>
+      #endif
       #ifdef PR_SET_KEEPCAPS
         #define VSF_SYSDEP_HAVE_SETKEEPCAPS
       #endif
