--- pymssql-0.7.1-orig/setup.py	2005-07-23 01:04:22.000000000 -0700
+++ py-mssql/setup.py	2005-07-23 21:54:07.249695000 -0700
@@ -21,16 +21,6 @@
 	libraries = ["ntwdblib", "msvcrt", "kernel32", "user32", "gdi32", "winspool", "comdlg32", "advapi32", "shell32", "ole32", "oleaut32", "uuid", "odbc32", "odbccp32", ]
 	data_files = []
 else:	# try some generic paths
-	include_dirs = [
-		'/usr/include', '/usr/local/include',
-		'/usr/include/freetds', '/usr/local/include/freetds',
-		'/usr/pkg/freetds/include'	# netbsd2
-	]
-	library_dirs = [
-		'/usr/lib', '/usr/local/lib',
-		'/usr/lib/freetds', '/usr/local/lib/freetds'
-		'/usr/pkg/freetds/lib'		# netbsd2
-	]
 	libraries = ["sybdb"]
 	data_files = []
 
@@ -50,8 +40,6 @@
 	url="http://pymssql.sourceforge.net",
 	py_modules=[ 'pymssql', ],
 	ext_modules=[Extension("_mssql", ["mssqldbmodule.c"],
-			include_dirs = include_dirs,
-			library_dirs = library_dirs,
 			libraries=libraries)],
 	classifiers = filter(None, classifiers.split("\n")),
 	data_files=data_files
