--- gnuplot-4.2.rc2/Makefile.in.orig	2006-11-25 04:13:38.000000000 +0100
+++ gnuplot-4.2.rc2/Makefile.in	2006-12-21 11:55:44.000000000 +0100
@@ -194,7 +194,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 AUTOMAKE_OPTIONS = foreign 1.2h
-SUBDIRS = config m4 term src docs $(LISPDIR) man demo tutorial share
+SUBDIRS = config m4 term src docs $(LISPDIR) man share
 EXTRA_DIST = BUGS CodeStyle Copyright FAQ GNUmakefile INSTALL INSTALL.gnu \
 Makefile.maint PATCHLEVEL PGPKEYS PORTING README README.1ST README.exp \
 VERSION configure.vms pm3d
