--- eaccelerator/eaccelerator.h.orig	2006-12-15 23:14:56.000000000 +0100
+++ eaccelerator/eaccelerator.h	2006-12-15 23:15:51.000000000 +0100
@@ -164,7 +164,7 @@
 #define eaccelerator_malloc_nolock(size) mm_malloc_nolock(eaccelerator_mm_instance->mm, size)
 #define eaccelerator_free_nolock(x)      mm_free_nolock(eaccelerator_mm_instance->mm, x)
 
-#if (defined (__GNUC__) && __GNUC__ >= 2)
+#if (defined (__GNUC__) && __GNUC__ >= 2) && !defined (__UCLIBC__)
 #define EACCELERATOR_PLATFORM_ALIGNMENT (__alignof__ (align_test))
 #else
 #define EACCELERATOR_PLATFORM_ALIGNMENT (sizeof(align_union))
--- eaccelerator/ea_store.c.orig	2006-12-15 23:19:01.000000000 +0100
+++ eaccelerator/ea_store.c	2006-12-15 23:18:24.000000000 +0100
@@ -778,7 +778,7 @@
 	/* Check if this is a parent class. If so, copy unconditionally */
 	if (parent) {
 		/* unpack the \0classname\0membername\0 style property name to seperate vars */
-		zend_unmangle_property_name(p->arKey, &cname, &mname);
+		// TODO zend_unmangle_property_name(p->arKey, &cname, &mname);
 	
 		/* lookup the member's info in parent and child */
 		if((zend_hash_find(&parent->properties_info, mname, strlen(mname)+1, &pinfo.ptr) == SUCCESS) &&
