--- 2.5/setup.py.orig	2010-02-15 16:58:11.000000000 +0000
+++ 2.5/setup.py	2010-02-17 06:14:12.788104653 +0000
@@ -93,21 +93,7 @@
 if not version:
     version = 'unknown'
 
-try:
-    from svn import core
-    assert core.SVN_VER_MAJOR == 1
-    assert core.SVN_VER_MINOR >= 5
-except ImportError:
-    raise Exception('You need to install the SWIG'
-                    ' bindings for Subversion to use this package.')
-except AssertionError:
-    raise Exception('You need at least Subversion 1.5 to use this package.')
-
 requires = []
-try:
-    import mercurial
-except ImportError:
-    requires.append('mercurial')
 
 setup(
     name = 'hgsubversion',
