--- mem.c.orig	2007-09-29 08:53:34.000000000 -0700
+++ mem.c	2007-09-29 08:53:41.000000000 -0700
@@ -1,6 +1,7 @@
 /* (C) 2007 by folkert@vanheusden.com
  * The GPL (GNU public license) applies to this sourcecode.
  */
+#include <sys/types.h>
 #include <errno.h>
 #include <regex.h>
 #include <signal.h>
@@ -8,7 +9,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/types.h>
 #include <time.h>
 #include <unistd.h>
 
