--- sane-backends/Makefile.in	2004-05-04 11:44:26.000000000 -0400
+++ sane-backends-patched/Makefile.in	2005-05-11 12:43:20.870000000 -0400
@@ -33,7 +33,7 @@
 
 @SET_MAKE@
 
-SUBDIRS	= include lib sanei backend frontend tools doc po
+SUBDIRS	= include lib sanei backend frontend tools po
 
 DISTFILES = AUTHORS COPYING ChangeLog ChangeLog-1.0.0 ChangeLog-1.0.1 \
   ChangeLog-1.0.2 ChangeLog-1.0.3 ChangeLog-1.0.4 ChangeLog-1.0.5 \
