--- configure.old	2006-06-19 16:38:35.000000000 +1000
+++ configure	2006-06-19 16:40:27.000000000 +1000
@@ -6443,11 +6443,11 @@
 echo "$as_me:$LINENO: checking for working snprintf" >&5
 echo $ECHO_N "checking for working snprintf... $ECHO_C" >&6
 if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
+  { { echo "$as_me:$LINENO: warning: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
+echo "$as_me: warning: cannot run test program while cross compiling
 See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+ }
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
