--- irssi/configure.in.orig	2007-04-25 11:13:56.000000000 -0700
+++ irssi/configure.in	2007-04-26 16:34:17.000000000 -0700
@@ -483,13 +483,8 @@
   if test ! -s conftest; then
     AC_MSG_RESULT([no, error compiling test program])
   else
-    status="`./conftest`"
-    if test "x$status" = "xok"; then
       DYNLIB_MODULES=yes
       AC_MSG_RESULT([yes])
-    else
-      AC_MSG_RESULT([no, error running: $status])
-    fi
   fi
 fi
 rm -rf conftest conftest.* libconftest.* .libs
@@ -549,7 +544,9 @@
 	if test -z "$perlpath"; then
        		perl_check_error="perl binary not found"
 	else
+            if test -z "$PERL_CFLAGS"; then
 		PERL_CFLAGS=`$perlpath -MExtUtils::Embed -e ccopts 2>/dev/null`
+            fi
 	fi
 
 	if test "x$ac_cv_prog_gcc" = "xyes" -a -z "`echo $host_os|grep 'bsd\|linux'`"; then
@@ -572,7 +569,9 @@
 		AC_MSG_RESULT([not found, building without Perl])
 		want_perl=no
 	else
+            if test -z "$PERL_LDFLAGS"; then
 		PERL_LDFLAGS=`$perlpath -MExtUtils::Embed -e ldopts 2>/dev/null`
+            fi
 
 		dnl * Perl 5.004 and older use perl_xxx variables while
 		dnl * later use PL_perl_xxx variables ..
