--- a/setup.py
+++ setup.py
@@ -37,24 +37,6 @@
     raise SystemExit(
         "Couldn't import standard zlib (incomplete Python install).")
 
-# The base IronPython distribution (as of 2.7.1) doesn't support bz2
-isironpython = False
-try:
-    isironpython = (platform.python_implementation()
-                    .lower().find("ironpython") != -1)
-except AttributeError:
-    pass
-
-if isironpython:
-    sys.stderr.write("warning: IronPython detected (no bz2 support)\n")
-else:
-    try:
-        import bz2
-        bz2.BZ2Compressor # silence unused import warning
-    except ImportError:
-        raise SystemExit(
-            "Couldn't import standard bz2 (incomplete Python install).")
-
 ispypy = "PyPy" in sys.version
 
 import os, stat, subprocess, time
