--- setup.py.orig	2010-04-09 05:33:20.093753821 +0000
+++ setup.py	2010-04-09 05:32:04.429466138 +0000
@@ -26,12 +26,6 @@
     raise SystemExit(
         "Couldn't import standard zlib (incomplete Python install).")
 
-try:
-    import bz2
-except:
-    raise SystemExit(
-        "Couldn't import standard bz2 (incomplete Python install).")
-
 import os, subprocess, time
 import shutil
 import tempfile
