*** ftpd-topfield/ftpd.c.orig	Wed Mar 23 02:11:45 2005
--- ftpd-topfield/ftpd.c	Sun Apr  3 20:36:04 2005
***************
*** 1110,1119 ****
  {
  	const char *mode = (dirent->type == 'f') ? "-rw-r--r--" : "drwxr-xr-x";
  	struct tm *tm = localtime(&dirent->stamp);
! 	char timebuf[20];
! 	strftime(timebuf, sizeof(timebuf), "%b %e %Y", tm);
  
! 	fprintf(fh, "%s   1 unknown unknown %20llu %s %s", mode, dirent->size, timebuf, filename ?: dirent->name);
  }
  
  /*
--- 1110,1119 ----
  {
  	const char *mode = (dirent->type == 'f') ? "-rw-r--r--" : "drwxr-xr-x";
  	struct tm *tm = localtime(&dirent->stamp);
! 	char timebuf[32];
! 	strftime(timebuf, sizeof(timebuf), " %b %e %Y ", tm);
  
! 	fprintf(fh, "%s 1 unknown unknown %12llu %s %s\r\n", mode, dirent->size, timebuf, dirent->name);
  }
  
  /*
***************
*** 1137,1142 ****
--- 1137,1143 ----
  /**
   * Dump in EPLF format.
   */
+  /*
  static void dump_dirent_as_eplf(FILE *fh, const char *filename, tf_dirent *dirent)
  {
  	fprintf(fh, "+");
***************
*** 1152,1157 ****
--- 1153,1159 ----
  		fprintf(fh, "/,\t%s\r\n", dirent->name);
  	}
  }
+ */
  
  void
  statfilecmd (const char *filename)
***************
*** 1619,1625 ****
  					}
  					count++;
  					if (LIST) {
! 						dump_dirent_as_eplf(dout, 0, &entries.entry[i]);
  					}
  					else {
  						fprintf(dout, "%s\r\n", entries.entry[i].name);
--- 1621,1628 ----
  					}
  					count++;
  					if (LIST) {
! 						dump_dirent(dout, 0, &entries.entry[i]);
! // TZDBUM						dump_dirent_as_eplf(dout, 0, &entries.entry[i]);
  					}
  					else {
  						fprintf(dout, "%s\r\n", entries.entry[i].name);
