*** ftpd-topfield/ftpd.c.orig	Wed Mar 23 02:11:45 2005
--- ftpd-topfield/ftpd.c	Sun Apr  3 20:36:04 2005
***************
*** 115,121 ****
  char	*hostname;             /* Who we are.  */
  int	usedefault = 1;	       /* For data transfers.  */
  char	tmpline[7];            /* Temp buffer use in OOB.  */
! int	turbo = 0;
  
  jmp_buf  errcatch;
  
--- 115,121 ----
  char	*hostname;             /* Who we are.  */
  int	usedefault = 1;	       /* For data transfers.  */
  char	tmpline[7];            /* Temp buffer use in OOB.  */
! int	turbo = 1;
  
  jmp_buf  errcatch;
  
