--- qemu/dyngen.h~	2005-04-27 13:52:05.000000000 -0700
+++ qemu/dyngen.h	2005-07-24 23:03:15.000000000 -0700
@@ -19,7 +19,9 @@
  */
 
 int __op_param1, __op_param2, __op_param3;
-int __op_gen_label1, __op_gen_label2, __op_gen_label3;
+int __op_gen_label1 __attribute__ ((section ("text"))) =0;
+int __op_gen_label2 __attribute__ ((section ("text"))) =0;
+int __op_gen_label3 __attribute__ ((section ("text"))) =0;
 int __op_jmp0, __op_jmp1, __op_jmp2, __op_jmp3;
 
 #ifdef __i386__
