--- server/Makefile.in~	2005-02-04 09:40:14.000000000 -0800
+++ server/Makefile.in	2005-02-19 18:09:15.577570020 -0800
@@ -24,7 +24,7 @@
 
 gen_test_char_OBJECTS = gen_test_char.lo util_debug.lo
 gen_test_char: $(gen_test_char_OBJECTS)
-	$(LINK) $(EXTRA_LDFLAGS) $(gen_test_char_OBJECTS) $(EXTRA_LIBS)
+	$(HOSTCC) $(ALL_CPPFLAGS) $(ALL_INCLUDES) gen_test_char.c util_debug.c -o gen_test_char
 
 test_char.h: gen_test_char
 	./gen_test_char > test_char.h
