--- wizd/wizd.h.orig	2005-05-15 13:07:23.380000000 +0000
+++ wizd/wizd.h	2005-05-15 13:12:40.430000000 +0000
@@ -60,15 +60,13 @@
 #ifdef NEED_CONFIG_FILE_DEFINITION
 char *config_file[] = {
 	"./wizd.conf",
-	"/usr/local/wizd/wizd.conf",
-	"/usr/local/etc/wizd.conf",
-	"/etc/wizd.conf",
+	"/opt/etc/wizd.conf",
 };
 #endif
 
 
 #define	DEFAULT_FLAG_DEBUG_LOG_OUTPUT	FALSE
-#define	DEFAULT_DEBUG_LOG_FILENAME		"/tmp/wizd_debug.log"
+#define	DEFAULT_DEBUG_LOG_FILENAME		"/opt/var/log/wizd_debug.log"
 
 
 #define	DEFAULT_FLAG_USE_SKIN		FALSE
@@ -76,7 +74,7 @@
 #define	DEFAULT_SKINDATA_NAME		"default"
 
 // execute path for CGI
-#define	DEFAULT_PATH	"/usr/bin:/bin:/usr/sbin:/usr/bin"
+#define	DEFAULT_PATH	"/opt/bin:/usr/bin:/bin:/opt/sbin:/usr/sbin:/sbin"
 
 #define	DEFAULT_FLAG_HIDE_SAME_SVI_NAME_DIRECTORY	FALSE
 
