--- wizd/wizd.conf.orig	2005-05-15 13:18:37.040000000 +0000
+++ wizd/wizd.conf	2005-05-15 13:24:56.550000000 +0000
@@ -11,7 +11,7 @@
 
 
 ## Base location for your media files
-document_root           ./docroot
+document_root           /opt/share/wizd
 
 ## Aliases show up in the base location as virtual folders
 ## When selected they get redirected to the path you enter here
@@ -23,11 +23,11 @@
 ## musicalias: allplay defaults to music playlists
 ## photoalias: allplay defaults to photo playlists
 
-#alias	DVD	/cygdrive/c/DVD
-#alias	Internet C:\Documents and Settings\Me\Favorites
-#moviealias	HDTV	d:\HDTV
-#musicalias	Music	c:\Documents and Settings\Me\My Music
-#photoalias	Photos	c:\Documents and Settings\Me\my pictures
+#alias	     DVD	/share/hdd/data/DVD
+#alias	     Internet	/share/hdd/data/favourites
+#moviealias  HDTV	/share/hdd/data/movie
+#musicalias  Music	/share/hdd/data/music
+#photoalias  Photos	/share/hdd/data/photo
 
 ## Enable this flag to allow file deletion from the LinkPlayer
 ## To delete files, press the "setup" button on your remote TWICE
@@ -82,7 +82,7 @@
 ## DebugLog (debug output file name)
 ##  default: /tmp/wizd_debug.log
 ##
-debug_log_filename     wizd_debug.log
+debug_log_filename     /opt/var/log/wizd_debug.log
 
 
 ## (Daemon conversion)
@@ -364,7 +364,7 @@
 ## (output error when executing CGI script)
 # default: /dev/null
 #debug_cgi_output     /dev/tty
-#debug_cgi_output     /tmp/wizd_cgi_debug.log
+#debug_cgi_output     /opt/var/log/wizd_cgi_debug.log
 
 
 ## Proxy use
