--- src/aclocal.m4.orig	2010-08-02 16:33:20.000000000 +0000
+++ src/aclocal.m4	2010-09-28 18:28:50.861824590 +0000
@@ -114,71 +114,11 @@
    		AC_MSG_ERROR([erlang not found])
    fi
 
-
-   cat >>conftest.erl <<_EOF
-
--module(conftest).
--author('alexey@sevcom.net').
-
--export([[start/0]]).
-
-start() ->
-    EIDirS = code:lib_dir("erl_interface") ++ "\n",
-    EILibS =  libpath("erl_interface") ++ "\n",
-    RootDirS = code:root_dir() ++ "\n",
-    file:write_file("conftest.out", list_to_binary(EIDirS ++ EILibS ++ ssldef() ++ RootDirS)),
-    halt().
-
-ssldef() -> 
-   OTP = (catch erlang:system_info(otp_release)),
-   if
-	OTP >= "R14" -> "-DSSL40\n";
-	OTP >= "R12" -> "-DSSL39\n";
-        true -> ""
-   end.
-
-%% return physical architecture based on OS/Processor
-archname() ->
-    ArchStr = erlang:system_info(system_architecture),
-    case os:type() of
-	{win32, _} -> "windows";
-	{unix,UnixName} ->
-	    Specs = string:tokens(ArchStr,"-"),
-	    Cpu = case lists:nth(2,Specs) of
-		      "pc" -> "x86";
-		      _ -> hd(Specs)
-		  end,
-	    atom_to_list(UnixName) ++ "-" ++ Cpu;
-	_ -> "generic"
-    end.
-
-%% Return arch-based library path or a default value if this directory
-%% does not exist
-libpath(App) ->
-    PrivDir    = code:priv_dir(App),
-    ArchDir    = archname(),
-    LibArchDir = filename:join([[PrivDir,"lib",ArchDir]]),
-    case file:list_dir(LibArchDir) of
-	%% Arch lib dir exists: We use it
-	{ok, _List}  -> LibArchDir;
-	%% Arch lib dir does not exist: Return the default value
-	%% ({error, enoent}):
-	_Error -> code:lib_dir("erl_interface") ++ "/lib"
-    end.
-
-_EOF
-
-   if ! $ERLC conftest.erl; then
-   	   AC_MSG_ERROR([could not compile sample program])
-   fi
-
-   if ! $ERL -s conftest -noshell; then
-       AC_MSG_ERROR([could not run sample program])
-   fi
-
-   if ! test -f conftest.out; then
-       AC_MSG_ERROR([erlang program was not properly executed, (conftest.out was not produced)])
-   fi
+   STAGING_LIB_DIR=@STAGING_LIB_DIR@
+   echo ${STAGING_LIB_DIR}/erlang/lib/erl_interface-* > conftest.out
+   echo ${STAGING_LIB_DIR}/erlang/lib/erl_interface-*/lib >> conftest.out
+   printf "\x2DDSSL40\n" >> conftest.out
+   echo ${STAGING_LIB_DIR}/erlang >> conftest.out
 
    # First line
    ERLANG_EI_DIR=`cat conftest.out | head -n 1`
