--- Python-3.0a3/Lib/site.py.orig	2008-01-23 00:24:23.000000000 -0800
+++ Python-3.0a3/Lib/site.py	2008-03-03 09:06:43.000000000 -0800
@@ -171,7 +171,7 @@
 
 def addsitepackages(known_paths):
     """Add site-packages (and possibly site-python) to sys.path"""
-    prefixes = [sys.prefix]
+    prefixes = [os.path.join(sys.prefix, "local"), sys.prefix]
     if sys.exec_prefix != sys.prefix:
         prefixes.append(sys.exec_prefix)
     for prefix in prefixes:
