--- sane-backends/backend/epjitsu.c.orig	2009-09-27 10:29:57.000000000 +0000
+++ sane-backends/backend/epjitsu.c	2009-09-29 06:27:18.372954729 +0000
@@ -3017,7 +3017,7 @@
                         gain_slope[idx] *= 0.75;
                     last_error[idx] = pixerror;
                     /* set the new gain */
-                    newgain = oldgain + (int) roundf(pixerror * gain_slope[idx]);
+                    newgain = oldgain + (int) (pixerror * gain_slope[idx] + .5);
                     if (newgain < 0)
                     {
                         low_pegs++;
