--- sane-backends/tools/Makefile.in.orig	2006-12-09 21:55:52.000000000 +0100
+++ sane-backends/tools/Makefile.in	2006-12-09 21:57:14.000000000 +0100
@@ -108,8 +108,9 @@
 umax_pp: umax_pp.o ../backend/umax_pp_low.o
 	@$(LIBTOOL) $(MLINK) $(LINK) umax_pp.o ../backend/umax_pp_low.o \
 	   $(LIBSANEI) $(LIBLIB) $(LIBS)
-sane-desc: sane-desc.o
-	@$(LIBTOOL) $(MLINK) $(LINK) sane-desc.o $(LIBSANEI) $(LIBLIB)
+sane-desc: sane-desc.c
+	gcc -I. sane-desc.c -I../include ../sanei/sanei_config.c ../sanei/sanei_init_debug.c -o sane-desc
+#	@$(LIBTOOL) $(MLINK) $(LINK) sane-desc.o $(LIBSANEI) $(LIBLIB)
 
 hotplug/libsane.usermap: $(wildcard ${top_srcdir}/doc/descriptions/*.desc) $(wildcard ${top_srcdir}/doc/descriptions-external/*.desc) sane-desc
 	@./sane-desc -m usermap -s ${top_srcdir}/doc/descriptions:${top_srcdir}/doc/descriptions-external \
