diff --git a/Modules/_cursesmodule.c b/Modules/_cursesmodule.c
index 7936aef..534d8c6 100644
--- a/Modules/_cursesmodule.c
+++ b/Modules/_cursesmodule.c
@@ -112,7 +112,8 @@ static const char PyCursesVersion[] = "2.2";
 #define CURSES_MODULE
 #include "py_curses.h"
 
-#if defined(HAVE_TERM_H) || defined(__sgi)
+//#if defined(HAVE_TERM_H) || defined(__sgi)
+#if 1
 /* For termname, longname, putp, tigetflag, tigetnum, tigetstr, tparm
    which are not declared in SysV curses and for setupterm. */
 #include <term.h>
