diff --git a/configure b/configure
index 36e117a..2431cce 100755
--- a/configure
+++ b/configure
@@ -14969,7 +14969,7 @@ $as_echo "$SOABI" >&6; }
 
 case $ac_sys_system in
     Linux*|GNU*|Darwin)
-	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
+	EXT_SUFFIX=${SHLIB_SUFFIX};;
     *)
 	EXT_SUFFIX=${SHLIB_SUFFIX};;
 esac
diff --git a/configure.ac b/configure.ac
index 620b48a..cbd529f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4669,7 +4669,7 @@ AC_MSG_RESULT($SOABI)
 AC_SUBST(EXT_SUFFIX)
 case $ac_sys_system in
     Linux*|GNU*|Darwin)
-	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
+	EXT_SUFFIX=${SHLIB_SUFFIX};;
     *)
 	EXT_SUFFIX=${SHLIB_SUFFIX};;
 esac
