--- Python-3.0a5/Lib/site.py.orig	2008-05-06 16:45:46.000000000 -0700
+++ Python-3.0a5/Lib/site.py	2008-05-15 14:47:30.000000000 -0700
@@ -57,7 +57,7 @@
 import builtins
 
 # Prefixes for site-packages; add additional prefixes like /usr/local here
-PREFIXES = [sys.prefix, sys.exec_prefix]
+PREFIXES = [os.path.join(sys.prefix, "local"), sys.prefix, sys.exec_prefix]
 # Enable per user site-packages directory
 # set it to False to disable the feature or True to force the feature
 ENABLE_USER_SITE = None
