diff --git a/Lib/site.py b/Lib/site.py
index ffd132b..25efc21 100644
--- a/Lib/site.py
+++ b/Lib/site.py
@@ -75,7 +75,7 @@ import builtins
 import _sitebuiltins
 
 # Prefixes for site-packages; add additional prefixes like /usr/local here
-PREFIXES = [sys.prefix, sys.exec_prefix]
+PREFIXES = [os.path.join(sys.prefix, "local"), sys.prefix, sys.exec_prefix]
 # Enable per user site-packages directory
 # set it to False to disable the feature or True to force the feature
 ENABLE_USER_SITE = None
