--- Python-3.4.2/Lib/site.py.orig	2014-10-08 11:18:12.000000000 +0300
+++ Python-3.4.2/Lib/site.py	2015-01-22 15:10:07.673061126 +0200
@@ -74,7 +74,7 @@
 import _sitebuiltins
 
 # Prefixes for site-packages; add additional prefixes like /usr/local here
-PREFIXES = [sys.prefix, sys.exec_prefix]
+PREFIXES = [os.path.join(sys.prefix, "local"), sys.prefix, sys.exec_prefix]
 # Enable per user site-packages directory
 # set it to False to disable the feature or True to force the feature
 ENABLE_USER_SITE = None
