diff --git a/Modules/Setup.dist b/Modules/Setup.dist
index f0eff5d..ea0e6cf 100644
--- a/Modules/Setup.dist
+++ b/Modules/Setup.dist
@@ -216,13 +216,13 @@ _symtable symtablemodule.c
 # The crypt module is now disabled by default because it breaks builds
 # on many systems (where -lcrypt is needed), e.g. Linux (I believe).
 
-#_crypt _cryptmodule.c # -lcrypt	# crypt(3); needs -lcrypt on some systems
+_crypt _cryptmodule.c -lcrypt	# crypt(3); needs -lcrypt on some systems
 
 
 # Some more UNIX dependent modules -- off by default, since these
 # are not supported by all UNIX systems:
 
-#nis nismodule.c -lnsl	# Sun yellow pages -- not everywhere
+nis nismodule.c -lnsl	# Sun yellow pages -- not everywhere
 #termios termios.c	# Steen Lumholt's termios module
 #resource resource.c	# Jeremy Hylton's rlimit interface
 
