--- a/Modules/Setup.dist
+++ b/Modules/Setup.dist
@@ -214,13 +214,13 @@
 #
 # First, look at Setup.config; configure may have set this for you.
 
-#_crypt _cryptmodule.c # -lcrypt	# crypt(3); needs -lcrypt on some systems
+_crypt _cryptmodule.c -lcrypt	# crypt(3); needs -lcrypt on some systems
 
 
 # Some more UNIX dependent modules -- off by default, since these
 # are not supported by all UNIX systems:
 
-#nis nismodule.c -lnsl	# Sun yellow pages -- not everywhere
+nis nismodule.c -lnsl	# Sun yellow pages -- not everywhere
 #termios termios.c	# Steen Lumholt's termios module
 #resource resource.c	# Jeremy Hylton's rlimit interface
 
