--- a/setup.py
+++ b/setup.py
@@ -16,7 +16,7 @@ from distutils.command.install_lib import install_lib
 from distutils.command.build_scripts import build_scripts
 from distutils.spawn import find_executable
 
-cross_compiling = "_PYTHON_HOST_PLATFORM" in os.environ
+cross_compiling = ("_PYTHON_HOST_PLATFORM" in os.environ) and (os.environ["_PYTHON_HOST_PLATFORM"] != '')
 
 # Add special CFLAGS reserved for building the interpreter and the stdlib
 # modules (Issue #21121).
@@ -43,7 +43,7 @@ host_platform = get_platform()
 COMPILED_WITH_PYDEBUG = ('--with-pydebug' in sysconfig.get_config_var("CONFIG_ARGS"))
 
 # This global variable is used to hold the list of modules to be disabled.
-disabled_module_list = []
+disabled_module_list = ['_tkinter'] # '_sqlite',
 
 def add_dir_to_list(dirlist, dir):
     """Add the directory 'dir' to the list 'dirlist' (after any relative
@@ -422,6 +422,10 @@ class PyBuildExt(build_ext):
             self.announce('WARNING: skipping import check for Cygwin-based "%s"'
                 % ext.name)
             return
+        if True:
+            self.announce('WARNING: skipping import check for cross compiled "%s"'
+                % ext.name)
+            return
         ext_filename = os.path.join(
             self.build_lib,
             self.get_ext_filename(self.get_ext_fullname(ext.name)))
@@ -537,16 +541,19 @@ class PyBuildExt(build_ext):
             os.unlink(tmpfile)
 
     def detect_modules(self):
+        add_dir_to_list(self.compiler.include_dirs, '.')
         # Ensure that /usr/local is always used, but the local build
         # directories (i.e. '.' and 'Include') must be first.  See issue
         # 10520.
-        if not cross_compiling:
-            add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
-            add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
+#        if not cross_compiling:
+#            add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
+#            add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
         # only change this for cross builds for 3.3, issues on Mageia
         if cross_compiling:
-            self.add_gcc_paths()
-        self.add_multiarch_paths()
+#            self.add_gcc_paths()
+            add_dir_to_list(self.compiler.include_dirs, '@STAGING_INCLUDE@')
+            add_dir_to_list(self.compiler.include_dirs, '@TOOLCHAIN_TARGET_INCLUDE@')
+#        self.add_multiarch_paths()
 
         # Add paths specified in the environment variables LDFLAGS and
         # CPPFLAGS for header and library files.
@@ -559,7 +566,7 @@ class PyBuildExt(build_ext):
                 ('LDFLAGS', '-L', self.compiler.library_dirs),
                 ('CPPFLAGS', '-I', self.compiler.include_dirs)):
             env_val = sysconfig.get_config_var(env_var)
-            if env_val:
+            if env_val and (not cross_compiling):
                 # To prevent optparse from raising an exception about any
                 # options in env_val that it doesn't know about we strip out
                 # all double dashes and any dashes followed by a character
@@ -582,18 +589,6 @@ class PyBuildExt(build_ext):
                     for directory in reversed(options.dirs):
                         add_dir_to_list(dir_list, directory)
 
-        if (not cross_compiling and
-                os.path.normpath(sys.base_prefix) != '/usr' and
-                not sysconfig.get_config_var('PYTHONFRAMEWORK')):
-            # OSX note: Don't add LIBDIR and INCLUDEDIR to building a framework
-            # (PYTHONFRAMEWORK is set) to avoid # linking problems when
-            # building a framework with different architectures than
-            # the one that is currently installed (issue #7473)
-            add_dir_to_list(self.compiler.library_dirs,
-                            sysconfig.get_config_var("LIBDIR"))
-            add_dir_to_list(self.compiler.include_dirs,
-                            sysconfig.get_config_var("INCLUDEDIR"))
-
         system_lib_dirs = ['/lib64', '/usr/lib64', '/lib', '/usr/lib']
         system_include_dirs = ['/usr/include']
         # lib_dirs and inc_dirs are used to search for files;
@@ -1241,8 +1236,10 @@ class PyBuildExt(build_ext):
         dbm_order = ['gdbm']
         # The standard Unix dbm module:
         if host_platform not in ['cygwin']:
-            config_args = [arg.strip("'")
-                           for arg in sysconfig.get_config_var("CONFIG_ARGS").split()]
+            if ("GNU_TARGET_NAME" in os.environ):
+                config_args = [ '--host=%s' % os.environ["GNU_TARGET_NAME"] ]
+            else:
+                config_args = [ ]
             dbm_args = [arg for arg in config_args
                         if arg.startswith('--with-dbmliborder=')]
             if dbm_args:
@@ -1300,7 +1297,6 @@ class PyBuildExt(build_ext):
                         if dbm_setup_debug: print("building dbm using bdb")
                         dbmext = Extension('_dbm', ['_dbmmodule.c'],
                                            library_dirs=dblib_dir,
-                                           runtime_library_dirs=dblib_dir,
                                            include_dirs=db_incs,
                                            define_macros=[
                                                ('HAVE_BERKDB_H', None),
@@ -2125,6 +2121,22 @@ class PyBuildExt(build_ext):
         return ext
 
     def _detect_openssl(self, inc_dirs, lib_dirs):
+        if cross_compiling:
+            ssl_ext = Extension(
+                '_ssl', ['_ssl.c'],
+                include_dirs=(),
+                library_dirs=(),
+                libraries=('ssl','crypto'),
+                depends=['socketmodule.h']
+            )
+            hashlib_ext = Extension(
+                '_hashlib', ['_hashopenssl.c'],
+                depends=['hashlib.h'],
+                include_dirs=(),
+                library_dirs=(),
+                libraries=('ssl','crypto'),
+            )
+            return ssl_ext, hashlib_ext
         config_vars = sysconfig.get_config_vars()
 
         def split_var(name, sep):
