--- Python-3.1/Modules/_localemodule.c.orig	2009-06-07 08:29:46.000000000 -0700
+++ Python-3.1/Modules/_localemodule.c	2009-06-29 10:07:57.000000000 -0700
@@ -24,7 +24,7 @@
 #include <langinfo.h>
 #endif
 
-#ifdef HAVE_LIBINTL_H
+#if defined(WITH_LIBINTL) && defined(HAVE_LIBINTL_H)
 #include <libintl.h>
 #endif
 
@@ -498,7 +498,7 @@
 }
 #endif /* HAVE_LANGINFO_H */
 
-#ifdef HAVE_LIBINTL_H
+#if defined(WITH_LIBINTL) && defined(HAVE_LIBINTL_H)
 
 PyDoc_STRVAR(gettext__doc__,
 "gettext(msg) -> string\n"
@@ -620,7 +620,7 @@
   {"nl_langinfo", (PyCFunction) PyLocale_nl_langinfo,
    METH_VARARGS, nl_langinfo__doc__},
 #endif
-#ifdef HAVE_LIBINTL_H
+#if defined(WITH_LIBINTL) && defined(HAVE_LIBINTL_H)
   {"gettext",(PyCFunction)PyIntl_gettext,METH_VARARGS,
     gettext__doc__},
   {"dgettext",(PyCFunction)PyIntl_dgettext,METH_VARARGS,
