diff --git a/Modules/_localemodule.c b/Modules/_localemodule.c
index f9eeeb7..b3d7dd9 100644
--- a/Modules/_localemodule.c
+++ b/Modules/_localemodule.c
@@ -25,7 +25,7 @@ This software comes with no warranty. Use at your own risk.
 #include <langinfo.h>
 #endif
 
-#ifdef HAVE_LIBINTL_H
+#if defined(WITH_LIBINTL) && defined(HAVE_LIBINTL_H)
 #include <libintl.h>
 #endif
 
@@ -487,7 +487,7 @@ PyLocale_nl_langinfo(PyObject* self, PyObject* args)
 }
 #endif /* HAVE_LANGINFO_H */
 
-#ifdef HAVE_LIBINTL_H
+#if defined(WITH_LIBINTL) && defined(HAVE_LIBINTL_H)
 
 PyDoc_STRVAR(gettext__doc__,
 "gettext(msg) -> string\n"
@@ -622,7 +622,7 @@ static struct PyMethodDef PyLocale_Methods[] = {
   {"nl_langinfo", (PyCFunction) PyLocale_nl_langinfo,
    METH_VARARGS, nl_langinfo__doc__},
 #endif
-#ifdef HAVE_LIBINTL_H
+#if defined(WITH_LIBINTL) && defined(HAVE_LIBINTL_H)
   {"gettext",(PyCFunction)PyIntl_gettext,METH_VARARGS,
     gettext__doc__},
   {"dgettext",(PyCFunction)PyIntl_dgettext,METH_VARARGS,
