From 0c0b575b529d4fe6dc0be179e701a302f27bc127 Mon Sep 17 00:00:00 2001
From: David Beazley <dave@dabeaz.com>
Date: Thu, 26 Apr 2012 10:15:09 -0500
Subject: [PATCH 3/3] Fixed lexer line tracking.

---
 test/calclex.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/calclex.py b/test/calclex.py
index 67d245f..302f0b0 100644
--- a/test/calclex.py
+++ b/test/calclex.py
@@ -36,7 +36,7 @@ t_ignore = " \t"
 
 def t_newline(t):
     r'\n+'
-    t.lineno += t.value.count("\n")
+    t.lexer.lineno += t.value.count("\n")
     
 def t_error(t):
     print("Illegal character '%s'" % t.value[0])
-- 
1.8.5.2

