--- Python-2.5b1/configure.in.orig	2006-06-19 01:14:09.000000000 -0700
+++ Python-2.5b1/configure.in	2006-06-22 22:20:57.000000000 -0700
@@ -9,6 +9,11 @@
 AC_CONFIG_SRCDIR([Include/object.h])
 AC_CONFIG_HEADER(pyconfig.h)
 
+# find compiler while respecting --host setting
+AC_CANONICAL_HOST()
+AC_CHECK_TOOLS(CC,gcc cc)
+AC_CHECK_TOOLS(CXX,g++ c++)
+
 dnl This is for stuff that absolutely must end up in pyconfig.h.
 dnl Please use pyport.h instead, if possible.
 AH_TOP([
@@ -676,9 +681,9 @@
 
 AC_MSG_RESULT($LDLIBRARY)
 
-AC_PROG_RANLIB
-AC_SUBST(AR)
-AC_CHECK_PROGS(AR, ar aal, ar)
+# find tools while respecting --host setting
+AC_CHECK_TOOL(RANLIB,ranlib)
+AC_CHECK_TOOLS(AR,ar aal,ar)
 
 AC_SUBST(SVNVERSION)
 AC_CHECK_PROG(SVNVERSION, svnversion, found, not-found)
@@ -3339,25 +3344,8 @@
 fi
 
 AC_MSG_CHECKING(for %zd printf() format support)
-AC_TRY_RUN([#include <stdio.h>
-#include <stddef.h>
-#include <string.h>
-
-int main()
-{
-    char buffer[4];
-
-    if(sprintf(buffer, "%zd", (size_t)123) < 0)
-       	return 1;
-
-    if (strncmp(buffer, "123", 3))
-	return 1;
-    
-    return 0;
-}],
-[AC_MSG_RESULT(yes)
- AC_DEFINE(PY_FORMAT_SIZE_T, "z", [Define to printf format modifier for Py_ssize_t])],
- AC_MSG_RESULT(no))
+AC_MSG_RESULT(yes)
+ AC_DEFINE(PY_FORMAT_SIZE_T, "z", [Define to printf format modifier for Py_ssize_t])
 
 AC_CHECK_TYPE(socklen_t,,
   AC_DEFINE(socklen_t,int,
@@ -3387,6 +3375,25 @@
 done
 AC_MSG_RESULT(done)
 
+AC_SUBST(BUILDPYTHON)
+AC_SUBST(BUILDPGEN)
+AC_SUBST(CROSS_COMPILE)
+if [[ $build != $host ]]; then
+    AC_MSG_NOTICE(Cross compiling: Configuring build python)
+    absconfigcommand=$(pwd)/$0
+    mkdir -p buildpython25
+    cd buildpython25
+    CC="" CXX="" AR="" RANLIB="" LDFLAGS="" $absconfigcommand --prefix=$prefix
+    cd ..
+    BUILDPYTHON='buildpython25/python$(BUILDEXE)'
+    BUILDPGEN='buildpython25/Parser/pgen$(BUILDEXE)'
+    CROSS_COMPILE=yes
+else
+    BUILDPYTHON='$(HOSTPYTHON)'
+    BUILDPGEN='$(HOSTPGEN)'
+    CROSS_COMPILE=no
+fi
+
 # generate output files
 AC_CONFIG_FILES(Makefile.pre Modules/Setup.config)
 AC_OUTPUT
