--- kissdx.conf	2007-01-23 23:22:31.000000000 +0100
+++ patched/kissdx.conf	2007-01-23 23:26:19.390625000 +0100
@@ -8,7 +8,7 @@
 # Path for persistent storage of info by kissdx.
 # The [Recently used] feature and the picture caching feature will store data here.
 # If option is omitted, kissdx will use picturepath, etc. for such storage.
-#persistentstoragepath	= /var/kissdx
+#persistentstoragepath	= /opt/var/kissdx
 
 # Allowed extensions for media files (comma separated, no spaces, case insensitive)
 # Omit a setting to send ALL files to the player in the corresponding directory listing.
