--- postgresql-8.2.0/src/timezone/Makefile.orig	2006-07-24 20:51:22.000000000 -0700
+++ postgresql-8.2.0/src/timezone/Makefile	2006-12-05 21:01:19.000000000 -0800
@@ -34,7 +34,8 @@
 	$(CC) $(CFLAGS) $(ZICOBJS) $(LDFLAGS) $(LIBS) -o $@$(X)
 
 install: all installdirs
-	./zic -d '$(DESTDIR)$(datadir)/timezone' -p '$(POSIXRULES)' $(TZDATAFILES)
+	gcc -o ./zic-cross  -I../include zic.c ialloc.c scheck.c localtime.c ../port/snprintf.c ../port/qsort.c
+	./zic-cross -d '$(DESTDIR)$(datadir)/timezone' -p '$(POSIXRULES)' $(TZDATAFILES)
 	$(MAKE) -C tznames $@
 
 installdirs:
