--- a/libpam/pam_private.h
+++ b/libpam/pam_private.h
@@ -24,9 +24,9 @@
 
 /* the Linux-PAM configuration file */
 
-#define PAM_CONFIG    "/etc/pam.conf"
-#define PAM_CONFIG_D  "/etc/pam.d"
-#define PAM_CONFIG_DF "/etc/pam.d/%s"
+#define PAM_CONFIG    "%OPTWARE_TARGET_PREFIX%/etc/pam.conf"
+#define PAM_CONFIG_D  "%OPTWARE_TARGET_PREFIX%/etc/pam.d"
+#define PAM_CONFIG_DF "%OPTWARE_TARGET_PREFIX%/etc/pam.d/%s"
 
 #define PAM_DEFAULT_SERVICE        "other"     /* lower case */
 #define PAM_DEFAULT_SERVICE_FILE   PAM_CONFIG_D "/" PAM_DEFAULT_SERVICE
@@ -36,7 +36,7 @@
  * the Linux-PAM lock file. If it exists Linux-PAM will abort. Use it
  * to block access to libpam
  */
-#define PAM_LOCK_FILE "/var/lock/subsys/PAM"
+#define PAM_LOCK_FILE "%OPTWARE_TARGET_PREFIX%/var/lock/subsys/PAM"
 #endif
 
 /* components of the pam_handle structure */
--- a/conf/pam.conf
+++ b/conf/pam.conf
@@ -1,5 +1,5 @@
 # ---------------------------------------------------------------------------#
-# /etc/pam.conf								     #
+# %OPTWARE_TARGET_PREFIX%/etc/pam.conf								     #
 #									     #
 # Last modified by Andrew G. Morgan <morgan@kernel.org>		             #
 # ---------------------------------------------------------------------------#
@@ -26,7 +26,7 @@
 # The PAM configuration file for the `ftp' service
 #
 ftp	auth       requisite  pam_listfile.so \
-		item=user sense=deny file=/etc/ftpusers onerr=succeed
+		item=user sense=deny file=%OPTWARE_TARGET_PREFIX%/etc/ftpusers onerr=succeed
 ftp	auth       requisite  pam_shells.so
 ftp	auth       required   pam_unix.so
 ftp	account    required   pam_unix.so
