--- configure	2005-04-13 22:35:02.000000000 +0200
+++ configure.patched	2005-09-10 20:12:40.000000000 +0200
@@ -9430,17 +9430,7 @@
 # ------------------------------------------------------------------------
 # Paths
 # ------------------------------------------------------------------------
-
-# Find the right directory to put the root-mode PID file in
-echo "$as_me:$LINENO: checking pid file location" >&5
-echo $ECHO_N "checking pid file location... $ECHO_C" >&6
-if test -d "/var/run"
-then
-	piddir="/var/run"
-elif test -d "/etc"
-then
-	piddir="/etc"
-fi
+piddir=/opt/var/run
 
 
 cat >>confdefs.h <<_ACEOF
