--- configure.in.orig	2009-03-14 13:15:34.000000000 +0000
+++ configure.in	2009-03-20 17:32:47.773731675 +0000
@@ -210,15 +210,7 @@
       LIBS="$LIBS $ICONV_LFLAGS"
    fi
    AC_MSG_CHECKING(for iconv usability in programs)
-   AC_TRY_RUN([
-	#include <iconv.h>
-	int main(int argc, char **argv) {
-	    iconv_t conv = iconv_open("ISO8859-1", "UTF-8");
-            if (conv != (iconv_t) -1) {
-	      return 0;
-            }
-	    return 1;
-      }],iconv_found="yes")
+   iconv_found="yes"
   if test "x$iconv_found" = "xno" ; then
      AC_MSG_RESULT(no)
   else
@@ -605,8 +597,8 @@
 *** libgnutls was not found. You may want to get it from ftp://ftp.gnutls.org/pub/gnutls/
 *** WeeChat will be built without GnuTLS support.]]))
     if test "x$found_gnutls" = "xyes" ; then
-        GNUTLS_CFLAGS=`libgnutls-config --cflags`
-        GNUTLS_LFLAGS=`libgnutls-config --libs`
+        GNUTLS_CFLAGS=`$LIBGNUTLS_CONFIG --cflags`
+        GNUTLS_LFLAGS=`$LIBGNUTLS_CONFIG --libs`
         AC_SUBST(GNUTLS_CFLAGS)
         AC_SUBST(GNUTLS_LFLAGS)
         AC_DEFINE(HAVE_GNUTLS)
