--- mc/m4.include/mc-with-screen.m4.orig	2010-01-03 04:50:50.277232242 +0000
+++ mc/m4.include/mc-with-screen.m4	2010-01-03 04:52:19.154278937 +0000
@@ -98,25 +98,7 @@
     dnl check if S-Lang have version 2.0 or newer
     if test x"$found_slang" = x"yes"; then
         AC_MSG_CHECKING([for S-Lang version 2.0 or newer])
-        AC_TRY_RUN([
-#ifdef HAVE_SLANG_SLANG_H
-#include <slang/slang.h>
-#else
-#include <slang.h>
-#endif
-int main (void)
-{
-#if SLANG_VERSION >= 20000
-    return 0;
-#else
-    return 1;
-#endif
-}
-],
-	    [mc_slang_is_valid_version=yes],
-	    [mc_slang_is_valid_version=no],
-	    [mc_slang_is_valid_version=no]
-	)
+	    mc_slang_is_valid_version=yes
 	if test x$mc_slang_is_valid_version = xno; then
             found_slang=no
             error_msg_slang="S-Lang library version 2.0 or newer not found"
