--- mc-4.6.2/doc/Makefile.in.orig	2009-02-01 19:46:24.000000000 +0000
+++ mc-4.6.2/doc/Makefile.in	2009-03-04 04:54:33.236835120 +0000
@@ -677,7 +677,7 @@
 
 
 mc.hlp: mc.1 $(srcdir)/xnc.hlp $(top_builddir)/src/man2hlp
-	- $(top_builddir)/src/man2hlp mc.1 $(srcdir)/xnc.hlp mc.hlp
+	- $(top_builddir)/src/man2hlp.host mc.1 $(srcdir)/xnc.hlp mc.hlp
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
