--- daemon/remote.c.orig	2013-01-08 17:32:09.000000000 -0800
+++ daemon/remote.c	2013-01-29 10:36:37.000000000 -0800
@@ -1656,6 +1656,7 @@
     else
     {
         int64_t      tag = -1;
+	int          itag;
         const char * str;
 
         if (tr_bencDictFindStr (&top, "result", &str))
@@ -1669,7 +1670,8 @@
             {
         tr_bencDictFindInt (&top, "tag", &tag);
 
-        switch (tag)
+	itag = tag;
+        switch (itag)
         {
             case TAG_SESSION:
                 printSession (&top); break;
