--- Makefile	Sun Sep 12 03:37:58 2004
+++ tcp_wrappers.makefile.new	Sun Sep 12 03:37:47 2004
@@ -143,7 +143,7 @@
 
 linux:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ=setenv.o \
+	LIBS= ARFLAGS=rv AUX_OBJ=setenv.o \
 	NETGROUP= TLI= EXTRA_CFLAGS="-DBROKEN_SO_LINGER" all
 
 # This is good for many SYSV+BSD hybrids with NIS, probably also for HP-UX 7.x.
@@ -375,7 +375,7 @@
 # library; some don't care as long as the modules are in the right order;
 # some systems don't even have a ranlib(1) command. Make your choice.
 
-RANLIB	= ranlib	# have ranlib (BSD-ish UNIX)
+#RANLIB	= ranlib	# have ranlib (BSD-ish UNIX)
 #RANLIB	= echo		# no ranlib (SYSV-ish UNIX)
 
 ARFLAGS	= rv		# most systems
@@ -590,7 +590,7 @@
 # look for access control information. Watch out for the quotes and
 # backslashes when you make changes.
 
-TABLES	= -DHOSTS_DENY=\"/etc/hosts.deny\" -DHOSTS_ALLOW=\"/etc/hosts.allow\"
+TABLES	= -DHOSTS_DENY=\"/opt/etc/hosts.deny\" -DHOSTS_ALLOW=\"/opt/etc/hosts.allow\" -DSYS_ERRLIST_DEFINED
 
 ####################################################
 # Optional: dealing with host name/address conflicts
--- tcp_wrappers_7.6/percent_m.c	1994-12-28 17:42:37.000000000 +0100
+++ tcpwrappers/percent_m.c	2006-05-29 11:49:52.000000000 +0200
@@ -13,10 +13,6 @@
 #include <string.h>
 
 extern int errno;
-#ifndef SYS_ERRLIST_DEFINED
-extern char *sys_errlist[];
-extern int sys_nerr;
-#endif
 
 #include "mystdarg.h"
 
@@ -29,11 +25,7 @@
 
     while (*bp = *cp)
 	if (*cp == '%' && cp[1] == 'm') {
-	    if (errno < sys_nerr && errno > 0) {
-		strcpy(bp, sys_errlist[errno]);
-	    } else {
-		sprintf(bp, "Unknown error %d", errno);
-	    }
+	    strcpy(bp, strerror(errno));
 	    bp += strlen(bp);
 	    cp += 2;
 	} else {
--- tcp_wrappers_7.6/tli.c	1997-03-21 19:27:26.000000000 +0100
+++ tcpwrappers/tli.c	2006-05-29 11:51:51.000000000 +0200
@@ -40,8 +40,6 @@
 
 extern char *nc_sperror();
 extern int errno;
-extern char *sys_errlist[];
-extern int sys_nerr;
 extern int t_errno;
 extern char *t_errlist[];
 extern int t_nerr;
@@ -305,12 +303,7 @@
 	    return (t_errlist[t_errno]);
 	}
     } else {
-	if (errno < 0 || errno >= sys_nerr) {
-	    sprintf(buf, "Unknown UNIX error %d", errno);
-	    return (buf);
-	} else {
-	    return (sys_errlist[errno]);
-	}
+	return (strerror(errno));
     }
 }
 
