--- mysql-4.1.16/sql/Makefile.am	2005-11-30 12:36:04.000000000 -0800
+++ mysql/sql/Makefile.am	2005-12-14 14:07:58.000000000 -0800
@@ -105,7 +105,7 @@
 			-DSHAREDIR="\"$(MYSQLSHAREdir)\"" \
 			@DEFS@
 
-BUILT_SOURCES =		sql_yacc.cc sql_yacc.h lex_hash.h
+BUILT_SOURCES =		sql_yacc.cc sql_yacc.h
 EXTRA_DIST =		udf_example.cc $(BUILT_SOURCES)
 AM_YFLAGS =		-d
 
@@ -137,16 +137,16 @@
 		@echo "If it fails, re-run configure with --with-low-memory"
 		$(CXXCOMPILE) $(LM_CFLAGS) -c $<
 
-lex_hash.h:	gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@
+#distclean-local:
+#		rm -f lex_hash.h
 
 # For testing of udf_example.so;  Works on platforms with gcc
 # (This is not part of our build process but only provided as an example)
 udf_example.so:	udf_example.cc
 		$(CXXCOMPILE) -shared -o $@ $<
 
-distclean-local:
-		rm -f lex_hash.h
+#distclean-local:
+#		rm -f lex_hash.h
 
 # Don't update the files from bitkeeper
 %::SCCS/s.%
