*** quagga/zebra/rt_netlink.c.orig	2005-05-30 23:11:12.916987453 +0200
--- quagga/zebra/rt_netlink.c	2005-05-30 23:26:13.459140446 +0200
***************
*** 1470,1478 ****
                        zlog_debug
                          ("netlink_route_multipath() (recursive, 1 hop): "
                           "%s %s/%d, type %s", lookup (nlmsg_str, cmd),
  			 (family == AF_INET) ? inet_ntoa (p->u.prefix4) :
! 			 inet6_ntoa (p->u.prefix6), p->prefixlen,
!                          nexthop_types_desc[nexthop->rtype]);
                      }
  
                    if (nexthop->rtype == NEXTHOP_TYPE_IPV4
--- 1470,1482 ----
                        zlog_debug
                          ("netlink_route_multipath() (recursive, 1 hop): "
                           "%s %s/%d, type %s", lookup (nlmsg_str, cmd),
+ #ifdef HAVE_IPV6
  			 (family == AF_INET) ? inet_ntoa (p->u.prefix4) :
! 			 inet6_ntoa (p->u.prefix6),
! #else
! 			inet_ntoa (p->u.prefix4),
! #endif /* HAVE_IPV6 */	
! 			 p->prefixlen, nexthop_types_desc[nexthop->rtype]);
                      }
  
                    if (nexthop->rtype == NEXTHOP_TYPE_IPV4
***************
*** 1524,1532 ****
                        zlog_debug
                          ("netlink_route_multipath() (single hop): "
                           "%s %s/%d, type %s", lookup (nlmsg_str, cmd),
  			 (family == AF_INET) ? inet_ntoa (p->u.prefix4) :
! 			 inet6_ntoa (p->u.prefix6), p->prefixlen,
! 			 nexthop_types_desc[nexthop->type]);
                      }
  
                    if (nexthop->type == NEXTHOP_TYPE_IPV4
--- 1528,1540 ----
                        zlog_debug
                          ("netlink_route_multipath() (single hop): "
                           "%s %s/%d, type %s", lookup (nlmsg_str, cmd),
+ #ifdef HAVE_IPV6
  			 (family == AF_INET) ? inet_ntoa (p->u.prefix4) :
! 			 inet6_ntoa (p->u.prefix6),
! #else
! 			inet_ntoa (p->u.prefix4),
! #endif  /* HAVE_IPV6 */	
! 			 p->prefixlen, nexthop_types_desc[nexthop->type]);
                      }
  
                    if (nexthop->type == NEXTHOP_TYPE_IPV4
***************
*** 1611,1618 ****
                      {
                        zlog_debug ("netlink_route_multipath() "
                           "(recursive, multihop): %s %s/%d type %s",
!                          lookup (nlmsg_str, cmd), (family == AF_INET) ? 
! 			 inet_ntoa (p->u.prefix4) : inet6_ntoa (p->u.prefix6),
                           p->prefixlen, nexthop_types_desc[nexthop->rtype]);
                      }
                    if (nexthop->rtype == NEXTHOP_TYPE_IPV4
--- 1619,1632 ----
                      {
                        zlog_debug ("netlink_route_multipath() "
                           "(recursive, multihop): %s %s/%d type %s",
!                          lookup (nlmsg_str, cmd), 
! #ifdef HAVE_IPV6
! 			(family == AF_INET) ?  inet_ntoa (p->u.prefix4) : 
! 				inet6_ntoa (p->u.prefix6),
! #else
! 			 inet_ntoa (p->u.prefix4),
! #endif /* HAVE_IPV6 */
! 			
                           p->prefixlen, nexthop_types_desc[nexthop->rtype]);
                      }
                    if (nexthop->rtype == NEXTHOP_TYPE_IPV4
***************
*** 1668,1676 ****
                      {
                        zlog_debug ("netlink_route_multipath() (multihop): "
                           "%s %s/%d, type %s", lookup (nlmsg_str, cmd),
  			 (family == AF_INET) ? inet_ntoa (p->u.prefix4) :
! 			 inet6_ntoa (p->u.prefix6), p->prefixlen,
!                          nexthop_types_desc[nexthop->type]);
                      }
                    if (nexthop->type == NEXTHOP_TYPE_IPV4
                        || nexthop->type == NEXTHOP_TYPE_IPV4_IFINDEX)
--- 1682,1694 ----
                      {
                        zlog_debug ("netlink_route_multipath() (multihop): "
                           "%s %s/%d, type %s", lookup (nlmsg_str, cmd),
+ #ifdef HAVE_IPV6
  			 (family == AF_INET) ? inet_ntoa (p->u.prefix4) :
! 			 inet6_ntoa (p->u.prefix6),
! #else
! 			inet_ntoa (p->u.prefix4),
! #endif /* HAVE_IPV6 */
! 			p->prefixlen, nexthop_types_desc[nexthop->type]);
                      }
                    if (nexthop->type == NEXTHOP_TYPE_IPV4
                        || nexthop->type == NEXTHOP_TYPE_IPV4_IFINDEX)
