--- transmission-0.82/libtransmission/transmission.h	2007-09-10 00:37:30.000000000 +0200
+++ transmission/libtransmission/transmission.h	2007-09-27 11:47:21.000000000 +0200
@@ -384,7 +384,7 @@
 
 const tr_info_t * tr_torrentInfo( const tr_torrent_t * );
 
-#if 0
+#if 1
 /***********************************************************************
  * tr_torrentScrape
  ***********************************************************************
--- transmission-0.82/libtransmission/tracker.c	2007-09-10 00:37:30.000000000 +0200
+++ transmission/libtransmission/tracker.c	2007-09-27 11:47:21.000000000 +0200
@@ -1041,7 +1041,7 @@
 }
 
 
-#if 0
+#if 1
 /* Blocking version */
 int tr_trackerScrape( tr_torrent_t * tor, int * s, int * l, int * d )
 {
--- transmission-0.82/cli/transmissioncli.c	2007-09-10 00:39:34.000000000 +0200
+++ transmission/cli/transmissioncli.c	2007-09-27 11:49:30.000000000 +0200
@@ -65,7 +65,7 @@
 "  -i, --info           Print metainfo and exit\n"
 "  -n  --nat-traversal  Attempt NAT traversal using NAT-PMP or UPnP IGD\n"
 "  -p, --port <int>     Port we should listen on (default = %d)\n"
-#if 0
+#if 1
 "  -s, --scrape         Print counts of seeders/leechers and exit\n"
 #endif
 "  -u, --upload <int>   Maximum upload rate (-1 = no limit, default = 20)\n"
@@ -73,7 +73,7 @@
 
 static int           showHelp      = 0;
 static int           showInfo      = 0;
-#if 0
+#if 1
 static int           showScrape    = 0;
 #endif
 static int           isPrivate     = 0;
@@ -216,12 +216,12 @@
         goto cleanup;
     }
 
-#if 0
+#if 1
     if( showScrape )
     {
         int seeders, leechers, downloaded;
 
-        if( tr_torrentScrape( tor, &seeders, &leechers, &downloaded ) )
+        if( tr_trackerScrape( tor, &seeders, &leechers, &downloaded ) )
         {
             printf( "Scrape failed.\n" );
         }
@@ -378,7 +378,7 @@
             case 'i':
                 showInfo = 1;
                 break;
-#if 0
+#if 1
             case 's':
                 showScrape = 1;
                 break;
