--- transmission/cli/Makefile.am.orig	2007-11-09 14:40:13.000000000 +0100
+++ transmission/cli/Makefile.am	2007-11-09 14:55:07.000000000 +0100
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I$(top_srcdir) $(LIBEVENT_CPPFLAGS)
 AM_CFLAGS = $(OPENSSL_CFLAGS) $(PTHREAD_CFLAGS)
 
-bin_PROGRAMS = transmission-cli
+bin_PROGRAMS = transmission-cli transmissiond
 
 dist_man_MANS = transmissioncli.1
 
@@ -13,3 +13,10 @@
     $(top_builddir)/third-party/miniupnp/libminiupnp.a \
     $(OPENSSL_LIBS) $(PTHREAD_LIBS) -lm
 
+transmissiond_SOURCES = transmissiond.c
+transmissiond_LDADD = \
+    $(top_builddir)/libtransmission/libtransmission.a \
+    $(top_builddir)/third-party/libevent/libevent.la \
+    $(top_builddir)/third-party/miniupnp/libminiupnp.a \
+    $(OPENSSL_LIBS) $(PTHREAD_LIBS) -lm
+
