--- transmission-0.96/libtransmission/transmission.h	2007-12-28 10:16:20.000000000 +0100
+++ transmission-r4345/libtransmission/transmission.h	2007-12-28 10:15:17.000000000 +0100
@@ -206,7 +206,15 @@
 int tr_torrentCount( tr_handle * h );
 
 
+/***********************************************************************
+ * tr_torrentIterate
+ ***********************************************************************
+ * Iterates on open torrents
+ **********************************************************************/
 typedef struct tr_torrent tr_torrent;
+typedef void (*tr_callback_t) ( tr_torrent *, void * );
+void tr_torrentIterate( tr_handle *, tr_callback_t, void * );
+
 
 /***********************************************************************
 *** Speed Limits
--- transmission-0.96/libtransmission/transmission.c	2007-12-28 10:16:20.000000000 +0100
+++ transmission-r4345/libtransmission/transmission.c	2007-12-28 10:15:17.000000000 +0100
@@ -359,6 +359,18 @@
     return h->torrentCount;
 }
 
+void
+tr_torrentIterate( tr_handle * h, tr_callback_t func, void * d )
+{
+    tr_torrent * tor, * next;
+
+    for( tor = h->torrentList; tor; tor = next )
+    {
+        next = tor->next;
+        func( tor, d );
+    }
+}
+
 static void
 tr_closeImpl( void * vh )
 {
