--- libtransmission/platform.c.orig	2008-11-10 15:52:25.000000000 +0000
+++ libtransmission/platform.c	2008-11-13 01:02:06.957161617 +0000
@@ -584,7 +584,7 @@
             /* XDG_DATA_DIRS are the backup directories */
             tmp = getenv( "XDG_DATA_DIRS" );
             if( !tmp || !*tmp )
-                tmp =  PACKAGE_DATA_DIR ":/usr/local/share/:/usr/share/";
+                tmp =  PACKAGE_DATA_DIR ":/opt/share/";
             while( tmp && *tmp ) {
                 const char * end = strchr( tmp, ':' );
                 if( end ) {
