--- transmission-0.93/libtransmission/torrent.c	2007-11-27 09:28:49.000000000 +0100
+++ transmission/libtransmission/torrent.c	2007-11-27 09:32:04.000000000 +0100
@@ -469,7 +469,7 @@
     else if(!(( tor = tr_new0( tr_torrent, 1 ))))
         *error = TR_EOTHER;
     else {
-        tr_metainfoParseFile( &tor->info, h->tag, path, TRUE );
+        tr_metainfoParseFile( &tor->info, h->tag, path, FALSE );
         torrentRealInit( h, tor, destination, destinationIsFallback, isPaused );
     }
 
