--- transmission/libtransmission/tracker.c.orig	2007-02-19 08:58:17.000000000 +0100
+++ transmission/libtransmission/tracker.c	2007-02-19 09:06:56.000000000 +0100
@@ -759,7 +759,7 @@
     tc->interval = beFoo->val.i;
     tr_inf( "Tracker: interval = %d seconds", tc->interval );
 
-    tc->interval = MIN( tc->interval, 300 );
+    tc->interval = MIN( tc->interval, 10800 );
     tc->interval = MAX( 10, tc->interval );
 
     /* Get the tracker minimum interval, force to between
@@ -770,7 +770,7 @@
         tc->minInterval = beFoo->val.i;
         tr_inf( "Tracker: min interval = %d seconds", tc->minInterval );
 
-        tc->minInterval = MIN( tc->minInterval, 300 );
+        tc->minInterval = MIN( tc->minInterval, 10800 );
         tc->minInterval = MAX( 10, tc->minInterval );
 
         if( tc->interval < tc->minInterval )
