diff -urN -x '.*' Transmission-0.7/mk/transmissiond.mk transmission/mk/transmissiond.mk
--- Transmission-0.6.1/mk/transmissiond.mk	1970-01-01 01:00:00.000000000 +0100
+++ transmission/mk/transmissiond.mk	2006-08-10 11:40:14.000000000 +0200
@@ -0,0 +1,35 @@
+# $Id: d.mk 265 2006-05-31 12:54:53Z titer $
+
+include ../mk/config.mk
+include ../mk/common.mk
+
+SRCS = transmissiond.c
+OBJS = $(SRCS:%.c=%.o)
+
+CFLAGS  += -I../libtransmission
+LDFLAGS += ../libtransmission/libtransmission.a
+
+transmissiond: $(OBJS) ../libtransmission/libtransmission.a
+	$(LINK_RULE)
+
+%.o: %.c ../mk/config.mk ../mk/common.mk ../mk/transmissiond.mk
+	$(CC_RULE)
+
+clean:
+	@echo "Clean transmissiond"
+	@$(RM) transmissiond
+	@echo "Clean $(OBJS)"
+	@$(RM) $(OBJS)
+
+.depend: $(SRCS) ../mk/config.mk ../mk/common.mk ../mk/transmissiond.mk
+	$(DEP_RULE)
+
+install: install-bin install-man
+
+install-bin: transmissiond
+	$(INSTALL_BIN_RULE)
+
+install-man:
+#	$(INSTALL_MAN_RULE)
+
+-include .depend
diff -urN -x '.*' Transmission-0.6.1/mk/default.mk transmission/mk/default.mk
--- Transmission-0.6.1/mk/default.mk	2006-06-25 23:34:31.000000000 +0200
+++ transmission/mk/default.mk	2006-08-10 11:40:14.000000000 +0200
@@ -3,7 +3,7 @@
 include mk/config.mk
 include mk/common.mk
 
-TARGETS = .cli
+TARGETS = .cli .transmissiond
 ifeq ($(GTK),yes)
 TARGETS += .gtk
 endif
@@ -21,6 +21,10 @@
 	@echo "* Building Transmission CLI client"
 	@$(MAKE) -C cli -f ../mk/cli.mk
 
+.transmissiond: .lib
+	@echo "* Building Transmission daemon for CGI interface"
+	@$(MAKE) -C cli -f ../mk/transmissiond.mk
+
 .gtk: .lib
 	@echo "* Building Transmission GTK+ client"
 	@$(MAKE) -C gtk -f ../mk/gtk.mk
@@ -35,6 +39,11 @@
 	@echo "* Installing Transmission CLI client"
 	@$(MAKE) -C cli -f ../mk/cli.mk install
 
+.install.transmissiond: .transmissiond
+	@echo "* Installing Transmission daemon for CGI interface"
+	@$(MAKE) -C cli -f ../mk/transmissiond.mk install
+
+
 .install.gtk: .gtk
 	@echo "* Installing Transmission GTK+ client"
 	@$(MAKE) -C gtk -f ../mk/gtk.mk install
