--- a/libtransmission/platform-quota.c
+++ libtransmission/platform-quota.c
@@ -286,7 +286,11 @@ getquota (const char * device)
 #elif defined(__APPLE__)
       spaceused = (int64_t) dq.dqb_curbytes;
 #elif defined(__UCLIBC__)
+#if _LINUX_QUOTA_VERSION < 2
       spaceused = (int64_t) btodb(dq.dqb_curblocks);
+#else
+      spaceused = (int64_t) btodb(dq.dqb_curspace);
+#endif
 #elif defined(__sun) || (defined(_LINUX_QUOTA_VERSION) && _LINUX_QUOTA_VERSION < 2)
       spaceused = (int64_t) dq.dqb_curblocks >> 1;
 #else
