--- git/read-cache.c.orig	2005-04-15 23:17:38.000000000 +0930
+++ git/read-cache.c	2005-04-15 23:20:02.000000000 +0930
@@ -288,17 +288,9 @@
 	/* nsec seems unreliable - not all filesystems support it, so
 	 * as long as it is in the inode cache you get right nsec
 	 * but after it gets flushed, you get zero nsec. */
-	if (ce->mtime.sec  != (unsigned int)st->st_mtim.tv_sec
-#ifdef NSEC
-	    || ce->mtime.nsec != (unsigned int)st->st_mtim.tv_nsec
-#endif
-	    )
+	if (ce->mtime.sec  != (unsigned int)st->st_mtime)
 		changed |= MTIME_CHANGED;
-	if (ce->ctime.sec  != (unsigned int)st->st_ctim.tv_sec
-#ifdef NSEC
-	    || ce->ctime.nsec != (unsigned int)st->st_ctim.tv_nsec
-#endif
-	    )
+	if (ce->ctime.sec  != (unsigned int)st->st_ctime)
 		changed |= CTIME_CHANGED;
 	if (ce->st_uid != (unsigned int)st->st_uid ||
 	    ce->st_gid != (unsigned int)st->st_gid)
