--- src/listener.c.orig	2007-12-07 07:16:47.000000000 -0800
+++ src/listener.c	2008-01-30 09:42:19.000000000 -0800
@@ -329,7 +329,11 @@
 	memset(&hints, 0, sizeof(hints));
 	hints.ai_family = PF_UNSPEC;
 	hints.ai_socktype = SOCK_STREAM;
+#ifdef AI_ADDRCONFIG
 	hints.ai_flags = AI_ADDRCONFIG | AI_PASSIVE;
+#else
+	hints.ai_flags = AI_PASSIVE;
+#endif
 
 	g_assert(!l->active);
 
--- lib/network.c.orig	2007-12-07 07:16:47.000000000 -0800
+++ lib/network.c	2008-01-30 09:47:59.000000000 -0800
@@ -656,7 +656,11 @@
 
 	memset(&hints_bind, 0, sizeof(hints_bind));
 	hints_bind.ai_family = res->ai_family;
+#ifdef AI_ADDRCONFIG
 	hints_bind.ai_flags = AI_ADDRCONFIG;
+#else
+	hints_bind.ai_flags = 0;
+#endif
 	hints_bind.ai_socktype = res->ai_socktype;
 	hints_bind.ai_protocol = res->ai_protocol;
 
@@ -731,7 +735,11 @@
 	memset(&hints, 0, sizeof(hints));
 	hints.ai_family = PF_UNSPEC;
 	hints.ai_socktype = SOCK_STREAM;
+#ifdef AI_ADDRCONFIG
 	hints.ai_flags = AI_ADDRCONFIG;
+#else
+	hints.ai_flags = 0;
+#endif
 
 	/* Lookup */
 	error = getaddrinfo(cs->host, cs->port, &hints, &addrinfo);
