--- swi-prolog/src/configure.in.orig	2007-07-27 02:54:33.000000000 -0700
+++ swi-prolog/src/configure.in	2007-08-02 19:55:03.000000000 -0700
@@ -383,16 +383,9 @@
   AC_CHECK_FUNCS(pthread_mutexattr_setkind_np pthread_mutexattr_settype)
   AC_CHECK_FUNCS(pthread_kill)
   AC_MSG_CHECKING("whether mutex support recursive locking")
-  echo '#include "confdefs.h"' > conftest.c
-  cat $srcdir/test/recmutex.c >> conftest.c
-  (eval $ac_link) 2>&5
-  if test -s conftest && ./conftest; then
     AC_MSG_RESULT(yes)
     AC_DEFINE(RECURSIVE_MUTEXES, 1,
 	      [Define to make use of standard (UNIX98) pthread recursive mutexes])
-  else
-    AC_MSG_RESULT(no)
-  fi
   AC_CHECK_FUNCS(pthread_sigmask pthread_setconcurrency)
 
 dnl ATOMIC inc/dec for reference counts
@@ -617,20 +610,7 @@
 AC_CHECK_FUNCS(clock_gettime)
 
 AC_MSG_CHECKING(for pthread support for cpu clocks)
-AC_TRY_RUN(
-[ #include <pthread.h>
-  #include <time.h>
-main()
-{
-  clockid_t clock_id;
-  struct timespec ts;
-  pthread_getcpuclockid(pthread_self(), &clock_id);
-  sleep(1);
-  clock_gettime(clock_id, &ts);
-  exit (ts.tv_sec);
-}], ac_pthread_cpuclocks="yes"
-    AC_MSG_RESULT(yes),
-    AC_MSG_RESULT(no))
+    ac_pthread_cpuclocks="yes"; AC_MSG_RESULT(yes)
 
 if test "x$ac_pthread_cpuclocks" = "xyes"; then
    AC_DEFINE(PTHREAD_CPUCLOCKS, 1,
@@ -645,18 +625,7 @@
    AC_MSG_RESULT(no))
 
 AC_MSG_CHECKING(for syscall support for cpu clocks)
-AC_TRY_RUN(
-[#include <time.h>
- #include <sys/syscall.h>
-main()
-{
-  struct timespec ts;
-  sleep(1);
-  syscall(__NR_clock_gettime, CLOCK_THREAD_CPUTIME_ID, &ts);
-  exit (ts.tv_sec);
-}], ac_linux_cpuclocks="yes"
-    AC_MSG_RESULT(yes),
-    AC_MSG_RESULT(no))
+    AC_MSG_RESULT(no)
 
 if test "x$ac_linux_cpuclocks" = "xyes"; then
    AC_DEFINE(LINUX_CPUCLOCKS, 1,
@@ -664,16 +633,9 @@
 else
 
 AC_MSG_CHECKING(linux 2.6 /proc)
-if test -f /proc/$$/task/$$/stat -a \
-	`cat /proc/$$/task/$$/stat | wc -w` = "41"; then
-   AC_MSG_RESULT(yes)
-   AC_DEFINE(LINUX_PROCFS, 26,
-	     [Define if you have Linux 2.6 compatible /proc])
-else
    AC_MSG_RESULT(no)
 fi
 fi
-fi
 
 AC_MSG_CHECKING(for sysconf support for _SC_NPROCESSORS_CONF)
 AC_TRY_COMPILE(
@@ -1174,15 +1136,9 @@
 dnl ****************************************************************
 
 AC_MSG_CHECKING("memory model");
-echo '#include "confdefs.h"' > conftest.c
-cat $srcdir/test/memmodel.c >> conftest.c
-(eval $ac_link) 2>&5
-if test -s conftest && eval `./conftest`; then
+STACK_DIRECTION=-1
     AC_DEFINE_UNQUOTED(STACK_DIRECTION, $STACK_DIRECTION)
     AC_MSG_RESULT(yes)
-else
-    AC_MSG_WARN("Failed to compile test/memmodel.c")
-fi
 rm -fr conftest*
 
 dnl ****************************************************************
@@ -1191,16 +1147,7 @@
 dnl ****************************************************************
 
 AC_MSG_CHECKING("for nop");
-echo '#include "confdefs.h"' > conftest.c
-cat $srcdir/test/nop.c >> conftest.c
-(eval $ac_link) 2>&5
-if test -s conftest && eval `./conftest`; then
     AC_MSG_RESULT(yes)
-else
-    AC_DEFINE(NO_ASM_NOP, 1,
-	      "Define if you can't use asm("nop") to separate two labels")
-    AC_MSG_RESULT(no)
-fi
 rm -fr conftest*
 
 
@@ -1289,13 +1236,11 @@
 mmap_tried=nosegv;
 mmap_test=mmap.c
 AC_MSG_CHECKING("if we can use mmap for stack allocation")
-while test ! "$MMAP_STACK" = "1" -a ! $mmap_tried = "stop"; do
 echo '#include "confdefs.h"' > conftest.c
 cat $srcdir/test/$mmap_test >> conftest.c
-(eval $ac_link) 2>&5
-if test -s conftest; then
-    res=`./conftest`
-    if test "$?" = 0 && eval $res; then
+    MMAP_STACK=1
+    HAVE_MAP_ANON=1
+    mmap_tried=segv
 	if test "$MMAP_STACK" = "1"; then
 	    if test "$HAVE_MAP_ANON" = "1"; then
 		AC_DEFINE(HAVE_MAP_ANON, 1,
@@ -1320,11 +1265,6 @@
 	else
 	    AC_MSG_RESULT("mmap is not good enough")
 	fi
-    else
-        AC_MSG_RESULT("could not run mmap test");
-    fi
-fi
-rm -fr conftest*
 if test ! "$MMAP_STACK" = "1"; then
   if test "$mmap_tried" = "sgi"; then
     mmap_tried=segv
@@ -1338,7 +1278,7 @@
 else
     mmap_tried=stop
 fi
-done
+    AC_DEFINE(NO_SEGV_HANDLING)
 fi # mmap
 
 case "$ARCH" in
