--- sqlite-3.2.1-orig/configure.ac	2005-03-23 08:08:39.000000000 -0800
+++ sqlite/configure.ac	2005-03-30 11:37:21.000000000 -0800
@@ -186,10 +186,10 @@
 default_build_cflags="-g"
 if test "$config_BUILD_CC" = ""; then
   AC_PROG_CC
-  if test "$cross_compiling" = "yes"; then
-    AC_MSG_ERROR([unable to find a compiler for building build tools])
-  fi
-  BUILD_CC=$CC
+#  if test "$cross_compiling" = "yes"; then
+#    AC_MSG_ERROR([unable to find a compiler for building build tools])
+#  fi
+  BUILD_CC=$HOSTCC
   default_build_cflags=$CFLAGS
 else
   BUILD_CC=$config_BUILD_CC
