--- Python-2.7/configure.in.orig	2010-05-28 15:28:47.000000000 +0000
+++ Python-2.7/configure.in	2010-09-05 01:50:11.256014039 +0000
@@ -12,7 +12,6 @@
        [],
        [m4_fatal([Autoconf version $1 is required for Python], 63)])
 ])
-version_required(2.65)
 
 AC_REVISION($Revision: 81582 $)
 AC_INIT(python, PYTHON_VERSION, http://bugs.python.org/)
@@ -28,6 +27,11 @@
     prefix=`echo "$prefix" | sed -e 's/\/$//g'`
 fi    
 
+# find compiler while respecting --host setting
+AC_CANONICAL_HOST()
+AC_CHECK_TOOLS(CC,gcc cc)
+AC_CHECK_TOOLS(CXX,g++ c++)
+
 dnl This is for stuff that absolutely must end up in pyconfig.h.
 dnl Please use pyport.h instead, if possible.
 AH_TOP([
@@ -916,7 +920,7 @@
 	    STRICT_PROTO="-Wstrict-prototypes"
 	fi
         # For gcc 4.x we need to use -fwrapv so lets check if its supported
-        if "$CC" -v --help 2>/dev/null |grep -- -fwrapv > /dev/null; then
+        if "$TARGET_CC" -v --help 2>/dev/null |grep -- -fwrapv > /dev/null; then
            WRAP="-fwrapv"
         fi
 	case $ac_cv_prog_cc_g in
@@ -4258,6 +4262,25 @@
 done
 AC_MSG_RESULT(done)
 
+AC_SUBST(BUILDPYTHON)
+AC_SUBST(BUILDPGEN)
+AC_SUBST(CROSS_COMPILE)
+if [[ $build != $host ]]; then
+    AC_MSG_NOTICE(Cross compiling: Configuring build python)
+    absconfigcommand=$(pwd)/$0
+    mkdir -p buildpython27
+    cd buildpython27
+    CC="" CXX="" AR="" RANLIB="" LDFLAGS="" $absconfigcommand --prefix=$prefix
+    cd ..
+    BUILDPYTHON='buildpython27/python$(BUILDEXE)'
+    BUILDPGEN='buildpython27/Parser/pgen$(BUILDEXE)'
+    CROSS_COMPILE=yes
+else
+    BUILDPYTHON='$(HOSTPYTHON)'
+    BUILDPGEN='$(HOSTPGEN)'
+    CROSS_COMPILE=no
+fi
+
 # generate output files
 AC_CONFIG_FILES(Makefile.pre Modules/Setup.config Misc/python.pc)
 AC_OUTPUT
