--- Python-2.7.9/Lib/site.py.orig	2014-12-10 17:59:40.000000000 +0200
+++ Python-2.7.9/Lib/site.py	2015-01-26 20:00:59.514417000 +0200
@@ -64,7 +64,7 @@
 import traceback
 
 # Prefixes for site-packages; add additional prefixes like /usr/local here
-PREFIXES = [sys.prefix, sys.exec_prefix]
+PREFIXES = [os.path.join(sys.prefix, "local"), sys.prefix, sys.exec_prefix]
 # Enable per user site-packages directory
 # set it to False to disable the feature or True to force the feature
 ENABLE_USER_SITE = None
