--- ctorrent-dnh2/btfiles.cpp.orig	Sun Jan  1 18:41:21 2006
+++ ctorrent-dnh2-new/btfiles.cpp	Sun Mar 19 19:54:53 2006
@@ -170,11 +170,13 @@
   return write(fd, &c, 1);
 #else
   // ftruncate() not allowed on [v]fat under linux
-  if( ftruncate(fd,length) < 0 ) {
+  int retval = ftruncate(fd,length);
+  if( retval < 0 ) {
     char c = (char)0;
     if(lseek(fd,length - 1, SEEK_SET) < 0 ) return -1;
     return write(fd, &c, 1);
   }
+  else return retval;
 #endif
 }
 
