diff -u ctorrent-dnh2.old/peer.cpp ctorrent-dnh2/peer.cpp
--- ctorrent-dnh2.old/peer.cpp	2006-01-02 20:40:24.000000000 -0200
+++ ctorrent-dnh2/peer.cpp	2006-07-05 10:04:52.000000000 -0300
@@ -525,12 +525,13 @@
 int btPeer::CancelPiece()
 {
   PSLICE ps = request_q.GetHead();
+  PSLICE aux;
   size_t idx;
   int cancel = 1;
   int retval;
 
   idx = ps->index;
-  for( ; ps; ps = ps->next){
+  while( ps ){
     if( ps->index != idx ) break;
     if( ps == request_q.NextSend() ) cancel = 0;
     if( cancel ){
@@ -543,7 +544,9 @@
         m_req_out = 0;
       }
     }
+    aux = ps->next;
     request_q.Remove(ps->index, ps->offset, ps->length);
+    ps = aux;
   }
   if( !m_req_out && g_next_dn == this ) g_next_dn = (btPeer *)0;
 
