--- libsigc++-2.2.3/Makefile.in.orig	2008-10-14 08:52:18.000000000 -0700
+++ libsigc++-2.2.3/Makefile.in	2009-07-04 10:22:48.000000000 -0700
@@ -210,7 +210,7 @@
 # other stuff here
 EXTRA_DIST = autogen.sh sigc++config.h.in libsigc++-2.0.spec.in
 # ACLOCAL_FLAGS = -I scripts
-SUBDIRS = sigc++ tests docs examples scripts MSVC_Net2005 MSVC_Net2008
+SUBDIRS = sigc++ docs scripts
 DIST_SUBDIRS = $(SUBDIRS)
 sigc_configdir = $(libdir)/sigc++-2.0/include
 sigc_config_DATA = sigc++config.h
