--- a/src/otherfunctions.cpp	2005-01-09 21:42:45.000000000 +0100
+++ amuled/src/otherfunctions.cpp	2005-01-09 21:50:52.000000000 +0100
@@ -72,13 +72,13 @@
 {
 
 	if (count < 1024)
-		return wxString::Format( wxT("%.0f %s"), (float)count, _("Bytes") );
+		return wxString::Format( wxT("%.0f %s"), (float)(uint32)count, _("Bytes") );
 	else if (count < 1048576)
-		return wxString::Format( wxT("%.0f %s"), (float)count/1024, _("KB") );
+		return wxString::Format( wxT("%.0f %s"), (float)(uint32)count/1024, _("KB") );
 	else if (count < 1073741824)
-		return wxString::Format( wxT("%.2f %s"), (float)count/1048576, _("MB") );
+		return wxString::Format( wxT("%.2f %s"), (float)(uint32)count/1048576, _("MB") );
 	else if (count < 1099511627776LL)
-		return wxString::Format( wxT("%.2f %s"), (float)count/1073741824, _("GB") );
+		return wxString::Format( wxT("%.2f %s"), (float)((uint32)(count/1024))/1048576, _("GB") );
 	else
 		return wxString::Format( wxT("%.3f %s"), (float)count/1099511627776LL, _("TB") );
 
@@ -90,13 +90,13 @@
 {
 
 	if (count < 1000)
-		return wxString::Format(wxT("%llu"), count);
+		return wxString::Format(wxT("%lu"), (uint32)count);
 	else if (count < 1000000)
-		return wxString::Format(wxT("%.0f%s"),(float)count/1000, _("K") );
+		return wxString::Format(wxT("%.0f%s"),(float)(uint32)count/1000, _("K") );
 	else if (count < 1000000000)
-		return wxString::Format(wxT("%.2f%s"),(float)count/1000000, _("M") );
+		return wxString::Format(wxT("%.2f%s"),(float)(uint32)count/1000000, _("M") );
 	else if (count < 1000000000000LL)
-		return wxString::Format(wxT("%.2f%s"),(float)count/1000000000LL, _("G") );
+		return wxString::Format(wxT("%.2f%s"),(float)((uint32)(count/1000))/1000000, _("G") );
 	else if (count < 1000000000000000LL)
 		return wxString::Format(wxT("%.2f%s"),(float)count/1000000000000LL, _("T") );
 
