--- apxs.in.orig	2013-12-26 20:01:53.000000000 +0200
+++ support/apxs.in	2015-01-10 13:11:04.797756113 +0200
@@ -189,13 +189,14 @@
 ##
 ##  Initial shared object support check
 ##
-unless ("@MOD_SO_ENABLED@" eq "yes") {
-    error("Sorry, no shared object support for Apache");
-    error("available under your platform. Make sure");
-    error("the Apache module mod_so is compiled into");
-    error("the server binary.");
-    exit 1;
-}
+### these checks will not work when cross-compiling into a staging area!
+#unless ("@MOD_SO_ENABLED@" eq "yes") {
+#    error("Sorry, no shared object support for Apache");
+#    error("available under your platform. Make sure");
+#    error("the Apache module mod_so is compiled into");
+#    error("the server binary.");
+#    exit 1;
+#}
 
 sub get_config_vars{
     my ($file, $rh_config) = @_;
@@ -550,9 +551,10 @@
 
     #   activate module via LoadModule/AddModule directive
     if ($opt_a or $opt_A) {
+	### when staging, this should be non-fatal
         if (not -f "$CFG_SYSCONFDIR/$CFG_TARGET.conf") {
             error("Config file $CFG_SYSCONFDIR/$CFG_TARGET.conf not found");
-            exit(1);
+            exit(0);
         }
 
         open(FP, "<$CFG_SYSCONFDIR/$CFG_TARGET.conf") || die;
