--- docs/conf/httpd.conf.in
+++ docs/conf/httpd.conf.in
@@ -65,6 +65,8 @@
 #
 @@LoadModule@@
 
+Include etc/apache2/conf.d/*.conf
+
 <IfModule unixd_module>
 #
 # If you wish httpd to run as a different user or group, you must run
@@ -74,8 +76,8 @@
 # It is usually good practice to create a dedicated user and group for
 # running httpd, as with most system services.
 #
-User daemon
-Group daemon
+User nobody
+Group #-1
 
 </IfModule>
 
