--- mod_python-3.1.4/configure.in	2005-01-29 13:25:27.000000000 -0800
+++ mod-python/configure.in	2005-04-21 22:41:54.572376000 -0700
@@ -98,7 +98,7 @@
   # check Apache version
   AC_MSG_CHECKING(Apache version)
   HTTPD="`${APXS} -q SBINDIR`/`${APXS} -q TARGET`"
-  ver=`$HTTPD -v | awk '/version/ {print $3}' | awk -F/ '{print $2}'`
+  ver=@APACHE_VERSION@
   AC_MSG_RESULT($ver)
 
   # make sure version begins with 2
@@ -188,8 +188,8 @@
 
 # find out python version
 AC_MSG_CHECKING(Python version)
-PyVERSION=`$PYTHON_BIN -c ['import sys; print sys.version[:3]'`]
-PyMAJVERSION=`$PYTHON_BIN -c ['import sys; print sys.version[:1]'`]
+PyVERSION=2.4
+PyMAJVERSION=2
 AC_MSG_RESULT($PyVERSION)
 
 # make sure Python is version 2
@@ -199,7 +199,7 @@
 
 # find out compiled in install prefix
 AC_MSG_CHECKING(Python install prefix)
-PyEXEC_INSTALLDIR=`$PYTHON_BIN -c "import sys; print sys.exec_prefix"`
+PyEXEC_INSTALLDIR=`cd ../../staging; pwd`/opt
 AC_MSG_RESULT($PyEXEC_INSTALLDIR)
 
 # this is where the Python libraries will get installed
