Description: Fix deprecated av_opt related calls
Author: Sebastian Ramacher <sramacher@debian.org>
Last-Update: 2013-11-11

--- a/export/export_ffmpeg.c
+++ b/export/export_ffmpeg.c
@@ -240,6 +240,15 @@
     opt_codec(&subtitle_stream_copy, &subtitle_codec_name, AVMEDIA_TYPE_SUBTITLE, arg);
 }
 
+static int av_set_string3_fallback(void *obj, const char *name, const char *val, int alloc, const AVOption **o_out)
+{
+    const AVOption *o = av_opt_find(obj, name, NULL, 0, 0);
+    if (o_out)
+        *o_out = o;
+    return av_opt_set(obj, name, val, 0);
+}
+
+
 static
 int opt_default(const char *opt, const char *arg){
     int type;
@@ -250,9 +259,9 @@
     for(type=0; type<AVMEDIA_TYPE_NB && ret>= 0; type++){
 		/* GLUE: +if */
 		if (type == AVMEDIA_TYPE_VIDEO) {
-        const AVOption *o2 = av_find_opt(avcodec_opts[0], opt, NULL, opt_types[type], opt_types[type]);
+        const AVOption *o2 = av_opt_find(avcodec_opts[0], opt, NULL, opt_types[type], 0);
         if(o2)
-            ret = av_set_string3(avcodec_opts[type], opt, arg, 1, &o);
+            ret = av_set_string3_fallback(avcodec_opts[type], opt, arg, 1, &o);
 		/* GLUE: +if */
 		}
     }
@@ -267,7 +276,7 @@
         if(opt[0] == 'a')
             ret = av_set_string3(avcodec_opts[AVMEDIA_TYPE_AUDIO], opt+1, arg, 1, &o);
         else */ if(opt[0] == 'v')
-            ret = av_set_string3(avcodec_opts[AVMEDIA_TYPE_VIDEO], opt+1, arg, 1, &o);
+            ret = av_set_string3_fallback(avcodec_opts[AVMEDIA_TYPE_VIDEO], opt+1, arg, 1, &o);
 		/* GLUE: disabling
         else if(opt[0] == 's')
             ret = av_set_string3(avcodec_opts[AVMEDIA_TYPE_SUBTITLE], opt+1, arg, 1, &o);
