--- transcode-1.0.3/export/aud_aux.c-org	2007-07-07 13:34:28.000000000 +0200
+++ transcode-1.0.3/export/aud_aux.c	2007-07-07 13:34:56.000000000 +0200
@@ -361,8 +361,7 @@
 
     pthread_mutex_lock(&init_avcodec_lock);
     avcodec_init();
-    register_avcodec(&ac3_encoder);
-    register_avcodec(&mp2_encoder);
+    avcodec_register_all();
     pthread_mutex_unlock(&init_avcodec_lock);
 
     switch (o_codec) {
