--- Python-2.4.2/Lib/site.py.orig	2005-12-27 20:59:19.000000000 -0800
+++ Python-2.4.2/Lib/site.py	2005-12-27 21:00:49.000000000 -0800
@@ -170,7 +170,7 @@
 
 def addsitepackages(known_paths):
     """Add site-packages (and possibly site-python) to sys.path"""
-    prefixes = [sys.prefix]
+    prefixes = [os.path.join(sys.prefix, "local"), sys.prefix]
     if sys.exec_prefix != sys.prefix:
         prefixes.append(sys.exec_prefix)
     for prefix in prefixes:
