--- a/scripts/build/libc/glibc.sh
+++ b/scripts/build/libc/glibc.sh
@@ -256,7 +256,7 @@ do_libc_backend_once() {
     RANLIB=${CT_TARGET}-ranlib                                      \
     "${CONFIG_SHELL}"                                               \
     "${src_dir}/configure"                                          \
-        --prefix=/usr                                               \
+        --prefix=%OPTWARE_TARGET_PREFIX%                                               \
         --build=${CT_BUILD}                                         \
         --host=${multi_target}                                      \
         --cache-file="$(pwd)/config.cache"                          \
@@ -307,6 +307,7 @@ do_libc_backend_once() {
         # use the 'install-headers' makefile target to install the
         # headers
         CT_DoExecLog ALL make ${JOBSFLAGS}                       \
+                         prefix=/usr                                \
                          install_root=${multi_root}                 \
                          install-bootstrap-headers=yes              \
                          "${extra_make_args[@]}"                    \
@@ -384,6 +385,7 @@ do_libc_backend_once() {
         CT_DoLog EXTRA "Installing C library"
         CT_DoExecLog ALL make ${JOBSFLAGS}                 \
                               "${extra_make_args[@]}"         \
+                              prefix=/usr                     \
                               install_root="${multi_root}"    \
                               install
 
@@ -518,7 +520,7 @@ do_libc_locales() {
     CFLAGS="${glibc_cflags}"               \
     ${CONFIG_SHELL}                        \
     "${src_dir}/configure"                 \
-        --prefix=/usr                      \
+        --prefix=%OPTWARE_TARGET_PREFIX%                      \
         --cache-file="$(pwd)/config.cache" \
         --without-cvs                      \
         --disable-profile                  \
@@ -537,6 +539,7 @@ do_libc_locales() {
 
     CT_DoLog EXTRA "Installing C library locales"
     CT_DoExecLog ALL make ${JOBSFLAGS}                  \
+                          prefix=/usr                      \
                           install_root="${CT_SYSROOT_DIR}" \
                           localedata/install-locales
 }
