--- a/config/kernel/linux.in
+++ b/config/kernel/linux.in
@@ -78,7 +78,7 @@ config KERNEL_V_3_4
 
 config KERNEL_V_3_2
     bool
-    prompt "3.2.77"
+    prompt "3.2.66"
 
 config KERNEL_V_2_6_32
     bool
@@ -99,7 +99,7 @@ config KERNEL_VERSION
     default "3.12.55" if KERNEL_V_3_12
     default "3.10.98" if KERNEL_V_3_10
     default "3.4.110" if KERNEL_V_3_4
-    default "3.2.77" if KERNEL_V_3_2
+    default "3.2.66" if KERNEL_V_3_2
     default "2.6.32.70" if KERNEL_V_2_6_32
 
 endif # ! KERNEL_LINUX_CUSTOM
