--- a/config/kernel/linux.in
+++ b/config/kernel/linux.in
@@ -82,7 +82,7 @@
 
 config KERNEL_V_2_6_32
     bool
-    prompt "2.6.32.70"
+    prompt "2.6.32"
     help
  
 endchoice
@@ -100,6 +100,6 @@
     default "3.10.98" if KERNEL_V_3_10
     default "3.4.110" if KERNEL_V_3_4
     default "3.2.77" if KERNEL_V_3_2
-    default "2.6.32.70" if KERNEL_V_2_6_32
+    default "2.6.32" if KERNEL_V_2_6_32
 
 endif # ! KERNEL_LINUX_CUSTOM
