--- mod_python-3.5.0/configure.in.orig	2015-02-06 21:52:43.293275136 +0200
+++ mod_python-3.5.0/configure.in	2015-02-07 11:28:09.293257767 +0200
@@ -76,7 +76,7 @@
   # check Apache version
   AC_MSG_CHECKING(Apache version)
   HTTPD="`${APXS} -q SBINDIR`/`${APXS} -q TARGET`"
-  HTTPD_VERSION=`$HTTPD -v | awk '/version/ {print $3}' | awk -F/ '{print $2}' | awk '{print $1}'`
+  HTTPD_VERSION=@APACHE_VERSION@
   APR_VERSION=`${APXS} -q APR_VERSION`
   AC_MSG_RESULT($HTTPD_VERSION)
 
@@ -167,9 +167,9 @@
 
 # find out python version
 AC_MSG_CHECKING(Python version)
-PyVERSION=`$PYTHON_BIN -c ['import sys; print(sys.version[:3])'`]
-PyMAJVERSION=`$PYTHON_BIN -c ['import sys; print(sys.version[:1])'`]
-PyMINVERSION=`$PYTHON_BIN -c ['import sys; print(sys.version.split(".")[1])'`]
+PyVERSION=2.7
+PyMAJVERSION=2
+PyMINVERSION=7
 AC_MSG_RESULT($PyVERSION)
 if test -z "$PyMAJVERSION"; then
   AC_MSG_ERROR([Unable to get Python version - is your Python working?])
