--- Makefile.in.orig	2007-05-31 23:18:12.000000000 -0700
+++ Makefile.in	2007-05-31 23:21:02.000000000 -0700
@@ -1107,7 +1107,7 @@
 gst.im: $(bin_PROGRAMS) $(srcdir)/kernel/stamp-classes
 	SMALLTALK_KERNEL="`cd $(srcdir)/kernel; pwd`" \
 	SMALLTALK_IMAGE="`pwd`" \
-	  ./gst -iQ /dev/null
+	  $(GNU_SMALLTALK_HOST_BUILD_DIR)/gst -iQ /dev/null
 
 all-local:
 
@@ -1141,7 +1141,7 @@
 	cd $(DESTDIR)$(imagedir) && \
 	SMALLTALK_KERNEL=$(DESTDIR)$(pkgdatadir)/kernel \
 	SMALLTALK_IMAGE=$(DESTDIR)$(imagedir) \
-	  $(DESTDIR)$(bindir)/gst -f $$srcdir/scripts/Finish.st \
+	  $(GNU_SMALLTALK_HOST_BUILD_DIR)/gst -f $$srcdir/scripts/Finish.st \
 	    "$(pkgdatadir)" "$(imagedir)" $(MODULES)
 
 uninstall-local:: gst-package
@@ -1152,7 +1152,7 @@
 installcheck-local:
 	if test -n "$(DESTDIR)"; then :; else \
 	  echo '[ PackageLoader fileInPackages: #(#MD5 #Regex) ] \
-	    ifCurtailed: [ ObjectMemory quit: 1 ]!' | $(bindir)/gst -q; \
+	    ifCurtailed: [ ObjectMemory quit: 1 ]!' | $(GNU_SMALLTALK_HOST_BUILD_DIR)/gst -q; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
