--- ion-3ds-20050607-orig/configure.ac	2005-06-07 06:06:03.000000000 -0700
+++ ion/configure.ac	2005-06-14 21:10:34.694321000 -0700
@@ -250,28 +250,7 @@
 save_LIBS="$LIBS"
 CFLAGS="$CFLAGS $LUA_INCLUDES"
 LIBS="$LIBS $LUA_LIBS"
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-
-#include <lua.h>
-#include <stdlib.h>
-#include <string.h>
-
-int main()
-{
-    if(strcmp(lua_version(), LUA_VERSION)!=0){
-        printf("Error: lua_version() does not match LUA_VERSION. ");
-        exit(EXIT_FAILURE);
-    }
-    printf("(found %s)... ", lua_version());
-    if(strcmp(lua_version(), "Lua 5.0")<0)
-        exit(EXIT_FAILURE);
-    exit(EXIT_SUCCESS);
-}
-
-]])],
-              [AC_MSG_RESULT([yes])],
-              [AC_MSG_RESULT([no])
-               exit 1])
+              AC_MSG_RESULT([yes])
 CFLAGS="$save_CFLAGS"
 LIBS="$save_LIBS"
 dnl }}}
@@ -342,30 +321,8 @@
 AC_MSG_CHECKING([for va_copy])
 save_CFLAGS="$CFLAGS"
 CFLAGS="$CFLAGS $C99_SOURCE"
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-
-#include <stdarg.h>
-#include <stdlib.h>
-
-void f(void *last, ...)
-{
-    va_list ap, aq;
-    va_start(ap, last);
-    va_copy(aq, ap);
-    va_end(ap);
-    exit(va_arg(aq, int));
-}
-
-int main()
-{
-    f(NULL, EXIT_SUCCESS);
-    exit(EXIT_FAILURE);
-}
-
-]])],
-                  [AC_MSG_RESULT([yes])
-                   _DCF_HAS_VA_COPY="-DCF_HAVE_VA_COPY"],
-                  [AC_MSG_RESULT([no])])
+                  AC_MSG_RESULT([yes])
+                   _DCF_HAS_VA_COPY="-DCF_HAVE_VA_COPY"
 CFLAGS="$save_CFLAGS"
 
 dnl }}}
