--- stunnel/tools/Makefile.in.orig	2005-01-07 17:05:12.000000000 +0100
+++ stunnel/tools/Makefile.in	2005-01-07 17:05:26.000000000 +0100
@@ -163,7 +163,7 @@
 	stunnel.spec stunnel.mak stunnel.cnf
 
 confdir = $(sysconfdir)/stunnel
-conf_DATA = stunnel.conf-sample stunnel.pem
+conf_DATA = stunnel.conf-sample
 docdir = $(datadir)/doc/stunnel
 examplesdir = $(docdir)/examples
 examples_DATA = ca.html ca.pl importCA.html importCA.sh script.sh \
