--- dropbear/options.h~	2004-10-20 10:15:00.000000000 +0930
+++ dropbear/options.h	2004-10-20 10:16:06.000000000 +0930
@@ -173,7 +173,7 @@
 
 /* This is used by the scp binary when used as a client binary. If you're
  * not using the Dropbear client, you'll need to change it */
-#define _PATH_SSH_PROGRAM "/usr/bin/dbclient"
+#define _PATH_SSH_PROGRAM "/opt/bin/dbclient"
 
 /* Multi-purpose binary configuration has now moved. Look at the top
  * of the Makefile for instructions, or INSTALL */
