--- a/configure
+++ configure
@@ -19392,19 +19392,28 @@
 fi
 
 
-have_inotify=no
-for ac_header in sys/inotify.h
-do :
-  ac_fn_c_check_header_mongrel "$LINENO" "sys/inotify.h" "ac_cv_header_sys_inotify_h" "$ac_includes_default"
-if test "x$ac_cv_header_sys_inotify_h" = xyes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_SYS_INOTIFY_H 1
-_ACEOF
- have_inotify=yes
-fi
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
 
-done
+#include <sys/inotify.h>
+
+int
+main ()
+{
 
+inotify_init();
+
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"; then :
+  have_inotify=yes
+else
+  have_inotify=no
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
 
  if test "x$have_inotify" = "xyes"; then
   HAVE_INOTIFY_TRUE=
@@ -19478,14 +19487,6 @@
 INTLTOOL_REQUIRED_VERSION_AS_INT=`echo 0.35.0 | awk -F. '{ print $ 1 * 1000 + $ 2 * 100 + $ 3; }'`
 INTLTOOL_APPLIED_VERSION=`intltool-update --version | head -1 | cut -d" " -f3`
 INTLTOOL_APPLIED_VERSION_AS_INT=`echo $INTLTOOL_APPLIED_VERSION | awk -F. '{ print $ 1 * 1000 + $ 2 * 100 + $ 3; }'`
-if test -n "0.35.0"; then
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for intltool >= 0.35.0" >&5
-$as_echo_n "checking for intltool >= 0.35.0... " >&6; }
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $INTLTOOL_APPLIED_VERSION found" >&5
-$as_echo "$INTLTOOL_APPLIED_VERSION found" >&6; }
-    test "$INTLTOOL_APPLIED_VERSION_AS_INT" -ge "$INTLTOOL_REQUIRED_VERSION_AS_INT" ||
-	as_fn_error $? "Your intltool is too old.  You need intltool 0.35.0 or later." "$LINENO" 5
-fi
 
 # Extract the first word of "intltool-update", so it can be a program name with args.
 set dummy intltool-update; ac_word=$2
@@ -23523,7 +23524,7 @@
 #
 # Avahi runtime dir
 #
-avahi_runtime_dir="/run"
+avahi_runtime_dir="\${prefix}/var/run"
 avahi_socket="${avahi_runtime_dir}/avahi-daemon/socket"
 
 
