--- lighttpd/configure.in.orig	2006-03-06 13:51:19.000000000 -0800
+++ lighttpd/configure.in	2006-03-06 13:51:32.000000000 -0800
@@ -250,23 +250,8 @@
     [WITH_PCRE=$withval],[WITH_PCRE=yes])
 AC_MSG_RESULT([$WITH_PCRE])
 
-if test "x$cross_compiling" = xno -a "$WITH_PCRE" != "no"; then
-  AC_PATH_PROG(PCRECONFIG, pcre-config)
-
-  if test x"$PCRECONFIG" != x; then 
-    PCRE_LIB=`$PCRECONFIG --libs`
-    CPPFLAGS="$CPPFLAGS `$PCRECONFIG --cflags`"
-    OLDLIBS="$LIBS"
-    LIBS="$LIBS $PCRE_LIB"
-    AC_CHECK_LIB(pcre, pcre_compile, [
-      AC_CHECK_HEADERS([pcre.h], [
         AC_DEFINE([HAVE_LIBPCRE], [1], [libpcre])
         AC_DEFINE([HAVE_PCRE_H], [1])
-      ])
-    ])
-    LIBS="$OLDLIBS"
-  fi
-fi
 
 AC_SUBST(PCRE_LIB)
 
