--- src/dnsmasq.h.orig	2007-02-05 21:38:36.000000000 -0800
+++ src/dnsmasq.h	2007-02-05 21:39:53.000000000 -0800
@@ -75,6 +75,12 @@
 
 #ifdef HAVE_LINUX_NETWORK
 #include <linux/capability.h>
+#ifndef ARPHRD_IEEE1394
+#   define ARPHRD_IEEE1394 24
+#endif
+#ifndef ARPHRD_EUI64
+#   define ARPHRD_EUI64 27
+#endif
 /* There doesn't seem to be a universally-available 
    userpace header for this. */
 extern int capset(cap_user_header_t header, cap_user_data_t data);
