--- kismet-2013-03-R1b/Makefile.in~	2013-03-27 16:41:48.000000000 +0200
+++ kismet-2013-03-R1b/Makefile.in	2015-04-26 18:33:10.045736164 +0300
@@ -105,7 +105,7 @@
 	@-echo "'Makefile.in' or 'configure' are more current than this Makefile.  You should re-run 'configure'."
 
 binsuidinstall:
-	$(INSTALL) -o $(INSTUSR) -g $(SUIDGROUP) -m 4550 $(CS) $(BIN)/$(CS); 
+	$(INSTALL) -m 4550 $(CS) $(BIN)/$(CS); 
 	
 commoninstall:
 	mkdir -p $(ETC)
@@ -113,26 +113,26 @@
 
 	if test -e $(NC); then \
 		echo "Installing client"; \
-		$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 555 scripts/kismet $(BIN)/kismet; \
-		$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 555 $(NC) $(BIN)/$(NC); \
+		$(INSTALL) -m 555 scripts/kismet $(BIN)/kismet; \
+		$(INSTALL) -m 555 $(NC) $(BIN)/$(NC); \
 	fi;
-	$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 555 $(PS) $(BIN)/$(PS); 
-	$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 555 $(DRONE) $(BIN)/$(DRONE);
+	$(INSTALL) -m 555 $(PS) $(BIN)/$(PS); 
+	$(INSTALL) -m 555 $(DRONE) $(BIN)/$(DRONE);
 
 	mkdir -p $(MAN)/man1
-	$(INSTALL) -o $(INSTUSR) -g $(MANGRP) -m 644 man/kismet.1 $(MAN)/man1/kismet.1
-	$(INSTALL) -o $(INSTUSR) -g $(MANGRP) -m 644 man/kismet_drone.1 $(MAN)/man1/kismet_drone.1
+	$(INSTALL) -m 644 man/kismet.1 $(MAN)/man1/kismet.1
+	$(INSTALL) -m 644 man/kismet_drone.1 $(MAN)/man1/kismet_drone.1
 
 	mkdir -p $(MAN)/man5
-	$(INSTALL) -o $(INSTUSR) -g $(MANGRP) -m 644 man/kismet.conf.5 $(MAN)/man5/kismet.conf.5
-	$(INSTALL) -o $(INSTUSR) -g $(MANGRP) -m 644 man/kismet_drone.conf.5 $(MAN)/man5/kismet_drone.conf.5
+	$(INSTALL) -m 644 man/kismet.conf.5 $(MAN)/man5/kismet.conf.5
+	$(INSTALL) -m 644 man/kismet_drone.conf.5 $(MAN)/man5/kismet_drone.conf.5
 
 	mkdir -p $(WAV)
-	$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 644 wav/new.wav $(WAV)/new.wav
-	$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 644 wav/packet.wav $(WAV)/packet.wav
-	$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 644 wav/alert.wav $(WAV)/alert.wav
-	$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 644 wav/gpslost.wav $(WAV)/gpslost.wav
-	$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 644 wav/gpslock.wav $(WAV)/gpslock.wav
+	$(INSTALL) -m 644 wav/new.wav $(WAV)/new.wav
+	$(INSTALL) -m 644 wav/packet.wav $(WAV)/packet.wav
+	$(INSTALL) -m 644 wav/alert.wav $(WAV)/alert.wav
+	$(INSTALL) -m 644 wav/gpslost.wav $(WAV)/gpslost.wav
+	$(INSTALL) -m 644 wav/gpslock.wav $(WAV)/gpslock.wav
 
 suidinstall: $(CS)
 	-groupadd -f $(SUIDGROUP)
@@ -143,8 +143,8 @@
 		echo "if there have been any changes to the base config you will need"; \
 		echo "to add them to your config file."; \
     else \
-		$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 644 conf/kismet.conf $(ETC)/kismet.conf;  \
-		echo install -o $(INSTUSR) -g $(INSTGRP) -m 644 conf/kismet.conf $(ETC)/kismet.conf;  \
+		$(INSTALL) -m 644 conf/kismet.conf $(ETC)/kismet.conf;  \
+		echo install -m 644 conf/kismet.conf $(ETC)/kismet.conf;  \
 		echo "Installed config into $(ETC)/kismet.conf."; \
 	fi
 	@if test -f $(ETC)/kismet_drone.conf; then \
@@ -152,8 +152,8 @@
 		echo "if there have been any changes to the base config you will need"; \
 		echo "to add them to your config file."; \
     else \
-		$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 644 conf/kismet_drone.conf $(ETC)/kismet_drone.conf;  \
-		echo install -o $(INSTUSR) -g $(INSTGRP) -m 644 conf/kismet_drone.conf $(ETC)/kismet_drone.conf;  \
+		$(INSTALL) -m 644 conf/kismet_drone.conf $(ETC)/kismet_drone.conf;  \
+		echo install -m 644 conf/kismet_drone.conf $(ETC)/kismet_drone.conf;  \
 		echo "Installed drone config into $(ETC)/kismet_drone.conf."; \
 	fi
 
@@ -175,8 +175,8 @@
 		echo "if there have been any changes to the base config you will need"; \
 		echo "to add them to your config file."; \
     else \
-		$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 644 conf/kismet.conf $(ETC)/kismet.conf;  \
-		echo install -o $(INSTUSR) -g $(INSTGRP) -m 644 conf/kismet.conf $(ETC)/kismet.conf;  \
+		$(INSTALL) -m 644 conf/kismet.conf $(ETC)/kismet.conf;  \
+		echo install -m 644 conf/kismet.conf $(ETC)/kismet.conf;  \
 		echo "Installed config into $(ETC)/kismet.conf."; \
 	fi
 	@if test -f $(ETC)/kismet_drone.conf; then \
@@ -184,8 +184,8 @@
 		echo "if there have been any changes to the base config you will need"; \
 		echo "to add them to your config file."; \
     else \
-		$(INSTALL) -o $(INSTUSR) -g $(INSTGRP) -m 644 conf/kismet_drone.conf $(ETC)/kismet_drone.conf;  \
-		echo install -o $(INSTUSR) -g $(INSTGRP) -m 644 conf/kismet_drone.conf $(ETC)/kismet_drone.conf;  \
+		$(INSTALL) -m 644 conf/kismet_drone.conf $(ETC)/kismet_drone.conf;  \
+		echo install -m 644 conf/kismet_drone.conf $(ETC)/kismet_drone.conf;  \
 		echo "Installed drone config into $(ETC)/kismet_drone.conf."; \
 	fi
 	@echo "Installed kismet into $(BIN)/."
