--- pklock.c.orig	2007-05-17 20:12:20.000000000 -0700
+++ pklock.c	2007-05-17 20:14:14.000000000 -0700
@@ -81,8 +81,7 @@
 	if ((n = read(fd, locker, MAXNAME)) < 1)
 	{
 		lseek(fd, 0, SEEK_SET);
-/*		strcpy(locker, getlogin()); */
-		cuserid(locker);
+		strcpy(locker, getlogin());
 		strcat(locker+strlen(locker), "@");
 		gethostname(locker+strlen(locker), 64);
 		write(fd, locker, strlen(locker));
