--- tethereal/configure.in.orig	2006-02-22 17:04:35.000000000 -0800
+++ tethereal/configure.in	2006-02-22 17:05:43.000000000 -0800
@@ -443,6 +443,7 @@
 # We can't just check for <inttypes.h> - some systems have one that
 # doesn't define all the PRI[doxu]64 macros.
 #
+if test -z "$ac_ethereal_inttypes_h_defines_formats" ; then
 AC_CHECK_HEADERS(inttypes.h,
   [
     #
@@ -484,6 +485,7 @@
     #
     ac_ethereal_inttypes_h_defines_formats=no
   ])
+fi
 if test "$ac_ethereal_inttypes_h_defines_formats" = yes; then
   AC_DEFINE(INTTYPES_H_DEFINES_FORMATS,,[Define if <inttypes.h> defines PRI[doxu]64 macros])
 else
