--- pure-ftpd-1.0.22/configure.ac.orig	2009-04-26 12:19:40.000000000 +0000
+++ pure-ftpd-1.0.22/configure.ac	2009-04-29 05:48:51.772543471 +0000
@@ -909,176 +909,28 @@
 ])  
 
 AC_MSG_CHECKING(whether snprintf is C99 conformant)
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <stdio.h>
-#ifdef STDC_HEADERS
-# include <stdlib.h>
-# include <stddef.h>
-#else
-# if HAVE_STDLIB_H
-#  include <stdlib.h>
-# endif
-#endif
-#ifdef HAVE_UNISTD_H
-# include <unistd.h>
-#endif
-
-int main(void)
-{
-    char buf[4];
-    
-    (void) fprintf(fopen("conftestval", "w"), "%d\n",
-        (int) snprintf(buf, sizeof buf, "12345678"));
-    return 0;
-}
-]])],[CONF_SNPRINTF_TYPE=`cat conftestval`
-],[],[])
-AC_MSG_RESULT(done)
-if test "x$CONF_SNPRINTF_TYPE" = "x" ; then
-  AC_MSG_WARN(your operating system doesn't implement snprintf)
-else
+CONF_SNPRINTF_TYPE=8
   AC_DEFINE_UNQUOTED(CONF_SNPRINTF_TYPE, $CONF_SNPRINTF_TYPE, [return value of an overflowed snprintf])
-fi
 
 
 AC_MSG_CHECKING(whether getgroups 0 is sane)
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <stdio.h>
-#ifdef STDC_HEADERS
-# include <stdlib.h>
-# include <stddef.h>
-#else
-# if HAVE_STDLIB_H
-#  include <stdlib.h>
-# endif
-#endif
-#ifdef HAVE_UNISTD_H
-# include <unistd.h>
-#endif
-
-int main(void)
-{
-    return getgroups(0, NULL) <= 0;
-}
-]])],[
 AC_MSG_RESULT(yes)
 AC_DEFINE(SAFE_GETGROUPS_0,,[Define is getgroups(0, NULL) works on your system])
-],[AC_MSG_RESULT(no)
-],[])
 
 AC_MSG_CHECKING(whether realpath likes unreadable directories)
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <stdio.h>
-#ifdef STDC_HEADERS
-# include <stdlib.h>
-# include <stddef.h>
-#else
-# if HAVE_STDLIB_H
-#  include <stdlib.h>
-# endif
-#endif
-#include <sys/types.h>
-#include <sys/stat.h>
-#ifdef HAVE_UNISTD_H
-# include <unistd.h>
-#endif
-#ifdef HAVE_LIMITS_H
-# include <limits.h>
-#endif
-#ifdef HAVE_SYS_PARAM_H
-# include <sys/param.h>
-#endif
-
-#ifndef MAXPATHLEN
-# define MAXPATHLEN PATH_MAX
-#endif
-
-int main(void)
-{
-    char x[MAXPATHLEN];
-    
-    if (mkdir("x", 0300) != 0) {
-        return 1;
-    }
-    if (mkdir("x/y", 0300) != 0) {
-        rmdir("x");
-        return 2;
-    }    
-    if (chdir("x") != 0) {
-        rmdir("x/y");    
-        rmdir("x");
-        return 3;
-    }
-    if (realpath("y", x) == NULL) {
-        rmdir("y");
-        rmdir("../x");
-        return 4;
-    }
-    rmdir("y");
-    rmdir("../x");
-    
-    return 0;    
-}
-]])],[
 AC_MSG_RESULT(yes)
 AC_DEFINE(REALPATH_WORKS_WITH_UNREADABLE_DIRECTORIES,,
 [Define if realpath() works on unreadable directories])
-],[AC_MSG_RESULT(no)
-AC_DEFINE(USE_BUILTIN_REALPATH)
-],[])
 
 AC_MSG_CHECKING(whether you already have a standard MD5 implementation)
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <stdio.h>
-#include <stdio.h>
-#include <string.h>
-#include <sys/types.h>
-#include <md5.h>
-
-int main(void)
-{
-    MD5_CTX ctx;
-    char b[33];
-    
-    MD5Init(&ctx);
-    MD5Update(&ctx, (const unsigned char *) "test", 4U);
-    MD5End(&ctx, b);
-    b[32] = 0;
-
-    return strcasecmp(b, "098f6bcd4621d373cade4e832627b4f6");
-}
-]])],[
 AC_MSG_RESULT(yes)
 AC_DEFINE(USE_SYSTEM_CRYPT_MD5,,[Define if you already have standard
 MD5 functions])
-],[AC_MSG_RESULT(no)
-],[])
 
 AC_MSG_CHECKING(whether you already have a standard SHA1 implementation)
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <stdio.h>
-#include <string.h>
-#include <sys/types.h>
-#include <sha1.h>
-
-int main(void)
-{
-    SHA1_CTX ctx;
-    char b[41];
-    
-    SHA1Init(&ctx);
-    SHA1Update(&ctx, (const unsigned char *) "test", 4U);
-    SHA1End(&ctx, b);
-    b[40] = 0;
-    
-    return strcasecmp(b, "a94a8fe5ccb19ba61c4c0873d391e987982fbbd3");
-}
-]])],[
 AC_MSG_RESULT(yes)
 AC_DEFINE(USE_SYSTEM_CRYPT_SHA1,,[Define if you already have standard
 SHA1 functions])
-],[AC_MSG_RESULT(no)
-],[])
 
 AC_MSG_CHECKING([whether we are inside a Virtuozzo virtual host])
 if test -d /proc/vz; then
