--- a/Makefile.am
+++ b/Makefile.am
@@ -8,7 +8,7 @@
 CXXFLAGS+= -DCONFDIR=\"$(hplip_confdir)\"
 
 hplipdir = $(datadir)/hplip
-apparmordir = /etc/apparmor.d
+apparmordir = %OPTWARE_TARGET_PREFIX%/etc/apparmor.d
 docdir = $(hpdocdir)
 htmldir = $(hphtmldir)
 doc_DATA = COPYING copyright prnt/hpijs/README_LIBJPG
@@ -105,7 +105,6 @@
 libhpmud_la_LDFLAGS += -lusb
 else
 libhpmud_la_SOURCES += io/hpmud/musb.c 
-libhpmud_la_CFLAGS  += -I/usr/include/libusb-1.0
 libhpmud_la_LDFLAGS += -lusb-1.0
 endif
 
@@ -127,7 +126,7 @@
 	ip/hpip.h ip/ipdefs.h ip/xform.h ip/xjpg_dct.h ip/xjpg_huf.h ip/xjpg_mrk.h
 
 #hpmud rules data dir
-rulessystemdir=/usr/lib/systemd/system
+rulessystemdir=%OPTWARE_TARGET_PREFIX%/lib/systemd/system
 dist_rulessystem_DATA =data/rules/hplip-printer@.service
 
 # hpmud.rules
@@ -140,11 +139,11 @@
 endif
 endif #HPLIP_CLASS_DRIVER
 
-halpredir = /usr/share/hal/fdi/preprobe/10osvendor
+halpredir = %OPTWARE_TARGET_PREFIX%/share/hal/fdi/preprobe/10osvendor
 dist_halpre_DATA = data/rules/20-hplip-devices.fdi
 
 # hplip.conf
-hplip_confdir = /etc/hp
+hplip_confdir = %OPTWARE_TARGET_PREFIX%/etc/hp
 hplip_conf_DATA = hplip.conf
 
 if !HPLIP_CLASS_DRIVER
@@ -154,7 +153,7 @@
 endif #HPLIP_CLASS_DRIVER
 
 # hplip.state
-hplip_statedir = /var/lib/hp
+hplip_statedir = %OPTWARE_TARGET_PREFIX%/var/lib/hp
 dist_hplip_state_DATA =
 
 if FULL_BUILD
@@ -229,11 +228,11 @@
 if !HPLIP_CLASS_DRIVER
 # AppArmor Changes
 if APPARMOR_BUILD
-apparmor_profiledir = /etc/apparmor.d
+apparmor_profiledir = %OPTWARE_TARGET_PREFIX%/etc/apparmor.d
 apparmor_profile_DATA = apparmor/usr.share.hplip
 dist_apparmor_profile_DATA = apparmor/usr.share.hplip
 
-apparmor_abstractiondir = /etc/apparmor.d/abstractions
+apparmor_abstractiondir = %OPTWARE_TARGET_PREFIX%/etc/apparmor.d/abstractions
 apparmor_abstraction_DATA = apparmor/abstractions/hplip
 dist_apparmor_abstraction_DATA = apparmor/abstractions/hplip
 
@@ -358,9 +357,6 @@
 hpmudext_la_CFLAGS += -Iprotocol/discovery
 endif
 
-if !LIBUSB01_BUILD
-hpmudext_la_CFLAGS +=-I/usr/include/libusb-1.0 
-endif
 endif #!HPLIP_CLASS_DRIVER
 # ui (qt3)
 if GUI_BUILD
@@ -445,9 +441,9 @@
 endif #HPLIP_CLASS_DRIVER
 # PolicyKit
 if POLICYKIT_INSTALL
-policykit_dbus_etcdir = /etc/dbus-1/system.d
+policykit_dbus_etcdir = %OPTWARE_TARGET_PREFIX%/etc/dbus-1/system.d
 dist_policykit_dbus_etc_DATA = data/policykit/com.hp.hplip.conf
-policykit_dbus_sharedir = /usr/share/dbus-1/system-services
+policykit_dbus_sharedir = %OPTWARE_TARGET_PREFIX%/share/dbus-1/system-services
 dist_policykit_dbus_share_DATA = data/policykit/com.hp.hplip.service
 policykit_policydir = $(policykit_dir)
 dist_policykit_policy_DATA = data/policykit/com.hp.hplip.policy
@@ -680,13 +676,13 @@
 if !HPLIP_CLASS_DRIVER
 #	   If scanner build, add hpaio entry to sane dll.conf.
 	if [ "$(scan_build)" = "yes" ]; then \
-	   $(mkinstalldirs) $(DESTDIR)/etc/sane.d; \
-	   if [ ! -f $(DESTDIR)/etc/sane.d/dll.conf ]; then \
-		  touch $(DESTDIR)/etc/sane.d/dll.conf; \
+	   $(mkinstalldirs) $(DESTDIR)%OPTWARE_TARGET_PREFIX%/etc/sane.d; \
+	   if [ ! -f $(DESTDIR)%OPTWARE_TARGET_PREFIX%/etc/sane.d/dll.conf ]; then \
+		  touch $(DESTDIR)%OPTWARE_TARGET_PREFIX%/etc/sane.d/dll.conf; \
 	   fi; \
-	   if ! grep ^hpaio $(DESTDIR)/etc/sane.d/dll.conf >/dev/null 2>/dev/null ; then \
-		  echo "Adding hpaio entry to /etc/sane.d/dll.conf." ; \
-		  echo hpaio >>$(DESTDIR)/etc/sane.d/dll.conf ; \
+	   if ! grep ^hpaio $(DESTDIR)%OPTWARE_TARGET_PREFIX%/etc/sane.d/dll.conf >/dev/null 2>/dev/null ; then \
+		  echo "Adding hpaio entry to %OPTWARE_TARGET_PREFIX%/etc/sane.d/dll.conf." ; \
+		  echo hpaio >>$(DESTDIR)%OPTWARE_TARGET_PREFIX%/etc/sane.d/dll.conf ; \
 	   fi; \
  	   if [ \( "$(UNAME)" = "x86_64" -a  -d "$(libdir)/x86_64-linux-gnu/sane" \) ]; then \
 	  	ln -sf $(libdir)/sane/libsane-hpaio.so $(libdir)/x86_64-linux-gnu/sane/ ; \
