Index: emu/Linux/mkfile
===================================================================
--- emu/Linux/mkfile	(revision 221)
+++ emu/Linux/mkfile	(working copy)
@@ -4,8 +4,8 @@
 
 #Configurable parameters
 
-CONF=emu			#default configuration
-CONFLIST=emu
+CONF=emu-g			#default configuration
+CONFLIST=emu-g
 CLEANCONFLIST=
 
 INSTALLDIR=$ROOT/$SYSTARG/$OBJTYPE/bin	#path of directory where kernel is installed
@@ -30,7 +30,7 @@
 HFILES=\
 
 CFLAGS='-DROOT="'$ROOT'"' -DEMU -I. -I../port -I$ROOT/$SYSTARG/$OBJTYPE/include -I$ROOT/include -I$ROOT/libinterp $CTHREADFLAGS $CFLAGS $EMUOPTIONS
-SYSLIBS= -lm -lX11 -lXext
+SYSLIBS= -lm
 KERNDATE=`{$NDATE}
 
 default:V:	$O.$CONF
