--- configure.orig	2015-02-03 19:08:56.029794787 +0200
+++ configure	2015-02-03 19:12:39.117786627 +0200
@@ -5128,25 +5128,7 @@
 fi
 
 if test $DEFAULT_AGENT = iptables; then
-	# test for the 64 bit libs first (18.04.2004 - Thomas Zehetbauer <thomasz@hostmaster.org>)
-	if test -d /usr/lib64/xtables; then
-		XTABLES_LIBDIR=/usr/lib64/xtables
-	elif test -d /lib64/xtables; then
-		XTABLES_LIBDIR=/lib64/xtables
-	elif test -d /usr/local/lib64/xtables; then
-		XTABLES_LIBDIR=/usr/local/lib64/xtables
-	elif test -d /usr/lib/xtables; then
-		XTABLES_LIBDIR=/usr/lib/xtables
-	elif test -d /lib/xtables; then
-		XTABLES_LIBDIR=/lib/xtables
-	elif test -d /usr/local/lib/xtables; then
-		XTABLES_LIBDIR=/usr/local/lib/xtables
-	else
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: iptables library directory not found, define
-				it at config.h" >&5
-$as_echo "$as_me: WARNING: iptables library directory not found, define
-				it at config.h" >&2;}
-	fi
+	XTABLES_LIBDIR=%OPTWARE_TARGET_PREFIX%/lib/xtables
 	IPTSLIBS=""
 	IPTSINC=""
 	IPTSSRC=""
