--- a/rpc/compat.h
+++ b/rpc/compat.h
@@ -14,16 +14,12 @@
 #include <poll.h>
 #include <pthread.h>
 
-#ifdef __UCLIBC__
-#include <bits/libc-lock.h>
-#else
 #undef __UCLIBC_MUTEX_STATIC
 #undef __UCLIBC_MUTEX_LOCK
 #undef __UCLIBC_MUTEX_UNLOCK
 #define __UCLIBC_MUTEX_STATIC(M,I) static pthread_mutex_t M = I
 #define __UCLIBC_MUTEX_LOCK(M) pthread_mutex_lock(&(M))
 #define __UCLIBC_MUTEX_UNLOCK(M) pthread_mutex_unlock(&(M))
-#endif
 
 #define smallint int
 
