--- src/lib/HTTP/HTTPMessage.cpp.orig	2007-11-20 09:16:22.000000000 +0000
+++ src/lib/HTTP/HTTPMessage.cpp	2008-05-19 06:03:41.000000000 +0000
@@ -778,7 +778,7 @@
   if(rxContentLength.Search(p_sMessage.c_str()))
   {
     string sContentLength = rxContentLength.Match(1);    
-    m_nContentLength = std::atoll(sContentLength.c_str());
+    m_nContentLength = atoll(sContentLength.c_str());
   }
   
   if((unsigned int)m_nContentLength >= p_sMessage.length())                      
