--- libgphoto2-2.2.1.org/camlibs/sonix/library.c	2006-04-19 03:54:18.000000000 +0200
+++ libgphoto2/camlibs/sonix/library.c	2006-08-20 18:22:29.000000000 +0200
@@ -41,7 +41,7 @@
 #else
 #  define _(String) (String)
 #  define N_(String) (String)
-#  define ngettext(String1,String2,Count) ((Count==1)?String1:String)
+#  define ngettext(String1,String2,Count) ((Count==1)?String1:String2)
 #endif
 
 #include <gphoto2-port.h>
