--- clips/clipssrc/setup.h-orig	2005-05-20 22:35:37.474321000 -0700
+++ clips/clipssrc/setup.h	2005-05-20 22:35:47.194321000 -0700
@@ -35,10 +35,10 @@
 /* Only one of these flags should be turned on (set to 1) at a time. */
 /*********************************************************************/
 
-#define GENERIC 0   /* Generic (any machine)                  */
+#define GENERIC 1   /* Generic (any machine)                  */
 #define UNIX_V  0   /* UNIX System V, 4.2bsd, HP Unix, Darwin */
 #define UNIX_7  0   /* UNIX System III Version 7 or Sun Unix  */
-#define MAC_MCW 1   /* Apple Macintosh, with CodeWarrior 9.4  */
+#define MAC_MCW 0   /* Apple Macintosh, with CodeWarrior 9.4  */
 #define MAC_XCD 0   /* Apple Macintosh, with Xcode 1.2		  */
 #define IBM_MCW 0   /* IBM PC, with CodeWarrior 9.4           */
 #define IBM_MSC 0   /* IBM PC, with Microsoft Visual C++ 6.0  */
