--- portmap_4/strerror.c	1992-06-11 22:53:13.000000000 +0200
+++ portmap/strerror.c	2006-05-29 11:31:09.000000000 +0200
@@ -41,8 +41,6 @@
 strerror(num)
 	int num;
 {
-	extern int sys_nerr;
-	extern char *sys_errlist[];
 #define	UPREFIX	"Unknown error: "
 	static char ebuf[40] = UPREFIX;		/* 64-bit number + slop */
 	register unsigned int errnum;
@@ -50,8 +48,7 @@
 	char tmp[40];
 
 	errnum = num;				/* convert to unsigned */
-	if (errnum < sys_nerr)
-		return(sys_errlist[errnum]);
+	return(strerror(errnum));
 
 	/* Do this by hand, so we don't include stdio(3). */
 	t = tmp;
