--- src/data/socket_file.cc.orig	2007-07-19 14:23:06.000000000 -0700
+++ src/data/socket_file.cc	2007-09-12 16:22:15.000000000 -0700
@@ -160,7 +160,14 @@
 
   uint64_t align = offset % MemoryChunk::page_size();
 
-  char* ptr = (char*)mmap(NULL, length + align, prot, flags, m_fd, offset - align);
+  offset = offset - align;
+
+#if __GLIBC__ == 2 && __GLIBC_MINOR__ < 3
+#warning "swap word of 64-bit offset to work around mmap64 bug of glibc"
+  offset = (offset >> 32) | (offset << 32);
+#endif
+
+  char* ptr = (char*)mmap(NULL, length + align, prot, flags, m_fd, offset);
   
   if (ptr == MAP_FAILED)
     return MemoryChunk();
