--- m4/am-check-python-headers.m4.orig	2007-07-31 11:10:27.000000000 -0700
+++ m4/am-check-python-headers.m4	2009-12-27 22:35:28.000000000 -0800
@@ -6,20 +6,7 @@
 dnl deduce PYTHON_INCLUDES
 
 AC_MSG_CHECKING(for Python headers using $PYTHON-config --includes)
-PYTHON_INCLUDES=`$PYTHON-config --includes`
-if test $? = 0; then
-  AC_MSG_RESULT($PYTHON_INCLUDES)
-else
-  AC_MSG_RESULT([failed, will try another way])
-  py_prefix=`$PYTHON -c "import sys; print sys.prefix"`
-  py_exec_prefix=`$PYTHON -c "import sys; print sys.exec_prefix"`
-  AC_MSG_CHECKING(for Python headers in $py_prefix and $py_exec_prefix)
-  PYTHON_INCLUDES="-I${py_prefix}/include/python${PYTHON_VERSION}"
-  if test "$py_prefix" != "$py_exec_prefix"; then
-    PYTHON_INCLUDES="$PYTHON_INCLUDES -I${py_exec_prefix}/include/python${PYTHON_VERSION}"
-  fi
-  AC_MSG_RESULT($PYTHON_INCLUDES)
-fi
+  PYTHON_INCLUDES="-I${staging_prefix}/include/python${PYTHON_VERSION}"
 
 AC_MSG_CHECKING(whether those headers are sufficient)
 AC_SUBST(PYTHON_INCLUDES)
