diff -urN thttpd/Makefile.in thttpd-patched/Makefile.in
--- thttpd/Makefile.in	Tue Apr  2 21:49:35 2002
+++ thttpd-patched/Makefile.in	Tue Apr  5 12:14:14 2005
@@ -99,9 +99,10 @@
 	    cd $$i ; \
 	    pwd ; \
 	    $(MAKE) $(MFLAGS) \
-		WEBDIR=$(WEBDIR) \
-		CGIBINDIR=$(CGIBINDIR) \
-		MANDIR=$(MANDIR) \
+		WEBDIR=$(DESTDIR)$(WEBDIR) \
+		CGIBINDIR=$(DESTDIR)$(CGIBINDIR) \
+		MANDIR=$(DESTDIR)$(MANDIR) \
+		BINDIR=$(DESTDIR)$(BINDIR) \
 		WEBGROUP=$(WEBGROUP) \
 	) ; done
 
@@ -121,9 +122,10 @@
 	    cd $$i ; \
 	    pwd ; \
 	    $(MAKE) $(MFLAGS) \
-		WEBDIR=$(WEBDIR) \
-		CGIBINDIR=$(CGIBINDIR) \
-		MANDIR=$(MANDIR) \
+		WEBDIR=$(DESTDIR)$(WEBDIR) \
+		CGIBINDIR=$(DESTDIR)$(CGIBINDIR) \
+		MANDIR=$(DESTDIR)$(MANDIR) \
+		BINDIR=$(DESTDIR)$(BINDIR) \
 		WEBGROUP=$(WEBGROUP) \
 		install \
 	) ; done
diff -urN thttpd/cgi-src/Makefile.in thttpd-patched/cgi-src/Makefile.in
--- thttpd/cgi-src/Makefile.in	Mon Nov 12 16:52:00 2001
+++ thttpd-patched/cgi-src/Makefile.in	Tue Apr  5 12:17:25 2005
@@ -26,9 +26,9 @@
 
 prefix =	@prefix@
 exec_prefix =	@exec_prefix@
-WEBDIR =	$(prefix)/www
-CGIBINDIR =	$(WEBDIR)/cgi-bin
-MANDIR =	@mandir@
+#WEBDIR =	$(prefix)/www
+#CGIBINDIR =	$(WEBDIR)/cgi-bin
+#MANDIR =	@mandir@
 
 CC =		@CC@
 CCOPT =		@V_CCOPT@
@@ -68,15 +68,11 @@
 
 install:	all
 	-mkdir -p $(CGIBINDIR)
-	rm -f $(CGIBINDIR)/redirect
 	cp redirect $(CGIBINDIR)/redirect
-	rm -f $(MANDIR)/man8/redirect.8
+	-mkdir -p $(MANDIR)/man8
 	cp redirect.8 $(MANDIR)/man8/redirect.8
-	rm -f $(CGIBINDIR)/ssi
 	cp ssi $(CGIBINDIR)/ssi
-	rm -f $(MANDIR)/man8/ssi.8
 	cp ssi.8 $(MANDIR)/man8/ssi.8
-	rm -f $(CGIBINDIR)/phf
 	cp phf $(CGIBINDIR)/phf
 
 clean:
diff -urN thttpd/extras/Makefile.in thttpd-patched/extras/Makefile.in
--- thttpd/extras/Makefile.in	Mon Nov 12 16:52:55 2001
+++ thttpd-patched/extras/Makefile.in	Tue Apr  5 12:16:22 2005
@@ -26,10 +26,10 @@
 
 prefix =	@prefix@
 exec_prefix =	@exec_prefix@
-BINDIR =	@sbindir@
-WEBDIR =	$(prefix)/www
-CGIBINDIR =	$(WEBDIR)/cgi-bin
-MANDIR =	@mandir@
+#BINDIR =	@sbindir@
+#WEBDIR =	$(prefix)/www
+#CGIBINDIR =	$(WEBDIR)/cgi-bin
+#MANDIR =	@mandir@
 
 CC =		@CC@
 CCOPT =		@V_CCOPT@
@@ -66,17 +66,15 @@
 
 
 install:	all
-	rm -f $(BINDIR)/makeweb $(BINDIR)/htpasswd $(BINDIR)/syslogtocern
+	-mkdir -p (BINDIR)
 	cp makeweb $(BINDIR)/makeweb
-	chgrp $(WEBGROUP) $(BINDIR)/makeweb
 	chmod 2755 $(BINDIR)/makeweb
 	cp htpasswd $(BINDIR)/htpasswd
 	cp syslogtocern $(BINDIR)/syslogtocern
-	rm -f $(MANDIR)/man1/makeweb.1
+	-mkdir -p $(MANDIR)/man1
 	cp makeweb.1 $(MANDIR)/man1/makeweb.1
-	rm -f $(MANDIR)/man1/htpasswd.1
 	cp htpasswd.1 $(MANDIR)/man1/htpasswd.1
-	rm -f $(MANDIR)/man8/syslogtocern.8
+	-mkdir -p $(MANDIR)/man8
 	cp syslogtocern.8 $(MANDIR)/man8/syslogtocern.8
 
 clean:
