diff -ur smartmontools-5.37/smartctl.cpp smartmontools/smartctl.cpp
--- smartmontools-5.37/smartctl.cpp	2006-10-09 13:45:12.000000000 +0200
+++ smartmontools/smartctl.cpp	2007-01-04 18:15:04.000000000 +0100
@@ -82,7 +82,7 @@
 }
 
 void printcopy(){
-  char *configargs=strlen(SMARTMONTOOLS_CONFIGURE_ARGS)?SMARTMONTOOLS_CONFIGURE_ARGS:"[no arguments given]";
+  const char *configargs=strlen(SMARTMONTOOLS_CONFIGURE_ARGS)?SMARTMONTOOLS_CONFIGURE_ARGS:"[no arguments given]";
 
   pout("smartctl comes with ABSOLUTELY NO WARRANTY. This\n");
   pout("is free software, and you are welcome to redistribute it\n");
diff -ur smartmontools-5.37/smartd.cpp smartmontools/smartd.cpp
--- smartmontools-5.37/smartd.cpp	2006-11-12 05:49:09.000000000 +0100
+++ smartmontools/smartd.cpp	2007-01-04 18:14:40.000000000 +0100
@@ -310,7 +310,7 @@
 
 // prints CVS identity information for the executable
 void PrintCVS(void){
-  char *configargs=strlen(SMARTMONTOOLS_CONFIGURE_ARGS)?SMARTMONTOOLS_CONFIGURE_ARGS:"[no arguments given]";
+  const char *configargs=strlen(SMARTMONTOOLS_CONFIGURE_ARGS)?SMARTMONTOOLS_CONFIGURE_ARGS:"[no arguments given]";
 
   PrintOut(LOG_INFO,(char *)copyleftstring);
   PrintOut(LOG_INFO,"CVS version IDs of files used to build this code are:\n");
