--- asterisk/dns.c.orig	2007-08-12 12:50:23.000000000 -0700
+++ asterisk/dns.c	2007-08-12 12:55:46.000000000 -0700
@@ -237,7 +237,14 @@
 #if defined(res_ndestroy)
 #define HAS_RES_NDESTROY
 #endif
-#else
+#endif
+
+#ifdef __UCLIBC__
+#undef HAS_RES_NINIT
+#undef HAS_RES_NDESTROY
+#endif
+
+#ifndef HAS_RES_NINIT
 AST_MUTEX_DEFINE_STATIC(res_lock);
 #if 0
 #warning "Warning, res_ninit is missing...  Could have reentrancy issues"
