--- asterisk-1.0.99/codecs/gsm/Makefile~	2005-04-02 11:29:21.000000000 -0800
+++ asterisk-1.0.99/codecs/gsm/Makefile	2005-04-02 11:55:15.000000000 -0800
@@ -284,6 +284,7 @@
 ifneq ($(shell uname -m), ppc64)
 ifneq ($(shell uname -m), alpha)
 ifneq ($(shell uname -m), sparc64)
+ifneq (${PROC},arm)
 GSM_OBJECTS+= $(SRC)/k6opt.o
 endif
 endif
@@ -291,6 +292,7 @@
 endif
 endif
 endif
+endif
 
 TOAST_OBJECTS =	$(SRC)/toast.o 		\
 		$(SRC)/toast_lin.o	\
