--- php-5.2.4/sapi/thttpd/config.m4.orig	2007-07-11 16:20:37.000000000 -0700
+++ php-5.2.4/sapi/thttpd/config.m4	2007-09-25 22:59:13.000000000 -0700
@@ -18,6 +18,10 @@
     patch="test -f $THTTPD/php_patched || \
     (cd $THTTPD && patch -p1 < $abs_srcdir/sapi/thttpd/thttpd_patch && touch php_patched)"
 
+  elif grep thttpd.2.25b $PHP_THTTPD/version.h >/dev/null; then
+    patch="test -f $THTTPD/php_patched || \
+    (cd $THTTPD && patch -p1 < $abs_srcdir/sapi/thttpd/thttpd-2.25b_patch && touch php_patched)"
+
   elif grep Premium $PHP_THTTPD/version.h >/dev/null; then
     patch=
   else
