--- nmap/libpcre/Makefile.in.orig	2005-01-08 13:10:58.000000000 +0100
+++ nmap/libpcre/Makefile.in	2005-01-08 13:11:18.000000000 +0100
@@ -105,11 +105,6 @@
 		pcre.h config.h Makefile
 		$(CC) -c $(CFLAGS) $(INCLS) $(DEFS) $(UTF8) $(top_srcdir)/study.c
 
-# An auxiliary program makes the default character table source
-
-$(top_srcdir)/chartables.c:   dftables
-		./dftables >$(top_srcdir)/chartables.c
-
 dftables.@BUILD_OBJEXT@:     $(top_srcdir)/dftables.c $(top_srcdir)/maketables.c \
 		$(top_srcdir)/internal.h pcre.h config.h Makefile
 		$(CC) -c $(CFLAGS) $(INCLS) $(DEFS) -I. $(top_srcdir)/dftables.c
