--- clamav/configure.in.orig	2007-08-20 15:58:17.000000000 -0700
+++ clamav/configure.in	2007-12-08 10:14:20.000000000 -0800
@@ -67,7 +67,9 @@
 AC_FUNC_FSEEKO
 
 dnl Check for broken snprintf (code by Phil Oleson <oz*nixil.net>)
-if test "x$ac_cv_func_snprintf" = "xyes" ; then
+if test "x$cross_compiling" = "xyes"; then
+	AC_MSG_WARN([Cross compiling: snprintf not tested])
+else if test "x$ac_cv_func_snprintf" = "xyes" ; then
         AC_MSG_CHECKING([whether snprintf correctly terminates long strings])
         AC_TRY_RUN(
                 [
@@ -82,6 +84,7 @@
                 ]
         )
 fi
+fi
 
 have_pthreads=no
 AC_CHECK_HEADER(pthread.h,[have_pthreads=yes],)
@@ -329,18 +332,20 @@
 AC_DEFINE_UNQUOTED(CONFDIR,"$cfg_dir",[where to look for the config file])
 
 dnl check for in_port_t definition
-AC_TRY_RUN([
+AC_TRY_COMPILE([
 #include <sys/types.h>
 #include <netinet/in.h>
-int main(int argc, char **argv) { in_port_t pt; pt = 0; return pt; }
-], AC_DEFINE(HAVE_IN_PORT_T,1,[in_port_t is defined]), AC_MSG_RESULT(in_port_t is not defined))
+],[
+in_port_t pt; pt = 0;
+], AC_DEFINE(HAVE_IN_PORT_T,1,[in_port_t is defined]))
 
 dnl check for in_addr_t definition
-AC_TRY_RUN([
+AC_TRY_COMPILE([
 #include <sys/types.h>
 #include <netinet/in.h>
-int main(int argc, char **argv) { in_addr_t pt; pt = 0; return pt; }
-], AC_DEFINE(HAVE_IN_ADDR_T,1,[in_addr_t is defined]), AC_MSG_RESULT(in_addr_t is not defined))
+],[
+in_addr_t pt; pt = 0;
+], AC_DEFINE(HAVE_IN_ADDR_T,1,[in_addr_t is defined]))
 
 case "$target_os" in
 linux*)
