--- a/setup.py
+++ b/setup.py
@@ -86,27 +86,7 @@
         self.set_undefined_options('build', ('build_lib', 'build_lib'))
 
     def run(self):
-        if self.develop:
-            js_basedir = os.path.join(os.path.dirname(__file__), 'deluge', 'ui', 'web', 'js')
-        else:
-            js_basedir = os.path.join(self.build_lib, 'deluge', 'ui', 'web', 'js')
-
-        js_source_dirs = [os.path.join(js_basedir, 'deluge-all'),
-                          os.path.join(js_basedir, 'extjs', 'ext-extensions')]
-
-        import_error = ''
-        try:
-            from minify_web_js import minify_js_dir
-        except ImportError as err:
-            import_error = err
-
-        for source_dir in js_source_dirs:
-            # If unable to import minify script and there is no existing minified file, raise error.
-            if import_error:
-                js_file = os.path.join(os.path.dirname(source_dir), os.path.basename(source_dir)) + '.js'
-                if not os.path.exists(js_file):
-                    raise ImportError(import_error)
-            minify_js_dir(source_dir)
+        js_basedir = os.path.join(os.path.dirname(__file__), 'deluge', 'ui', 'web', 'js')
 
 
 class BuildTranslations(cmd.Command):
