--- a/setup.py
+++ b/setup.py
@@ -114,25 +114,6 @@
     def run(self):
         js_basedir = os.path.join(os.path.dirname(__file__), self.JS_DIR)
 
-        try:
-            from minify_web_js import minify_js_dir
-            import_error = ''
-        except ImportError as err:
-            import_error = err
-
-        for js_src_dir in self.JS_SRC_DIRS:
-            source_dir = os.path.join(js_basedir, js_src_dir)
-            try:
-                minify_js_dir(source_dir)
-            except NameError:
-                js_file = source_dir + '.js'
-                if os.path.isfile(js_file):
-                    print('Unable to minify but found existing minified: {}'.format(js_file))
-                else:
-                    # Unable to minify and no existing minified file found so exiting.
-                    print('Import error: %s' % import_error)
-                    sys.exit(1)
-
         # Create the gettext.js file for translations.
         try:
             from gen_web_gettext import create_gettext_js
