diff --git a/version.py b/version.py
index 276b5c5..40ecc20 100644
--- a/version.py
+++ b/version.py
@@ -40,7 +40,7 @@
 def call_git_describe(prefix="", suffix=""):
     cmd = "git describe --tags --match %s[0-9]*" % prefix
     try:
-        version = Popen(cmd.split(), stdout=PIPE).communicate()[0]
+        version = Popen(cmd.split(), stdout=PIPE, stderr=PIPE).communicate()[0]
         version = version.strip().replace(prefix, "")
         if "-" in version:
             version = ".dev".join(version.replace(suffix, "").split("-")[:2])
@@ -58,9 +58,9 @@ def get_version(prefix="", suffix=""):
 
     version = call_git_describe(prefix, suffix)
 
-    if version is None:
+    if not version:
         version = release_version
-    if version is None:
+    if not version:
         raise ValueError("Cannot find the version number!")
 
     if version != release_version:
