--- cyrus-imapd-2.3.16/perl/Makefile.in.orig	2009-05-06 16:48:04.000000000 +0300
+++ cyrus-imapd-2.3.16/perl/Makefile.in	2015-02-13 11:46:02.283727098 +0200
@@ -74,6 +74,9 @@
 cyrus_prefix = @cyrus_prefix@
 
 CC = @CC@
+CPPFLAGS=@CPPFLAGS@
+LDFLAGS=@LDFLAGS@
+
 PLMAKE = imap/Makefile
 
 all::
@@ -81,6 +84,8 @@
 	do \
 		(cd $$d; echo "### Making" all "in" `pwd`;	\
 			if [ -f Makefile.PL -a ! -f Makefile ]; then \
+			   CPPFLAGS="$(CPPFLAGS)" \
+			   LDFLAGS="$(LDFLAGS)" \
 			   LIB_RT="$(LIB_RT)" \
 			   ZLIB="$(ZLIB)" \
 			   BDB_LIB="$(BDB_LIB)" BDB_INC="$(BDB_INC)" \
--- cyrus-imapd-2.3.16/perl/sieve/Makefile.in.orig	2009-05-06 16:48:04.000000000 +0300
+++ cyrus-imapd-2.3.16/perl/sieve/Makefile.in	2015-02-13 11:48:23.667378372 +0200
@@ -74,6 +74,9 @@
 ZLIB=@ZLIB@
 
 CC=@CC@
+CPPFLAGS=@CPPFLAGS@
+LDFLAGS=@LDFLAGS@
+
 PERL=@PERL@
 
 all::
@@ -81,6 +84,8 @@
 	do \
 		(cd $$d; echo "### Making" all "in" `pwd`;	\
 			if [ -f Makefile.PL -a ! -f Makefile ]; then \
+			   CPPFLAGS="$(CPPFLAGS)" \
+			   LDFLAGS="$(LDFLAGS)" \
 			   LIB_RT="$(LIB_RT)" \
 			   ZLIB="$(ZLIB)" \
 			   BDB_LIB="$(BDB_LIB)" BDB_INC="$(BDB_INC)" \
